<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="caitlinpage" />

<meta name="date" content="2025-09-24" />

<title>bedFileBetas</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">3_cf_nano</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/caitlinpage/methyl_nano_cf">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">bedFileBetas</h1>
<h4 class="author">caitlinpage</h4>
<h4 class="date">2025-09-24</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-10-03
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>methyl_nano_cf/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250606code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250606)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250606code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250606)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomcaitlinpagemethylnanocftree4aacc2b7a356630975e4383888dfa49e1be72703targetblank4aacc2ba">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/caitlinpage/methyl_nano_cf/tree/4aacc2b7a356630975e4383888dfa49e1be72703" target="_blank">4aacc2b</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomcaitlinpagemethylnanocftree4aacc2b7a356630975e4383888dfa49e1be72703targetblank4aacc2ba"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/caitlinpage/methyl_nano_cf/tree/4aacc2b7a356630975e4383888dfa49e1be72703" target="_blank">4aacc2b</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    renv.lock
    Ignored:    renv/

Unstaged changes:
    Modified:   .Rprofile
    Modified:   .gitignore
    Modified:   analysis/joinAlphaAndBeta.Rmd
    Modified:   analysis/plotAlphaAndBeta.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/bedFileBetas.Rmd</code>) and HTML
(<code>docs/bedFileBetas.html</code>) files. If you’ve configured a
remote Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/caitlinpage/methyl_nano_cf/blob/4aacc2b7a356630975e4383888dfa49e1be72703/analysis/bedFileBetas.Rmd" target="_blank">4aacc2b</a>
</td>
<td>
caitlinpage
</td>
<td>
2025-10-03
</td>
<td>
wflow_publish("analysis/bedFileBetas.Rmd")
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/caitlinpage/methyl_nano_cf/blob/58ee5a3368b68425405fa87c23639f35381a6e16/analysis/bedFileBetas.Rmd" target="_blank">58ee5a3</a>
</td>
<td>
caitlinpage
</td>
<td>
2025-09-25
</td>
<td>
init commit - clean version of all ways I’ve tried to get betas
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="description" class="section level2">
<h2>Description</h2>
<p>So I’ve identified in a few files that weird things are happening
with bed files and beta files and beta values and them not lining up as
expected with alpha or just not lining up with positions</p>
<p>so making this the one stop shop for bed/beta organisation (because
my solution to tidy things is always make another file)</p>
<ul>
<li>first found problem in plotAlphaAndBeta.html</li>
<li>in bed_betas - which is the bed files</li>
<li>and position</li>
</ul>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<pre class="r"><code>library(data.table)
library(plyranges)</code></pre>
<pre><code>Loading required package: BiocGenerics</code></pre>
<pre><code>
Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    IQR, mad, sd, var, xtabs</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    anyDuplicated, aperm, append, as.data.frame, basename, cbind,
    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,
    get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,
    match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,
    Position, rank, rbind, Reduce, rownames, sapply, saveRDS, setdiff,
    table, tapply, union, unique, unsplit, which.max, which.min</code></pre>
<pre><code>Loading required package: IRanges</code></pre>
<pre><code>Loading required package: S4Vectors</code></pre>
<pre><code>Loading required package: stats4</code></pre>
<pre><code>
Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:data.table&#39;:

    first, second</code></pre>
<pre><code>The following object is masked from &#39;package:utils&#39;:

    findMatches</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    expand.grid, I, unname</code></pre>
<pre><code>
Attaching package: &#39;IRanges&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:data.table&#39;:

    shift</code></pre>
<pre><code>Loading required package: GenomicRanges</code></pre>
<pre><code>Loading required package: GenomeInfoDb</code></pre>
<pre><code>Warning: replacing previous import &#39;S4Arrays::makeNindexFromArrayViewport&#39; by
&#39;DelayedArray::makeNindexFromArrayViewport&#39; when loading &#39;SummarizedExperiment&#39;</code></pre>
<pre><code>
Attaching package: &#39;plyranges&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:IRanges&#39;:

    slice</code></pre>
<pre><code>The following object is masked from &#39;package:data.table&#39;:

    between</code></pre>
<pre><code>The following object is masked from &#39;package:stats&#39;:

    filter</code></pre>
<pre class="r"><code>library(tidyr)</code></pre>
<pre><code>
Attaching package: &#39;tidyr&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:S4Vectors&#39;:

    expand</code></pre>
<pre class="r"><code>library(reshape2)</code></pre>
<pre><code>
Attaching package: &#39;reshape2&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:tidyr&#39;:

    smiths</code></pre>
<pre><code>The following objects are masked from &#39;package:data.table&#39;:

    dcast, melt</code></pre>
<pre class="r"><code>library(dplyr)</code></pre>
<pre><code>
Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:plyranges&#39;:

    between, n, n_distinct</code></pre>
<pre><code>The following objects are masked from &#39;package:GenomicRanges&#39;:

    intersect, setdiff, union</code></pre>
<pre><code>The following object is masked from &#39;package:GenomeInfoDb&#39;:

    intersect</code></pre>
<pre><code>The following objects are masked from &#39;package:IRanges&#39;:

    collapse, desc, intersect, setdiff, slice, union</code></pre>
<pre><code>The following objects are masked from &#39;package:S4Vectors&#39;:

    first, intersect, rename, setdiff, setequal, union</code></pre>
<pre><code>The following objects are masked from &#39;package:BiocGenerics&#39;:

    combine, intersect, setdiff, union</code></pre>
<pre><code>The following objects are masked from &#39;package:data.table&#39;:

    between, first, last</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    filter, lag</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    intersect, setdiff, setequal, union</code></pre>
<pre class="r"><code>library(ggplot2)</code></pre>
<pre class="r"><code>all_samples &lt;- readRDS(&quot;/researchers/caitlin.page/cf_nano/r_output/all_samples.rds&quot;)
overlap_bins_reads &lt;- readRDS(&quot;/researchers/caitlin.page/cf_nano/r_output/overlap_bins_reads.rds&quot;)
cg_sites &lt;- readRDS(&quot;/researchers/caitlin.page/cf_nano/r_output/cg_sites.rds&quot;)
bed_betas &lt;- readRDS(&quot;/researchers/caitlin.page/cf_nano/r_output/bed_betas.rds&quot;)
wt_beta_chr22 &lt;- readRDS(&quot;/researchers/caitlin.page/cf_nano/r_output/wt_beta_chr22.rds&quot;)
indiv_reads &lt;- readRDS(&quot;/researchers/caitlin.page/cf_nano/r_output/indiv_reads.rds&quot;)</code></pre>
<ul>
<li>bed_betas : just beta value and position from available data
downloaded</li>
<li>wt_beta : wgbs_tools beta values</li>
<li>modkit_bed</li>
</ul>
<p>USING: bed_betas (downloaded data)</p>
<p>WHY: * something wrong with wt_beta - unmeth is always 0 (clearly
wrong) * modkit_bed has weird position things happening that I don’t
like</p>
<ul>
<li>can see this in (checkAlphaAndBet.html)</li>
</ul>
</div>
<div id="bed_betas" class="section level1">
<h1>bed_betas</h1>
<pre class="r"><code>bed_betas$pos &lt;- paste0(bed_betas$seqnames, &quot;-&quot;, bed_betas$start)</code></pre>
<pre class="r"><code>bed_betas_hu5.10 &lt;- bed_betas %&gt;% filter(sample == &quot;hu5.10&quot;)</code></pre>
<ul>
<li>decision from (checkAlphaAndBeta.html) beta values match expected
beta values (calculated from indiv_reads) when expected beta is
calculated by filtering to reads with a call probability of &gt; 0.66
this value was selected as it’s what <a
href="https://github.com/epi2me-labs/modbam2bed">modbam2bed</a> default
filtering option is which is what the <a
href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-022-02710-1#Sec10">paper</a>
used to make the processed bed file they used in their own analysis</li>
</ul>
<pre class="r"><code>overlaps_bed_reads &lt;- find_overlaps(as_granges(bed_betas_hu5.10), as_granges(indiv_reads)) %&gt;% data.frame()
overlaps_bed_reads$pos &lt;- paste0(overlaps_bed_reads$seqnames, &quot;-&quot;, overlaps_bed_reads$start)

overlaps_bed_reads_filt &lt;- overlaps_bed_reads %&gt;% 
  filter(call_prob &gt; 0.66) %&gt;% 
  group_by(ref_position) %&gt;% 
  mutate(expec_beta_filt = mean(call_code == &quot;m&quot;)) %&gt;% 
  ungroup() %&gt;% 
  mutate(match_beta_filt = ifelse(beta == expec_beta_filt, TRUE, FALSE)) %&gt;% 
  filter(match_beta_filt == TRUE)

nrow(overlaps_bed_reads)</code></pre>
<pre><code>[1] 992941</code></pre>
<pre class="r"><code>nrow(overlaps_bed_reads_filt)</code></pre>
<pre><code>[1] 677646</code></pre>
<pre class="r"><code>head(overlaps_bed_reads_filt)</code></pre>
<pre><code># A tibble: 6 × 36
  seqnames   start    end width strand  name sample type   beta bed_sample pos  
  &lt;fct&gt;      &lt;int&gt;  &lt;int&gt; &lt;int&gt; &lt;fct&gt;  &lt;dbl&gt; &lt;chr&gt;  &lt;chr&gt; &lt;dbl&gt; &lt;chr&gt;      &lt;chr&gt;
1 chr22     1.05e7 1.05e7     1 *          0 hu5.10 heal…     0 HU005.10.  chr2…
2 chr22     1.05e7 1.05e7     1 *          1 hu5.10 heal…     1 HU005.10.  chr2…
3 chr22     1.05e7 1.05e7     1 *          1 hu5.10 heal…     1 HU005.10.  chr2…
4 chr22     1.05e7 1.05e7     1 *          1 hu5.10 heal…     1 HU005.10.  chr2…
5 chr22     1.05e7 1.05e7     1 *          1 hu5.10 heal…     1 HU005.10.  chr2…
6 chr22     1.05e7 1.05e7     1 *          0 hu5.10 heal…     0 HU005.10.  chr2…
# ℹ 25 more variables: read_id &lt;chr&gt;, forward_read_position &lt;int&gt;,
#   ref_position &lt;int&gt;, chrom &lt;chr&gt;, mod_strand &lt;chr&gt;, ref_strand &lt;chr&gt;,
#   ref_mod_strand &lt;chr&gt;, fw_soft_clipped_start &lt;int&gt;,
#   fw_soft_clipped_end &lt;int&gt;, alignment_start &lt;int&gt;, alignment_end &lt;int&gt;,
#   read_length &lt;int&gt;, call_prob &lt;dbl&gt;, call_code &lt;chr&gt;, base_qual &lt;int&gt;,
#   ref_kmer &lt;chr&gt;, query_kmer &lt;chr&gt;, canonical_base &lt;chr&gt;,
#   modified_primary_base &lt;chr&gt;, fail &lt;lgl&gt;, inferred &lt;lgl&gt;, …</code></pre>
</div>
<div id="wt_beta" class="section level1">
<h1>wt_beta</h1>
<pre class="r"><code>wt_beta_chr22[1:5,]</code></pre>
<pre><code>             pos seqnames    start      end HU005.10.meth HU005.10.cov
1 chr22-10510235    chr22 10510235 10510236             0            0
2 chr22-10510275    chr22 10510275 10510276             0            0
3 chr22-10510284    chr22 10510284 10510285             0            0
4 chr22-10510338    chr22 10510338 10510339             0            0
5 chr22-10510437    chr22 10510437 10510438             0            0
  HU005.10.unmeth HU005.10.beta HU005.11.meth HU005.11.cov HU005.11.unmeth
1               0           NaN             0            0               0
2               0           NaN             0            0               0
3               0           NaN             0            0               0
4               0           NaN             0            0               0
5               0           NaN             0            0               0
  HU005.11.beta HU005.12.meth HU005.12.cov HU005.12.unmeth HU005.12.beta
1           NaN             0            0               0           NaN
2           NaN             0            0               0           NaN
3           NaN             0            0               0           NaN
4           NaN             0            0               0           NaN
5           NaN             0            0               0           NaN
  ISPRO.bc01.meth ISPRO.bc01.cov ISPRO.bc01.unmeth ISPRO.bc01.beta
1               1              3                 2       0.3333333
2               0              3                 3       0.0000000
3               0              3                 3       0.0000000
4               0              2                 2       0.0000000
5               0              0                 0             NaN
  ISPRO.bc02.meth ISPRO.bc02.cov ISPRO.bc02.unmeth ISPRO.bc02.beta
1               0              0                 0             NaN
2               0              0                 0             NaN
3               0              0                 0             NaN
4               0              0                 0             NaN
5               0              0                 0             NaN
  ISPRO.bc03.meth ISPRO.bc03.cov ISPRO.bc03.unmeth ISPRO.bc03.beta
1               0              0                 0             NaN
2               0              0                 0             NaN
3               0              0                 0             NaN
4               0              0                 0             NaN
5               0              0                 0             NaN
  ISPRO.bc04.meth ISPRO.bc04.cov ISPRO.bc04.unmeth ISPRO.bc04.beta
1               0              2                 2             0.0
2               1              2                 1             0.5
3               1              2                 1             0.5
4               1              1                 0             1.0
5               0              0                 0             NaN
  ISPRO.bc05.meth ISPRO.bc05.cov ISPRO.bc05.unmeth ISPRO.bc05.beta
1               0              0                 0             NaN
2               0              0                 0             NaN
3               0              0                 0             NaN
4               0              0                 0             NaN
5               0              1                 1               0
  ISPRO.bc08.meth ISPRO.bc08.cov ISPRO.bc08.unmeth ISPRO.bc08.beta
1               1              1                 0               1
2               0              1                 1               0
3               0              1                 1               0
4               0              0                 0             NaN
5               0              0                 0             NaN
  ISPRO.bc09.meth ISPRO.bc09.cov ISPRO.bc09.unmeth ISPRO.bc09.beta
1               0              0                 0             NaN
2               0              0                 0             NaN
3               0              0                 0             NaN
4               0              0                 0             NaN
5               0              0                 0             NaN
  ISPRO.bc10.meth ISPRO.bc10.cov ISPRO.bc10.unmeth ISPRO.bc10.beta
1               0              0                 0             NaN
2               0              0                 0             NaN
3               0              0                 0             NaN
4               0              0                 0             NaN
5               0              0                 0             NaN
  ISPRO.bc11.meth ISPRO.bc11.cov ISPRO.bc11.unmeth ISPRO.bc11.beta
1               0              0                 0             NaN
2               0              0                 0             NaN
3               0              0                 0             NaN
4               0              0                 0             NaN
5               0              0                 0             NaN
  ISPRO.S1.meth ISPRO.S1.cov ISPRO.S1.unmeth ISPRO.S1.beta
1             2            3               1     0.6666667
2             1            2               1     0.5000000
3             2            2               0     1.0000000
4             0            0               0           NaN
5             0            1               1     0.0000000</code></pre>
<pre class="r"><code>summary(wt_beta_chr22$HU005.10.meth)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.0000  0.0000  0.0000  0.1674  0.0000  8.0000 </code></pre>
<pre class="r"><code>summary(wt_beta_chr22$HU005.10.unmeth)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
      0       0       0       0       0       0 </code></pre>
<pre class="r"><code>summary(wt_beta_chr22$HU005.10.cov)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.0000  0.0000  0.0000  0.1674  0.0000  8.0000 </code></pre>
<pre class="r"><code>wt_beta_chr22 %&gt;%  
  ggplot(aes(x = HU005.10.beta)) +
  geom_density()</code></pre>
<pre><code>Warning: Removed 539208 rows containing non-finite outside the scale range
(`stat_density()`).</code></pre>
<p><img src="figure/bedFileBetas.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="modkit" class="section level1">
<h1>modkit</h1>
<ul>
<li>if they match the beta values from megalodon that would be good</li>
</ul>
<pre class="r"><code>list.files(&quot;/researchers/caitlin.page/cf_nano/modkit_output/bed&quot;)</code></pre>
<pre><code> [1] &quot;hu5.10.bed&quot;        &quot;hu5.10.bed.gz.tbi&quot; &quot;hu5.11.bed&quot;       
 [4] &quot;ispro.bc1.bed&quot;     &quot;ispro.bc10.bed&quot;    &quot;ispro.bc11.bed&quot;   
 [7] &quot;ispro.bc2.bed&quot;     &quot;ispro.bc3.bed&quot;     &quot;ispro.bc4.bed&quot;    
[10] &quot;ispro.bc5.bed&quot;     &quot;ispro.bc8.bed&quot;     &quot;ispro.bc9.bed&quot;    
[13] &quot;ispro.s1.bed&quot;      &quot;sort_hu5.10.bed&quot;  </code></pre>
<pre class="r"><code>modkit_bed_hu5.10 &lt;- fread(&quot;/researchers/caitlin.page/cf_nano/modkit_output/bed/hu5.10.bed&quot;)</code></pre>
<pre class="r"><code>colnames(modkit_bed_hu5.10) &lt;- c(&quot;seqnames&quot;, &quot;start&quot;, &quot;end&quot;, &quot;mod_base_code&quot;, &quot;score_same_as_coverage&quot;, &quot;strand&quot;, &quot;start_pos&quot;, &quot;end_pos&quot;, &quot;colour&quot;, &quot;coverage&quot;, &quot;beta&quot;, &quot;n_mod&quot;, &quot;n_canon&quot;, &quot;n_other_mod&quot;, &quot;n_delete&quot;, &quot;n_fail&quot;, &quot;n_diff&quot;, &quot;n_nocall&quot;)

modkit_bed_hu5.10 &lt;- modkit_bed_hu5.10 %&gt;% filter(seqnames == &quot;chr22&quot;)
modkit_bed_hu5.10$pos &lt;- paste0(modkit_bed_hu5.10$seqnames, &quot;-&quot;, modkit_bed_hu5.10$start)</code></pre>
<pre class="r"><code>nrow(indiv_reads)</code></pre>
<pre><code>[1] 105981</code></pre>
<pre class="r"><code>#indiv_reads %&gt;% filter(ref_position %in% modkit_bed_hu5.10$start) %&gt;% nrow()
#indiv_reads %&gt;% filter(ref_position %in% modkit_bed_hu5.10$end)</code></pre>
<pre class="r"><code>#bed_betas_hu5.10 %&gt;% filter(start %in% modkit_bed_hu5.10$end)</code></pre>
<p>92973</p>
<pre class="r"><code>modkit_bed_hu5.10[1:5,]</code></pre>
<pre><code>   seqnames    start      end mod_base_code score_same_as_coverage strand
     &lt;char&gt;    &lt;int&gt;    &lt;int&gt;        &lt;char&gt;                  &lt;int&gt; &lt;char&gt;
1:    chr22 10521104 10521105             m                      1      -
2:    chr22 10521105 10521106             m                      1      -
3:    chr22 10521122 10521123             m                      1      -
4:    chr22 10521123 10521124             m                      1      -
5:    chr22 10521130 10521131             m                      1      -
   start_pos  end_pos  colour coverage  beta n_mod n_canon n_other_mod n_delete
       &lt;int&gt;    &lt;int&gt;  &lt;char&gt;    &lt;int&gt; &lt;num&gt; &lt;int&gt;   &lt;int&gt;       &lt;int&gt;    &lt;int&gt;
1:  10521104 10521105 255,0,0        1     0     0       1           0        0
2:  10521105 10521106 255,0,0        1     0     0       1           0        0
3:  10521122 10521123 255,0,0        1     0     0       1           0        0
4:  10521123 10521124 255,0,0        1     0     0       1           0        0
5:  10521130 10521131 255,0,0        1     0     0       1           0        0
   n_fail n_diff n_nocall            pos
    &lt;int&gt;  &lt;int&gt;    &lt;int&gt;         &lt;char&gt;
1:      0      0        0 chr22-10521104
2:      0      0        0 chr22-10521105
3:      0      0        0 chr22-10521122
4:      0      0        0 chr22-10521123
5:      0      0        0 chr22-10521130</code></pre>
<pre class="r"><code>find_overlaps(as_granges(modkit_bed_hu5.10), as_granges(bed_betas_hu5.10)) %&gt;% data.frame() %&gt;% nrow()</code></pre>
<pre><code>[1] 66601</code></pre>
<pre class="r"><code>find_overlaps(as_granges(modkit_bed_hu5.10), as_granges(bed_betas_hu5.10)) %&gt;% data.frame() %&gt;% distinct() %&gt;% nrow()</code></pre>
<pre><code>[1] 66601</code></pre>
<pre class="r"><code>find_overlaps(as_granges(modkit_bed_hu5.10), as_granges(cg_sites)) %&gt;% data.frame() %&gt;% nrow()</code></pre>
<pre><code>[1] 77313</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.1 (2024-06-14)
Platform: x86_64-pc-linux-gnu
Running under: Red Hat Enterprise Linux 9.5 (Plow)

Matrix products: default
BLAS/LAPACK: FlexiBLAS OPENBLAS-OPENMP;  LAPACK version 3.9.0

locale:
 [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
 [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8   
 [7] LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       

time zone: Australia/Melbourne
tzcode source: system (glibc)

attached base packages:
[1] stats4    stats     graphics  grDevices datasets  utils     methods  
[8] base     

other attached packages:
 [1] ggplot2_3.5.2        dplyr_1.1.4          reshape2_1.4.4      
 [4] tidyr_1.3.1          plyranges_1.26.0     GenomicRanges_1.58.0
 [7] GenomeInfoDb_1.42.3  IRanges_2.40.1       S4Vectors_0.44.0    
[10] BiocGenerics_0.52.0  data.table_1.17.8    workflowr_1.7.1     

loaded via a namespace (and not attached):
 [1] tidyselect_1.2.1            farver_2.1.2               
 [3] Biostrings_2.74.1           bitops_1.0-9               
 [5] fastmap_1.2.0               RCurl_1.98-1.17            
 [7] GenomicAlignments_1.42.0    promises_1.3.3             
 [9] XML_3.99-0.18               digest_0.6.37              
[11] lifecycle_1.0.4             processx_3.8.6             
[13] magrittr_2.0.3              compiler_4.4.1             
[15] rlang_1.1.6                 sass_0.4.10                
[17] tools_4.4.1                 utf8_1.2.6                 
[19] yaml_2.3.10                 rtracklayer_1.66.0         
[21] knitr_1.50                  labeling_0.4.3             
[23] S4Arrays_1.6.0              curl_6.4.0                 
[25] DelayedArray_0.32.0         RColorBrewer_1.1-3         
[27] plyr_1.8.9                  abind_1.4-8                
[29] BiocParallel_1.40.2         withr_3.0.2                
[31] purrr_1.1.0                 grid_4.4.1                 
[33] git2r_0.36.2                scales_1.4.0               
[35] SummarizedExperiment_1.36.0 cli_3.6.5                  
[37] rmarkdown_2.29              crayon_1.5.3               
[39] generics_0.1.4              rstudioapi_0.17.1          
[41] httr_1.4.7                  rjson_0.2.23               
[43] cachem_1.1.0                stringr_1.5.1              
[45] zlibbioc_1.52.0             parallel_4.4.1             
[47] BiocManager_1.30.26         XVector_0.46.0             
[49] restfulr_0.0.16             matrixStats_1.5.0          
[51] vctrs_0.6.5                 Matrix_1.7-0               
[53] jsonlite_2.0.0              callr_3.7.6                
[55] jquerylib_0.1.4             glue_1.8.0                 
[57] codetools_0.2-20            ps_1.9.1                   
[59] stringi_1.8.7               gtable_0.3.6               
[61] later_1.4.2                 BiocIO_1.16.0              
[63] UCSC.utils_1.2.0            tibble_3.3.0               
[65] pillar_1.11.0               htmltools_0.5.8.1          
[67] GenomeInfoDbData_1.2.13     R6_2.6.1                   
[69] rprojroot_2.1.0             evaluate_1.0.4             
[71] lattice_0.22-6              Biobase_2.66.0             
[73] Rsamtools_2.22.0            renv_1.1.5                 
[75] httpuv_1.6.16               bslib_0.9.0                
[77] Rcpp_1.1.0                  SparseArray_1.6.2          
[79] whisker_0.4.1               xfun_0.52                  
[81] fs_1.6.6                    MatrixGenerics_1.18.1      
[83] getPass_0.2-4               pkgconfig_2.0.3            </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
