<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="caitlinpage" />

<meta name="date" content="2025-10-29" />

<title>buildFunctions</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">3_cf_nano</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/caitlinpage/methyl_nano_cf">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">buildFunctions</h1>
<h4 class="author">caitlinpage</h4>
<h4 class="date">2025-10-29</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-11-07
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>methyl_nano_cf/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250606code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250606)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250606code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250606)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomcaitlinpagemethylnanocftree3318860008fdfc633c87e1a75ab64c04703795aftargetblank3318860a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/caitlinpage/methyl_nano_cf/tree/3318860008fdfc633c87e1a75ab64c04703795af" target="_blank">3318860</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomcaitlinpagemethylnanocftree3318860008fdfc633c87e1a75ab64c04703795aftargetblank3318860a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/caitlinpage/methyl_nano_cf/tree/3318860008fdfc633c87e1a75ab64c04703795af" target="_blank">3318860</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/figure/
    Ignored:    renv.lock
    Ignored:    renv/

Unstaged changes:
    Modified:   .Rprofile
    Modified:   .gitignore
    Modified:   analysis/00_ObtainData.Rmd
    Modified:   analysis/rrbsData.Rmd
    Modified:   code/seqFastqToReadMeth.txt

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/buildFunctions.Rmd</code>) and
HTML (<code>docs/buildFunctions.html</code>) files. If you’ve configured
a remote Git repository (see <code>?wflow_git_remote</code>), click on
the hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/caitlinpage/methyl_nano_cf/blob/3318860008fdfc633c87e1a75ab64c04703795af/analysis/buildFunctions.Rmd" target="_blank">3318860</a>
</td>
<td>
caitlinpage
</td>
<td>
2025-11-07
</td>
<td>
wflow_publish("analysis/buildFunctions.Rmd")
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/caitlinpage/methyl_nano_cf/blob/92fe9d0dfe2b1953079bb1f237b50bf741351116/analysis/buildFunctions.Rmd" target="_blank">92fe9d0</a>
</td>
<td>
caitlinpage
</td>
<td>
2025-11-07
</td>
<td>
removed unncessary code
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/caitlinpage/methyl_nano_cf/blob/3ee657fc7c0ce45b9ef3668b6b92167b0888dddc/analysis/buildFunctions.Rmd" target="_blank">3ee657f</a>
</td>
<td>
caitlinpage
</td>
<td>
2025-11-07
</td>
<td>
tidied up rrbs fns
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/caitlinpage/methyl_nano_cf/blob/ff83440d111b699e38477098c6247ee8c62d5974/analysis/buildFunctions.Rmd" target="_blank">ff83440</a>
</td>
<td>
caitlinpage
</td>
<td>
2025-11-07
</td>
<td>
copied code to make fns from meth_pattern
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/caitlinpage/methyl_nano_cf/blob/3070c251a3b8a848388d44a689c817c481c0221a/analysis/buildFunctions.Rmd" target="_blank">3070c25</a>
</td>
<td>
caitlinpage
</td>
<td>
2025-11-07
</td>
<td>
writing fns for nanopore and wgbs
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/caitlinpage/methyl_nano_cf/blob/c54e1912330cd360bb1a279aa25356a66fcc7375/analysis/buildFunctions.Rmd" target="_blank">c54e191</a>
</td>
<td>
Caitlin Page
</td>
<td>
2025-11-07
</td>
<td>
fleshed out
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/caitlinpage/methyl_nano_cf/blob/0d6123695ee583c0803d8f977b6b259a7dda46c6/analysis/buildFunctions.Rmd" target="_blank">0d61236</a>
</td>
<td>
caitlinpage
</td>
<td>
2025-11-07
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/caitlinpage/methyl_nano_cf/blob/b122f9d5d4c5bfc120865f381a5c55363e901300/analysis/buildFunctions.Rmd" target="_blank">b122f9d</a>
</td>
<td>
caitlinpage
</td>
<td>
2025-10-31
</td>
<td>
new file focussed on fns
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<pre class="r"><code>library(plyranges)</code></pre>
<pre><code>Loading required package: BiocGenerics</code></pre>
<pre><code>
Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    IQR, mad, sd, var, xtabs</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    anyDuplicated, aperm, append, as.data.frame, basename, cbind,
    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,
    get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,
    match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,
    Position, rank, rbind, Reduce, rownames, sapply, saveRDS, setdiff,
    table, tapply, union, unique, unsplit, which.max, which.min</code></pre>
<pre><code>Loading required package: IRanges</code></pre>
<pre><code>Loading required package: S4Vectors</code></pre>
<pre><code>Loading required package: stats4</code></pre>
<pre><code>
Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:utils&#39;:

    findMatches</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    expand.grid, I, unname</code></pre>
<pre><code>Loading required package: GenomicRanges</code></pre>
<pre><code>Loading required package: GenomeInfoDb</code></pre>
<pre><code>Warning: replacing previous import &#39;S4Arrays::makeNindexFromArrayViewport&#39; by
&#39;DelayedArray::makeNindexFromArrayViewport&#39; when loading &#39;SummarizedExperiment&#39;</code></pre>
<pre><code>
Attaching package: &#39;plyranges&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:IRanges&#39;:

    slice</code></pre>
<pre><code>The following object is masked from &#39;package:stats&#39;:

    filter</code></pre>
<pre class="r"><code>library(stringr)
library(BSgenome.Hsapiens.UCSC.hg38)</code></pre>
<pre><code>Loading required package: BSgenome</code></pre>
<pre><code>Loading required package: Biostrings</code></pre>
<pre><code>Loading required package: XVector</code></pre>
<pre><code>
Attaching package: &#39;Biostrings&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:base&#39;:

    strsplit</code></pre>
<pre><code>Loading required package: BiocIO</code></pre>
<pre><code>Loading required package: rtracklayer</code></pre>
<pre><code>
Attaching package: &#39;rtracklayer&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:BiocIO&#39;:

    FileForFormat</code></pre>
<pre class="r"><code>library(tidyr)</code></pre>
<pre><code>
Attaching package: &#39;tidyr&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:S4Vectors&#39;:

    expand</code></pre>
<pre class="r"><code>library(dplyr)</code></pre>
<pre><code>
Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:Biostrings&#39;:

    collapse, intersect, setdiff, setequal, union</code></pre>
<pre><code>The following object is masked from &#39;package:XVector&#39;:

    slice</code></pre>
<pre><code>The following objects are masked from &#39;package:plyranges&#39;:

    between, n, n_distinct</code></pre>
<pre><code>The following objects are masked from &#39;package:GenomicRanges&#39;:

    intersect, setdiff, union</code></pre>
<pre><code>The following object is masked from &#39;package:GenomeInfoDb&#39;:

    intersect</code></pre>
<pre><code>The following objects are masked from &#39;package:IRanges&#39;:

    collapse, desc, intersect, setdiff, slice, union</code></pre>
<pre><code>The following objects are masked from &#39;package:S4Vectors&#39;:

    first, intersect, rename, setdiff, setequal, union</code></pre>
<pre><code>The following objects are masked from &#39;package:BiocGenerics&#39;:

    combine, intersect, setdiff, union</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    filter, lag</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    intersect, setdiff, setequal, union</code></pre>
<pre class="r"><code>library(ggplot2)
library(data.table)</code></pre>
<pre><code>
Attaching package: &#39;data.table&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    between, first, last</code></pre>
<pre><code>The following object is masked from &#39;package:plyranges&#39;:

    between</code></pre>
<pre><code>The following object is masked from &#39;package:GenomicRanges&#39;:

    shift</code></pre>
<pre><code>The following object is masked from &#39;package:IRanges&#39;:

    shift</code></pre>
<pre><code>The following objects are masked from &#39;package:S4Vectors&#39;:

    first, second</code></pre>
<ul>
<li><p>starting a new file because outputs from modkit (nanopore) and
bismark (wgbs + rrbs) are very very different</p></li>
<li><p>and I want to get them in the same place</p></li>
<li><p>then this also stays centralised as a way to make sure if I
modify pieces other functions still run</p></li>
<li><p>and I love making new files for new ideas</p></li>
<li><p>so use this as an end to end workflow as well</p></li>
<li><p>going from read+cpg to context with beta and alpha</p></li>
</ul>
</div>
<div id="files" class="section level1">
<h1>files</h1>
<ul>
<li><p>pat file (with extra info) : gives alpha</p></li>
<li><p>beta file</p></li>
<li><p>og read level: has everything</p></li>
<li><p>technically can get beta from alpha</p></li>
<li><p>nanopore issue was figuring out the filtering metrics from modkit
for that to be accurate</p></li>
<li><p>will have to go through the same process for bismark
files</p></li>
</ul>
</div>
<div id="processing-files" class="section level1">
<h1>processing files</h1>
<ul>
<li>from practical standpoint: use datatable and fread because files are
LARGE</li>
</ul>
<div id="helpers" class="section level2">
<h2>helpers</h2>
<pre class="r"><code>cg_sites &lt;- cbind(data.frame(Biostrings::matchPattern(&quot;CG&quot;, BSgenome.Hsapiens.UCSC.hg38[[&quot;chr22&quot;]])), seqnames = &quot;chr22&quot;) %&gt;% 
  mutate(pos = paste0(seqnames, &quot;-&quot;, start)) %&gt;%
  relocate(pos, seqnames)</code></pre>
<pre><code>Warning in .local(x, row.names, optional, ...): &#39;optional&#39; argument was ignored</code></pre>
<pre class="r"><code>cg_sites$index &lt;- 1:nrow(cg_sites)
c(number_of_sites = nrow(cg_sites))</code></pre>
<pre><code>number_of_sites 
         634646 </code></pre>
<pre class="r"><code>head(cg_sites)</code></pre>
<pre><code>             pos seqnames    start      end width seq index
1 chr22-10510235    chr22 10510235 10510236     2  CG     1
2 chr22-10510275    chr22 10510275 10510276     2  CG     2
3 chr22-10510284    chr22 10510284 10510285     2  CG     3
4 chr22-10510338    chr22 10510338 10510339     2  CG     4
5 chr22-10510437    chr22 10510437 10510438     2  CG     5
6 chr22-10510532    chr22 10510532 10510533     2  CG     6</code></pre>
</div>
<div id="nanopore" class="section level2">
<h2>nanopore</h2>
<ul>
<li><p>need modkit read file from modkit_extract_full</p></li>
<li><p>bed file from downloaded data (bed file from modkit was
weird)</p></li>
<li><p>fread is a nightmare</p></li>
</ul>
<pre class="r"><code>#fread(path_to_modkit_output)
process_modkit &lt;- function(modkit_fread, path_to_bed_file) {
  reads &lt;- modkit_fread
  reads &lt;- reads$ref_position + 1 # convert to 1 based
  reads &lt;- reads[, ref_position := ifelse(strand == &quot;-&quot;, ref_position - 1, ref_position)] # convert to 1 based
  reads &lt;- reads[, read_and_pos := paste0(reads$read_id, &quot;:&quot;, reads$ref_position)]
  reads$start &lt;- reads$ref_position
  reads$end &lt;- reads$ref_position
  
  bed &lt;- fread(path_to_bed_file)
  
  alpha_and_beta2 &lt;- find_overlaps(as_granges(reads), as_granges(bed)) %&gt;% data.frame() %&gt;% 
    filter(call_prob &gt; 0.66) %&gt;% # solution for alpha and beta match
    group_by(read_id) %&gt;% 
    .[order(.$alignment_start),] %&gt;% 
    mutate(call_code = ifelse(call_code == &quot;m&quot;, &quot;M&quot;, &quot;U&quot;)) %&gt;% 
    mutate(meth_pattern = paste(call_code, collapse = &quot;&quot;)) %&gt;% 
    mutate(num_meth = str_count(meth_pattern, pattern = &quot;M&quot;),
         total = str_length(meth_pattern),
         alpha = num_meth/total) %&gt;% 
    mutate(genom_positions = paste0(ref_position, collapse = &quot;,&quot;)) %&gt;% 
    ungroup()
  alpha_and_beta2 &lt;- alpha_and_beta2 %&gt;% data.frame()
  alpha_and_beta2
}

modkit_to_pat &lt;- function(alpha_and_beta) {
  alpha_and_beta2_pat &lt;- alpha_and_beta2 %&gt;% 
    mutate(start_pos = sapply(str_split(genom_positions, &quot;,&quot;), head, 1),
           end_pos = sapply(str_split(genom_positions, &quot;,&quot;), tail, 1)) %&gt;% 
    filter(ref_position == start_pos) %&gt;% 
    distinct(read_id, seqnames, start_pos, meth_pattern, alpha, end_pos, beta) %&gt;% 
    group_by(seqnames, start_pos, meth_pattern, alpha, end_pos, beta) %&gt;% 
    summarise(n=n()) %&gt;% 
    ungroup()

  alpha_and_beta2_pat &lt;- alpha_and_beta2_pat %&gt;% 
    mutate(start_pos = as.integer(start_pos), 
           end_pos = as.integer(end_pos))
  alpha_and_beta2_pat
}</code></pre>
</div>
<div id="wgbsrrbs" class="section level2">
<h2>wgbs/rrbs</h2>
<ul>
<li>fread is a nightmare</li>
<li>it won’t run within a fn - doesn’t trust the path</li>
<li>but it’s way faster than any other alternative</li>
<li>so for now - read in the file with fread and then put that into the
fn</li>
</ul>
<pre class="r"><code>#read &lt;- fread(file = path_to_bismark_res)
#bed &lt;- fread(file = path_to_bed) # for betas - otherwise will make them myself
process_wgbs &lt;- function(bismark, bed = NULL) {
  read &lt;- bismark
  names(read) &lt;- c(&quot;read_id&quot;, &quot;strand&quot;, &quot;seqnames&quot;, &quot;start&quot;, &quot;meth_status&quot;)
  read$end &lt;- read$start
  
 # if (is.null(bed) == TRUE) {
#    read_overlap &lt;- read
 # } else {
#  read_overlap &lt;- find_overlaps(as_granges(read), as_granges(bed)) %&gt;% data.frame()
 # }
  read_overlap &lt;- read
  read_overlap[, meth_status := ifelse(meth_status == &quot;Z&quot;, &quot;M&quot;, &quot;U&quot;)]
  read_overlap[, meth_pattern := paste0(meth_status, collapse = &quot;&quot;), by = read_id]
  read_overlap[, num_meth := str_count(meth_pattern, pattern = &quot;M&quot;)]
  read_overlap[, total := str_length(meth_pattern)]
  read_overlap[, alpha := num_meth/total]
  read_overlap[, genom_positions := paste0(start, collapse = &quot;,&quot;), by = read_id]

  read_overlap &lt;- read_overlap %&gt;% .[order(.$read_id),]
  
  betas &lt;- betas_for_wgbs(read_overlap)
  betas &lt;- data.frame(betas)
  read_overlap &lt;- read_overlap %&gt;% group_by(read_id) %&gt;% 
    mutate(read_start = min(start), read_end = max(end), 
           genom_positions = paste0(start, collapse = &quot;,&quot;)) %&gt;% 
    ungroup()
  read_overlap &lt;- read_overlap %&gt;% data.frame() %&gt;% 
    mutate(beta = betas[match(.$start, betas$start), &quot;beta&quot;])
  
  read_overlap
}</code></pre>
<pre class="r"><code>betas_for_wgbs &lt;- function(bismark_processed) {
  res &lt;- bismark_processed %&gt;% group_by(start, meth_status) %&gt;% 
  .[order(.$start),] %&gt;%
  mutate(meth_count_pos = n(), unmeth_count_pos = n()) %&gt;%
  mutate(meth_count_pos = ifelse(meth_status == &quot;M&quot;, meth_count_pos, NA),
         unmeth_count_pos = ifelse(meth_status == &quot;U&quot;, unmeth_count_pos, NA)) %&gt;% 
    ungroup() %&gt;% 
    distinct(seqnames, start, end, meth_count_pos, unmeth_count_pos) %&gt;%  
 group_by(start) %&gt;% 
    .[order(.$meth_count_pos),] %&gt;%
  fill(meth_count_pos, .direction = &quot;down&quot;) %&gt;%
  fill(unmeth_count_pos, .direction = &quot;up&quot;) %&gt;% 
  replace(is.na(.), 0) %&gt;%
  mutate(beta = meth_count_pos/(meth_count_pos + unmeth_count_pos)) %&gt;%
  ungroup() %&gt;%
  distinct(seqnames, start, end, meth_count_pos, unmeth_count_pos, beta)
  
  res
}</code></pre>
<pre class="r"><code>pat_for_wgbs &lt;- function(process_res) {
  pat &lt;- process_res %&gt;% 
  mutate(start_pos = sapply(str_split(genom_positions, &quot;,&quot;), head, 1),
         end_pos = sapply(str_split(genom_positions, &quot;,&quot;), tail, 1)) %&gt;% 
  filter(start == start_pos) %&gt;%
  distinct(read_id, seqnames, start_pos, meth_pattern, alpha, end_pos, beta) %&gt;% 
  group_by(seqnames, start_pos, meth_pattern, alpha, end_pos, beta) %&gt;% 
  summarise(n=n()) %&gt;%
  ungroup()  %&gt;%
  mutate(start_pos = as.integer(start_pos),
         end_pos = as.integer(end_pos))
  
  pat
}</code></pre>
<pre class="r"><code>bis &lt;- fread(&quot;/researchers/caitlin.page/rrbs_encode_cell_line/CpG_OT_ENCFF000LYO_bismark_bt2.txt&quot;)</code></pre>
<pre><code>Warning in
fread(&quot;/researchers/caitlin.page/rrbs_encode_cell_line/CpG_OT_ENCFF000LYO_bismark_bt2.txt&quot;):
Detected 1 column names but the data has 5 columns (i.e. invalid file). Added 4
extra default column names at the end.</code></pre>
<pre class="r"><code>for_beta &lt;- process_wgbs(bis)
for_beta &lt;- for_beta %&gt;% filter(seqnames == &quot;chr22&quot;)
rrbs_for_beta_pat &lt;- pat_for_wgbs(for_beta)</code></pre>
<pre><code>`summarise()` has grouped output by &#39;seqnames&#39;, &#39;start_pos&#39;, &#39;meth_pattern&#39;,
&#39;alpha&#39;, &#39;end_pos&#39;. You can override using the `.groups` argument.</code></pre>
</div>
</div>
<div id="plotting" class="section level1">
<h1>plotting</h1>
<pre class="r"><code>plot_fn &lt;- function(chrom = chrom, region = c(start, end), data_big = data_big, cg_sites = cg_sites) {
  chr &lt;- chrom
  start &lt;- region[1]
  end &lt;- region[2]
  
  data &lt;- data_big
  cpgs &lt;- cg_sites
  
  data &lt;- data %&gt;% 
    filter(seqnames == chr,
           start &gt;= start,
           end &lt;= end) 
  data2 &lt;- data %&gt;%
    distinct(read_id, alignment_start, alignment_end, alpha)
  
  plot &lt;- data2 %&gt;% 
    ggplot() +
    geom_segment(aes(x = alignment_start, xend = alignment_end, #read length
                     y = alpha, yend = alpha), alpha = 0.2) +
    geom_point(data = data, 
             aes(x = ref_position, y = alpha, colour = call_code)) + # alpha dots, 1 per read
    geom_point(data = distinct(data, ref_position, beta),            # beta as X
            aes(x = ref_position, y = beta), shape = 4, size = 3) +
    geom_segment(data = filter(cpgs, start &gt;= start, end &lt;= end),    # green lines showing cpg positions
                 aes(x = start, xend = start, y = -0.15, yend = -0.05), colour = &quot;green&quot;)
  plot
}</code></pre>
<pre class="r"><code># provide vector of read ids
rrbs_plot &lt;- function(long_data, read_ids, cpgs = cg_sites) {
  long_data %&gt;% filter(read_id %in% read_ids) %&gt;% 
    distinct(read_id, read_start, read_end, alpha) %&gt;%
  ggplot() +
  geom_segment(aes(x = read_start, xend = read_end, y = alpha, yend = alpha), alpha = 0.2) +
  geom_point(data = filter(long_data, read_id %in% read_ids), 
             aes(x = start, y = alpha, colour = meth_status)) +
  geom_point(data = distinct(filter(long_data, read_id %in% read_ids), start, beta), 
            aes(x = start, y = beta), shape = 4, size = 3) +
  geom_segment(data = filter(cg_sites, start &gt;= min(filter(long_data, read_id %in% read_ids)$start), 
                             end &lt;= max(filter(long_data, read_id %in% read_ids)$end)), 
               aes(x = start, xend = start, y = -0.15, yend = -0.05), colour = &quot;green&quot;)
}</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.1 (2024-06-14)
Platform: x86_64-pc-linux-gnu
Running under: Rocky Linux 9.6 (Blue Onyx)

Matrix products: default
BLAS/LAPACK: FlexiBLAS OPENBLAS-OPENMP;  LAPACK version 3.9.0

locale:
 [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
 [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8   
 [7] LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       

time zone: Australia/Melbourne
tzcode source: system (glibc)

attached base packages:
[1] stats4    stats     graphics  grDevices datasets  utils     methods  
[8] base     

other attached packages:
 [1] data.table_1.17.8                 ggplot2_3.5.2                    
 [3] dplyr_1.1.4                       tidyr_1.3.1                      
 [5] BSgenome.Hsapiens.UCSC.hg38_1.4.5 BSgenome_1.74.0                  
 [7] rtracklayer_1.66.0                BiocIO_1.16.0                    
 [9] Biostrings_2.74.1                 XVector_0.46.0                   
[11] stringr_1.5.1                     plyranges_1.26.0                 
[13] GenomicRanges_1.58.0              GenomeInfoDb_1.42.3              
[15] IRanges_2.40.1                    S4Vectors_0.44.0                 
[17] BiocGenerics_0.52.0              

loaded via a namespace (and not attached):
 [1] SummarizedExperiment_1.36.0 gtable_0.3.6               
 [3] rjson_0.2.23                xfun_0.52                  
 [5] bslib_0.9.0                 Biobase_2.66.0             
 [7] lattice_0.22-6              vctrs_0.6.5                
 [9] tools_4.4.1                 bitops_1.0-9               
[11] generics_0.1.4              curl_6.4.0                 
[13] parallel_4.4.1              tibble_3.3.0               
[15] pkgconfig_2.0.3             Matrix_1.7-0               
[17] RColorBrewer_1.1-3          lifecycle_1.0.4            
[19] GenomeInfoDbData_1.2.13     farver_2.1.2               
[21] compiler_4.4.1              git2r_0.36.2               
[23] Rsamtools_2.22.0            codetools_0.2-20           
[25] httpuv_1.6.16               htmltools_0.5.8.1          
[27] sass_0.4.10                 RCurl_1.98-1.17            
[29] yaml_2.3.10                 later_1.4.2                
[31] pillar_1.11.0               crayon_1.5.3               
[33] jquerylib_0.1.4             whisker_0.4.1              
[35] BiocParallel_1.40.2         cachem_1.1.0               
[37] DelayedArray_0.32.0         abind_1.4-8                
[39] tidyselect_1.2.1            digest_0.6.37              
[41] stringi_1.8.7               purrr_1.1.0                
[43] restfulr_0.0.16             rprojroot_2.1.0            
[45] fastmap_1.2.0               grid_4.4.1                 
[47] cli_3.6.5                   SparseArray_1.6.2          
[49] magrittr_2.0.3              S4Arrays_1.6.0             
[51] XML_3.99-0.18               withr_3.0.2                
[53] scales_1.4.0                UCSC.utils_1.2.0           
[55] promises_1.3.3              rmarkdown_2.29             
[57] httr_1.4.7                  matrixStats_1.5.0          
[59] workflowr_1.7.1             evaluate_1.0.4             
[61] knitr_1.50                  rlang_1.1.6                
[63] Rcpp_1.1.0                  glue_1.8.0                 
[65] BiocManager_1.30.26         renv_1.1.5                 
[67] rstudioapi_0.17.1           jsonlite_2.0.0             
[69] R6_2.6.1                    MatrixGenerics_1.18.1      
[71] GenomicAlignments_1.42.0    fs_1.6.6                   
[73] zlibbioc_1.52.0            </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
