<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="caitlinpage" />

<meta name="date" content="2025-09-03" />

<title>checkAlphaAndBetaMatch</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">3_cf_nano</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/caitlinpage/methyl_nano_cf">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">checkAlphaAndBetaMatch</h1>
<h4 class="author">caitlinpage</h4>
<h4 class="date">2025-09-03</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-09-24
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>methyl_nano_cf/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250606code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250606)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250606code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250606)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomcaitlinpagemethylnanocftree330f8ae4336defef83046265e4fc67c6e0dc5909targetblank330f8aea">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/caitlinpage/methyl_nano_cf/tree/330f8ae4336defef83046265e4fc67c6e0dc5909" target="_blank">330f8ae</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomcaitlinpagemethylnanocftree330f8ae4336defef83046265e4fc67c6e0dc5909targetblank330f8aea"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/caitlinpage/methyl_nano_cf/tree/330f8ae4336defef83046265e4fc67c6e0dc5909" target="_blank">330f8ae</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/figure/
    Ignored:    renv.lock
    Ignored:    renv/

Unstaged changes:
    Modified:   .Rprofile
    Modified:   .gitignore
    Modified:   analysis/binningGenome.Rmd
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/checkAlphaAndBetaMatch.Rmd</code>) and HTML
(<code>docs/checkAlphaAndBetaMatch.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/caitlinpage/methyl_nano_cf/blob/330f8ae4336defef83046265e4fc67c6e0dc5909/analysis/checkAlphaAndBetaMatch.Rmd" target="_blank">330f8ae</a>
</td>
<td>
caitlinpage
</td>
<td>
2025-09-24
</td>
<td>
wflow_publish("analysis/checkAlphaAndBetaMatch.Rmd")
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/caitlinpage/methyl_nano_cf/blob/aca159b16ea5fc5cd0ab64c32dbbc9b59e6f2773/analysis/checkAlphaAndBetaMatch.Rmd" target="_blank">aca159b</a>
</td>
<td>
caitlinpage
</td>
<td>
2025-09-05
</td>
<td>
made bed files with modkit and still don’t align
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/caitlinpage/methyl_nano_cf/blob/1c84beea0b02a472d86c5bfa5df02b8e0f6b3db9/analysis/checkAlphaAndBetaMatch.Rmd" target="_blank">1c84bee</a>
</td>
<td>
caitlinpage
</td>
<td>
2025-09-03
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/caitlinpage/methyl_nano_cf/blob/cbc1d0c3dc79e065f5d1e3d90085cbe83b5f4838/analysis/checkAlphaAndBetaMatch.Rmd" target="_blank">cbc1d0c</a>
</td>
<td>
caitlinpage
</td>
<td>
2025-09-03
</td>
<td>
inital commit - try to figure out if alpha and beta actually are
aligning properly or what’s going on
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Alpha values are obtained from <a
href="modkitOutputToReadFormat.html">alpha</a></p>
<p>Beta values were first done in research sprint work: <a
href="researchSprint.html">day 1</a>, <a href="researchSprintB.html">day
2</a></p>
<p>Beta values were then obtained again <a
href="plotAlphaAndBeta.html">here???</a></p>
<p>Goal was to plot alpha and beta together, <a
href="plotAlphaAndBeta.html">plots</a></p>
<p>But I think something went wrong and I no longer trust either my
alpha or beta values an</p>
<div id="packages" class="section level2">
<h2>Packages</h2>
<pre class="r"><code>library(data.table)

library(plyranges)</code></pre>
<pre><code>Loading required package: BiocGenerics</code></pre>
<pre><code>
Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    IQR, mad, sd, var, xtabs</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    anyDuplicated, aperm, append, as.data.frame, basename, cbind,
    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,
    get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,
    match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,
    Position, rank, rbind, Reduce, rownames, sapply, saveRDS, setdiff,
    table, tapply, union, unique, unsplit, which.max, which.min</code></pre>
<pre><code>Loading required package: IRanges</code></pre>
<pre><code>Loading required package: S4Vectors</code></pre>
<pre><code>Loading required package: stats4</code></pre>
<pre><code>
Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:data.table&#39;:

    first, second</code></pre>
<pre><code>The following object is masked from &#39;package:utils&#39;:

    findMatches</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    expand.grid, I, unname</code></pre>
<pre><code>
Attaching package: &#39;IRanges&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:data.table&#39;:

    shift</code></pre>
<pre><code>Loading required package: GenomicRanges</code></pre>
<pre><code>Loading required package: GenomeInfoDb</code></pre>
<pre><code>Warning: replacing previous import &#39;S4Arrays::makeNindexFromArrayViewport&#39; by
&#39;DelayedArray::makeNindexFromArrayViewport&#39; when loading &#39;SummarizedExperiment&#39;</code></pre>
<pre><code>
Attaching package: &#39;plyranges&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:IRanges&#39;:

    slice</code></pre>
<pre><code>The following object is masked from &#39;package:data.table&#39;:

    between</code></pre>
<pre><code>The following object is masked from &#39;package:stats&#39;:

    filter</code></pre>
<pre class="r"><code>library(tidyr)</code></pre>
<pre><code>
Attaching package: &#39;tidyr&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:S4Vectors&#39;:

    expand</code></pre>
<pre class="r"><code>library(dplyr)</code></pre>
<pre><code>
Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:plyranges&#39;:

    between, n, n_distinct</code></pre>
<pre><code>The following objects are masked from &#39;package:GenomicRanges&#39;:

    intersect, setdiff, union</code></pre>
<pre><code>The following object is masked from &#39;package:GenomeInfoDb&#39;:

    intersect</code></pre>
<pre><code>The following objects are masked from &#39;package:IRanges&#39;:

    collapse, desc, intersect, setdiff, slice, union</code></pre>
<pre><code>The following objects are masked from &#39;package:S4Vectors&#39;:

    first, intersect, rename, setdiff, setequal, union</code></pre>
<pre><code>The following objects are masked from &#39;package:BiocGenerics&#39;:

    combine, intersect, setdiff, union</code></pre>
<pre><code>The following objects are masked from &#39;package:data.table&#39;:

    between, first, last</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    filter, lag</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    intersect, setdiff, setequal, union</code></pre>
<pre class="r"><code>library(ggplot2)</code></pre>
<pre class="r"><code>all_samples &lt;- readRDS(&quot;/researchers/caitlin.page/cf_nano/r_output/all_samples.rds&quot;)
overlap_bins_reads &lt;- readRDS(&quot;/researchers/caitlin.page/cf_nano/r_output/overlap_bins_reads.rds&quot;)
cg_sites &lt;- readRDS(&quot;/researchers/caitlin.page/cf_nano/r_output/cg_sites.rds&quot;)
bed_betas &lt;- readRDS(&quot;/researchers/caitlin.page/cf_nano/r_output/bed_betas.rds&quot;)
wt_beta_chr22 &lt;- readRDS(&quot;/researchers/caitlin.page/cf_nano/r_output/wt_beta_chr22.rds&quot;)
indiv_reads &lt;- readRDS(&quot;/researchers/caitlin.page/cf_nano/r_output/indiv_reads.rds&quot;)</code></pre>
</div>
<div id="code-for-plot" class="section level2">
<h2>Code for plot</h2>
<ul>
<li>bin 3780 is the most dense (most number of cpgs)</li>
<li>use that to plot</li>
</ul>
<pre class="r"><code>overlap_bins_reads %&gt;% .[order(.$num_reads_in_bin, decreasing = TRUE),] %&gt;% .[1, c(&quot;bin_num&quot;, &quot;num_reads_in_bin&quot;)]</code></pre>
<pre><code># A tibble: 1 × 2
  bin_num num_reads_in_bin
    &lt;int&gt;            &lt;int&gt;
1    3780              968</code></pre>
<pre class="r"><code>bin_cpgs &lt;- all_samples %&gt;% filter(read_id %in% filter(overlap_bins_reads, bin_num == 3780)$read_id) %&gt;% .$genom_positions %&gt;% strsplit(&quot;,&quot;) %&gt;% unlist() %&gt;% as.double()
bin_cpgs &lt;- data.frame(bin_cpgs)
bin_cpgs &lt;- cbind(bin_cpgs, meth = all_samples %&gt;% filter(read_id %in% filter(overlap_bins_reads, bin_num == 3780)$read_id) %&gt;% .$meth_pattern %&gt;% strsplit(&quot;&quot;) %&gt;% unlist())
bin_cpgs &lt;- bin_cpgs %&gt;% mutate(site_is_unmeth = ifelse(meth == &quot;T&quot;, TRUE, FALSE))

alpha_vec &lt;- all_samples %&gt;% filter(read_id %in% filter(overlap_bins_reads, bin_num == 3780)$read_id) %&gt;% .$alpha 
rep_vec &lt;- all_samples %&gt;% filter(read_id %in% filter(overlap_bins_reads, bin_num == 3780)$read_id) %&gt;% .$total 

bin_cpgs &lt;- cbind(bin_cpgs, alpha = rep(x = alpha_vec, times = rep_vec))
# sample
alpha_vec &lt;- all_samples %&gt;% filter(read_id %in% filter(overlap_bins_reads, bin_num == 3780)$read_id) %&gt;% .$sample 
rep_vec &lt;- all_samples %&gt;% filter(read_id %in% filter(overlap_bins_reads, bin_num == 3780)$read_id) %&gt;% .$total 
bin_cpgs &lt;- cbind(bin_cpgs, sample = rep(x = alpha_vec, times = rep_vec))
# type
alpha_vec &lt;- all_samples %&gt;% filter(read_id %in% filter(overlap_bins_reads, bin_num == 3780)$read_id) %&gt;% .$type 
rep_vec &lt;- all_samples %&gt;% filter(read_id %in% filter(overlap_bins_reads, bin_num == 3780)$read_id) %&gt;% .$total 
bin_cpgs &lt;- cbind(bin_cpgs, type = rep(x = alpha_vec, times = rep_vec))</code></pre>
<pre class="r"><code>overlap_bins_reads %&gt;% filter(bin_num == 3780) %&gt;%
  ggplot() +
  geom_segment(aes(x = read_start, xend = read_end, y = alpha, yend = alpha, colour = sample, group = sample)) +
  geom_point(data = bin_cpgs, aes(x = bin_cpgs, y = alpha, shape = site_is_unmeth, colour = sample, group = sample)) +
  scale_shape_manual(values = c(`TRUE` = 1, `FALSE` = 16),
                     name = &quot;Meth status&quot;, labels = c(`TRUE` = &quot;Unmeth&quot;, `FALSE` = &quot;Meth&quot;)) +
  labs(title = &quot;bin 3780&quot;) +
  facet_wrap(~type, nrow = 2)</code></pre>
<p><img src="figure/checkAlphaAndBetaMatch.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" />
* this plot has alpha values and the methylation state for each cpg on a
read</p>
<ul>
<li>split the plot by sample</li>
</ul>
<pre class="r"><code>bed_betas %&gt;% filter(sample == &quot;hu5.10&quot;) %&gt;% summary(.)</code></pre>
<pre><code>    seqnames         start               end               width   strand   
 chr22  :92973   Min.   :10521181   Min.   :10521181   Min.   :1   +:    0  
 chr1   :    0   1st Qu.:23521265   1st Qu.:23521265   1st Qu.:1   -:    0  
 chr2   :    0   Median :35373136   Median :35373136   Median :1   *:92973  
 chr3   :    0   Mean   :33589451   Mean   :33589451   Mean   :1            
 chr4   :    0   3rd Qu.:43827134   3rd Qu.:43827134   3rd Qu.:1            
 chr5   :    0   Max.   :50807889   Max.   :50807889   Max.   :1            
 (Other):    0                                                              
      name           sample              type                beta       
 Min.   :0.0000   Length:92973       Length:92973       Min.   :0.0000  
 1st Qu.:0.0000   Class :character   Class :character   1st Qu.:0.0000  
 Median :1.0000   Mode  :character   Mode  :character   Median :1.0000  
 Mean   :0.6989                                         Mean   :0.6989  
 3rd Qu.:1.0000                                         3rd Qu.:1.0000  
 Max.   :1.0000                                         Max.   :1.0000  
                                                                        
  bed_sample       
 Length:92973      
 Class :character  
 Mode  :character  
                   
                   
                   
                   </code></pre>
<pre class="r"><code>summary(wt_beta_chr22$HU005.10.unmeth)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
      0       0       0       0       0       0 </code></pre>
<pre class="r"><code>overlap_bins_reads %&gt;% filter(bin_num == 3780) %&gt;%
  ggplot() +
  geom_segment(aes(x = read_start, xend = read_end, y = alpha, yend = alpha, colour = type, group = sample)) +
  geom_point(data = bin_cpgs, aes(x = bin_cpgs, y = alpha, shape = site_is_unmeth, colour = type, group = sample)) +
  scale_shape_manual(values = c(`TRUE` = 1, `FALSE` = 16),
                     name = &quot;Meth status&quot;, labels = c(`TRUE` = &quot;Unmeth&quot;, `FALSE` = &quot;Meth&quot;)) +
  labs(title = &quot;bin 3780&quot;) +
  facet_wrap(~sample)</code></pre>
<p><img src="figure/checkAlphaAndBetaMatch.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>overlap_bins_reads %&gt;% filter(bin_num == 3780, sample == &quot;hu5.10&quot;) %&gt;%
  ggplot() +
  geom_segment(aes(x = read_start, xend = read_end, y = alpha, yend = alpha, colour = type, group = sample)) +
  geom_point(data = filter(bin_cpgs, sample == &quot;hu5.10&quot;), aes(x = bin_cpgs, y = alpha, shape = site_is_unmeth, colour = type, group = sample)) +
  scale_shape_manual(values = c(`TRUE` = 1, `FALSE` = 16),
                     name = &quot;Meth status&quot;, labels = c(`TRUE` = &quot;Unmeth&quot;, `FALSE` = &quot;Meth&quot;)) +
  labs(title = &quot;bin 3780: hu5.10&quot;)</code></pre>
<p><img src="figure/checkAlphaAndBetaMatch.Rmd/unnamed-chunk-7-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>overlap_bins_reads %&gt;% filter(bin_num == 3780, sample == &quot;hu5.10&quot;, alpha %in% c(0,1)) </code></pre>
<pre><code># A tibble: 7 × 21
  seqnames    start    end width strand bin_num read_id read_length type  sample
  &lt;fct&gt;       &lt;int&gt;  &lt;int&gt; &lt;int&gt; &lt;fct&gt;    &lt;int&gt; &lt;chr&gt;         &lt;int&gt; &lt;chr&gt; &lt;chr&gt; 
1 chr22    18895001 1.89e7  5000 *         3780 a323a4…         298 heal… hu5.10
2 chr22    18895001 1.89e7  5000 *         3780 d4aa41…         184 heal… hu5.10
3 chr22    18895001 1.89e7  5000 *         3780 4d7a8e…         254 heal… hu5.10
4 chr22    18895001 1.89e7  5000 *         3780 42b151…         296 heal… hu5.10
5 chr22    18895001 1.89e7  5000 *         3780 d855ca…         202 heal… hu5.10
6 chr22    18895001 1.89e7  5000 *         3780 b62fe6…         366 heal… hu5.10
7 chr22    18895001 1.89e7  5000 *         3780 7c3ff4…         174 heal… hu5.10
# ℹ 11 more variables: index_cg &lt;int&gt;, meth_pattern &lt;chr&gt;, num_meth &lt;int&gt;,
#   total &lt;int&gt;, alpha &lt;dbl&gt;, read_start &lt;int&gt;, read_end &lt;int&gt;,
#   cancer_group &lt;chr&gt;, num_reads_in_bin &lt;int&gt;, var_alpha &lt;dbl&gt;,
#   var_alpha_per_sample_per_bin &lt;dbl&gt;</code></pre>
<ul>
<li>oh if reads overlap each other they’re not visible</li>
</ul>
<pre class="r"><code>overlap_bins_reads %&gt;% filter(bin_num == 3780, sample == &quot;hu5.10&quot;) %&gt;%
  ggplot() +
  geom_segment(aes(x = read_start, xend = read_end, y = alpha, yend = alpha, colour = read_id, group = sample)) +
  geom_point(data = filter(bin_cpgs, sample == &quot;hu5.10&quot;), aes(x = bin_cpgs, y = alpha, shape = site_is_unmeth, colour = sample, group = sample)) +
  scale_shape_manual(values = c(`TRUE` = 1, `FALSE` = 16),
                     name = &quot;Meth status&quot;, labels = c(`TRUE` = &quot;Unmeth&quot;, `FALSE` = &quot;Meth&quot;)) +
  labs(title = &quot;bin 3780: hu5.10&quot;)</code></pre>
<p><img src="figure/checkAlphaAndBetaMatch.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" />
* need it to have the circle colours as well - read id in the bin_cpg *
but also maybe making the line thicker and then opaque would help?</p>
<pre class="r"><code>overlap_bins_reads %&gt;% filter(bin_num == 3780, sample == &quot;hu5.10&quot;) %&gt;%
  ggplot() +
  geom_segment(aes(x = read_start, xend = read_end, y = alpha, yend = alpha, colour = sample, group = sample), alpha = 0.3) +
  geom_point(data = filter(bin_cpgs, sample == &quot;hu5.10&quot;), aes(x = bin_cpgs, y = alpha, shape = site_is_unmeth, colour = sample, group = sample)) +
  scale_shape_manual(values = c(`TRUE` = 1, `FALSE` = 16),
                     name = &quot;Meth status&quot;, labels = c(`TRUE` = &quot;Unmeth&quot;, `FALSE` = &quot;Meth&quot;)) +
  labs(title = &quot;bin 3780: hu5.10&quot;)</code></pre>
<p><img src="figure/checkAlphaAndBetaMatch.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" />
* that shows it a little bit</p>
<pre class="r"><code>overlap_bins_reads %&gt;% filter(bin_num == 3780) %&gt;%
  ggplot() +
  geom_segment(aes(x = read_start, xend = read_end, y = alpha, yend = alpha, colour = sample, group = sample), alpha = 0.2) +
  geom_point(data = bin_cpgs, aes(x = bin_cpgs, y = alpha, shape = site_is_unmeth, colour = sample, group = sample)) +
  scale_shape_manual(values = c(`TRUE` = 1, `FALSE` = 16),
                     name = &quot;Meth status&quot;, labels = c(`TRUE` = &quot;Unmeth&quot;, `FALSE` = &quot;Meth&quot;)) +
  labs(title = &quot;bin 3780&quot;) +
  facet_wrap(~type, nrow = 2)</code></pre>
<p><img src="figure/checkAlphaAndBetaMatch.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>c(bins_20_highest_var = overlap_bins_reads %&gt;% .[order(.$var_alpha, decreasing = TRUE),] %&gt;% .$bin_num %&gt;% unique() %&gt;% .[1:20],
bins_20_lowest_var = overlap_bins_reads %&gt;% .[order(.$var_alpha, decreasing = FALSE),] %&gt;% .$bin_num %&gt;% unique() %&gt;% .[1:20])</code></pre>
<pre><code> bins_20_highest_var1  bins_20_highest_var2  bins_20_highest_var3 
                 2377                  9598                  3261 
 bins_20_highest_var4  bins_20_highest_var5  bins_20_highest_var6 
                 2224                  6883                  2119 
 bins_20_highest_var7  bins_20_highest_var8  bins_20_highest_var9 
                 3387                  6878                  2122 
bins_20_highest_var10 bins_20_highest_var11 bins_20_highest_var12 
                 2129                  5217                  6840 
bins_20_highest_var13 bins_20_highest_var14 bins_20_highest_var15 
                 2577                  6861                  3065 
bins_20_highest_var16 bins_20_highest_var17 bins_20_highest_var18 
                 3052                  3174                  3308 
bins_20_highest_var19 bins_20_highest_var20   bins_20_lowest_var1 
                 2130                  2226                  2167 
  bins_20_lowest_var2   bins_20_lowest_var3   bins_20_lowest_var4 
                 4072                  9265                 10058 
  bins_20_lowest_var5   bins_20_lowest_var6   bins_20_lowest_var7 
                 9413                  8516                  4375 
  bins_20_lowest_var8   bins_20_lowest_var9  bins_20_lowest_var10 
                 9243                  9960                  7568 
 bins_20_lowest_var11  bins_20_lowest_var12  bins_20_lowest_var13 
                 9261                 10046                 10092 
 bins_20_lowest_var14  bins_20_lowest_var15  bins_20_lowest_var16 
                10056                  6284                  9340 
 bins_20_lowest_var17  bins_20_lowest_var18  bins_20_lowest_var19 
                 9428                  5713                  8796 
 bins_20_lowest_var20 
                 5686 </code></pre>
<ul>
<li>distribution of alpha by bin</li>
</ul>
<pre class="r"><code>high_var_bins_20 &lt;- overlap_bins_reads %&gt;% .[order(.$var_alpha, decreasing = TRUE),] %&gt;% .$bin_num %&gt;% unique() 
low_var_bins_20 &lt;- overlap_bins_reads %&gt;% .[order(.$var_alpha, decreasing = FALSE),] %&gt;% .$bin_num %&gt;% unique() </code></pre>
<pre class="r"><code>overlap_bins_reads %&gt;% filter(bin_num %in% high_var_bins_20[1:20]) %&gt;%
  ggplot(aes(x = alpha, colour = as.factor(bin_num))) +
  geom_density() +
  labs(title = &quot;Distribution of alpha in bins with highest var&quot;)</code></pre>
<p><img src="figure/checkAlphaAndBetaMatch.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>overlap_bins_reads %&gt;% filter(bin_num %in% high_var_bins_20[1:20]) %&gt;%
  ggplot(aes(x = alpha, colour = as.factor(bin_num))) +
  geom_density() +
  labs(title = &quot;Distribution of alpha in bins with highest var&quot;) +
  facet_wrap(~type)</code></pre>
<p><img src="figure/checkAlphaAndBetaMatch.Rmd/unnamed-chunk-14-2.png" width="672" style="display: block; margin: auto;" />
* something weird going on in one the bins in the healthy samples *
would have been nice if there was a distinct difference between the luad
and healthy * but also this is highest var bins overall - not
necessarily bins with high var for either * but I guess what we want
would be bins with low var individually - but the meth is different</p>
<pre class="r"><code>overlap_bins_reads %&gt;% filter(bin_num %in% low_var_bins_20[1:20]) %&gt;%
  ggplot(aes(x = alpha, colour = as.factor(bin_num))) +
  geom_density() +
  labs(title = &quot;Distribution of alpha in bins with lowest var&quot;)</code></pre>
<p><img src="figure/checkAlphaAndBetaMatch.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>overlap_bins_reads %&gt;% filter(bin_num %in% low_var_bins_20[1:20]) %&gt;%
  ggplot(aes(x = alpha, colour = as.factor(bin_num))) +
  geom_density() +
  labs(title = &quot;Distribution of alpha in bins with lowest var&quot;) +
  facet_wrap(~type)</code></pre>
<p><img src="figure/checkAlphaAndBetaMatch.Rmd/unnamed-chunk-15-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>overlap_bins_reads %&gt;% filter(bin_num %in% low_var_bins_20[1:10]) %&gt;%
  ggplot(aes(x = alpha, colour = as.factor(bin_num))) +
  geom_density() +
  labs(title = &quot;Distribution of alpha in bins with lowest var&quot;)</code></pre>
<p><img src="figure/checkAlphaAndBetaMatch.Rmd/unnamed-chunk-15-3.png" width="672" style="display: block; margin: auto;" />
* this is nice</p>
<ul>
<li>adding in betas</li>
</ul>
<pre class="r"><code>match_names &lt;- cbind(sample_names = unique(overlap_bins_reads$sample)[c(12,8,13,3,9,10,1,2,5,6,11,4,7)], bed_names = unique(bed_betas$sample)) %&gt;% data.frame()

bed_betas$bed_sample &lt;- bed_betas$sample
bed_betas &lt;- bed_betas %&gt;% mutate(sample = match_names[match(.$bed_sample, match_names$bed_names), &quot;sample_names&quot;])</code></pre>
<pre class="r"><code>overlap_bins_reads %&gt;% filter(bin_num == 3780) %&gt;%
  ggplot() +
  geom_segment(aes(x = read_start, xend = read_end, y = alpha, yend = alpha, colour = sample, group = sample), alpha = 0.2) +
  geom_point(data = bin_cpgs, aes(x = bin_cpgs, y = alpha, shape = site_is_unmeth, colour = sample, group = sample)) +
  scale_shape_manual(values = c(`TRUE` = 1, `FALSE` = 16),
                     name = &quot;Meth status&quot;, labels = c(`TRUE` = &quot;Unmeth&quot;, `FALSE` = &quot;Meth&quot;)) +
  geom_line(data = filter(bed_betas, start &gt;= min(bin_cpgs$bin_cpgs), start &lt;= max(bin_cpgs$bin_cpgs)),
            aes(x = start, y = beta, colour = sample, group = sample),
            linetype = &quot;dashed&quot;) +
  labs(title = &quot;bin 3780&quot;) +
  facet_wrap(~type, nrow = 2)</code></pre>
<p><img src="figure/checkAlphaAndBetaMatch.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>overlap_bins_reads %&gt;% filter(bin_num == 3780, sample == &quot;hu5.10&quot;) %&gt;%
  ggplot() +
  geom_segment(aes(x = read_start, xend = read_end, y = alpha, yend = alpha, colour = sample, group = sample), alpha = 0.2) +
  geom_point(data = filter(bin_cpgs, sample == &quot;hu5.10&quot;), aes(x = bin_cpgs, y = alpha, shape = site_is_unmeth, colour = sample, group = sample)) +
  scale_shape_manual(values = c(`TRUE` = 1, `FALSE` = 16),
                     name = &quot;Meth status&quot;, labels = c(`TRUE` = &quot;Unmeth&quot;, `FALSE` = &quot;Meth&quot;)) +
  geom_line(data = filter(bed_betas, sample == &quot;hu5.10&quot;, start &gt;= min(bin_cpgs$bin_cpgs), start &lt;= max(bin_cpgs$bin_cpgs)),
            aes(x = start, y = beta, colour = sample, group = sample),
            linetype = &quot;dashed&quot;) +
  labs(title = &quot;bin 3780: hu5.10&quot;) +
  scale_x_continuous(limits = c(18896000, 18897000))</code></pre>
<pre><code>Warning: Removed 2 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
<pre><code>Warning: Removed 2 rows containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
<pre><code>Warning: Removed 6 rows containing missing values or values outside the scale range
(`geom_line()`).</code></pre>
<p><img src="figure/checkAlphaAndBetaMatch.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>overlap_bins_reads %&gt;% filter(bin_num == 3780, sample == &quot;hu5.10&quot;) </code></pre>
<pre><code># A tibble: 10 × 21
   seqnames   start    end width strand bin_num read_id read_length type  sample
   &lt;fct&gt;      &lt;int&gt;  &lt;int&gt; &lt;int&gt; &lt;fct&gt;    &lt;int&gt; &lt;chr&gt;         &lt;int&gt; &lt;chr&gt; &lt;chr&gt; 
 1 chr22     1.89e7 1.89e7  5000 *         3780 0d26cd…         519 heal… hu5.10
 2 chr22     1.89e7 1.89e7  5000 *         3780 f57e62…         659 heal… hu5.10
 3 chr22     1.89e7 1.89e7  5000 *         3780 a323a4…         298 heal… hu5.10
 4 chr22     1.89e7 1.89e7  5000 *         3780 d4aa41…         184 heal… hu5.10
 5 chr22     1.89e7 1.89e7  5000 *         3780 4d7a8e…         254 heal… hu5.10
 6 chr22     1.89e7 1.89e7  5000 *         3780 42b151…         296 heal… hu5.10
 7 chr22     1.89e7 1.89e7  5000 *         3780 d855ca…         202 heal… hu5.10
 8 chr22     1.89e7 1.89e7  5000 *         3780 b62fe6…         366 heal… hu5.10
 9 chr22     1.89e7 1.89e7  5000 *         3780 7c3ff4…         174 heal… hu5.10
10 chr22     1.89e7 1.89e7  5000 *         3780 d549e1…         283 heal… hu5.10
# ℹ 11 more variables: index_cg &lt;int&gt;, meth_pattern &lt;chr&gt;, num_meth &lt;int&gt;,
#   total &lt;int&gt;, alpha &lt;dbl&gt;, read_start &lt;int&gt;, read_end &lt;int&gt;,
#   cancer_group &lt;chr&gt;, num_reads_in_bin &lt;int&gt;, var_alpha &lt;dbl&gt;,
#   var_alpha_per_sample_per_bin &lt;dbl&gt;</code></pre>
<pre class="r"><code>filter(bed_betas, sample == &quot;hu5.10&quot;, start &gt;= min(bin_cpgs$bin_cpgs), start &lt;= max(bin_cpgs$bin_cpgs))</code></pre>
<pre><code>   seqnames    start      end width strand name sample    type beta bed_sample
1     chr22 18891008 18891008     1      *  0.5 hu5.10 healthy  0.5     hu5.10
2     chr22 18892444 18892444     1      *  1.0 hu5.10 healthy  1.0     hu5.10
3     chr22 18892835 18892835     1      *  1.0 hu5.10 healthy  1.0     hu5.10
4     chr22 18893016 18893016     1      *  1.0 hu5.10 healthy  1.0     hu5.10
5     chr22 18895995 18895995     1      *  1.0 hu5.10 healthy  1.0     hu5.10
6     chr22 18895996 18895996     1      *  0.0 hu5.10 healthy  0.0     hu5.10
7     chr22 18896177 18896177     1      *  1.0 hu5.10 healthy  1.0     hu5.10
8     chr22 18896178 18896178     1      *  1.0 hu5.10 healthy  1.0     hu5.10
9     chr22 18896209 18896209     1      *  1.0 hu5.10 healthy  1.0     hu5.10
10    chr22 18896210 18896210     1      *  1.0 hu5.10 healthy  1.0     hu5.10
11    chr22 18896219 18896219     1      *  1.0 hu5.10 healthy  1.0     hu5.10
12    chr22 18896220 18896220     1      *  1.0 hu5.10 healthy  1.0     hu5.10
13    chr22 18896226 18896226     1      *  0.0 hu5.10 healthy  0.0     hu5.10
14    chr22 18896246 18896246     1      *  1.0 hu5.10 healthy  1.0     hu5.10
15    chr22 18896247 18896247     1      *  1.0 hu5.10 healthy  1.0     hu5.10
16    chr22 18896304 18896304     1      *  1.0 hu5.10 healthy  1.0     hu5.10
17    chr22 18896305 18896305     1      *  1.0 hu5.10 healthy  1.0     hu5.10
18    chr22 18896602 18896602     1      *  0.0 hu5.10 healthy  0.0     hu5.10
19    chr22 18896751 18896751     1      *  0.0 hu5.10 healthy  0.0     hu5.10
20    chr22 18896782 18896782     1      *  0.0 hu5.10 healthy  0.0     hu5.10
21    chr22 18896832 18896832     1      *  0.0 hu5.10 healthy  0.0     hu5.10
22    chr22 18896853 18896853     1      *  1.0 hu5.10 healthy  1.0     hu5.10</code></pre>
<pre class="r"><code>all_samples %&gt;% filter(sample == &quot;hu5.10&quot;, read_id %in% filter(overlap_bins_reads, bin_num == 3780, sample == &quot;hu5.10&quot;)$read_id)</code></pre>
<pre><code>                                read_id seqnames    start      end read_length
1  0d26cd7f-14c3-4877-9682-dd3f19d9a54c    chr22 18895773 18896292         519
2  f57e628a-7b6d-4d3f-92ff-a7ea08d588ed    chr22 18895877 18896536         659
3  a323a461-a8dc-4f05-a5b0-88da3eef09be    chr22 18896258 18896556         298
4  d4aa4141-9433-45c3-b275-5d6160ff2e05    chr22 18896269 18896453         184
5  4d7a8e95-2bed-4d47-a9ee-3fddc339e0a2    chr22 18896282 18896536         254
6  42b151bd-0062-4542-b99e-10253ee72cbe    chr22 18896293 18896589         296
7  d855ca96-9e2f-4568-8782-5f4b8d244137    chr22 18896293 18896495         202
8  b62fe613-d079-4cc6-9ba7-57c4cea632e1    chr22 18896372 18896738         366
9  7c3ff42f-7d65-4ff6-98b4-2dedfd25191f    chr22 18896473 18896647         174
10 d549e183-ed12-4d5d-b6c2-40420a7c8739    chr22 18896652 18896935         283
      type sample index_cg meth_pattern
1  healthy hu5.10    99859       TCCCCC
2  healthy hu5.10    99859      CCCCTCC
3  healthy hu5.10    99865            C
4  healthy hu5.10    99865            C
5  healthy hu5.10    99865            T
6  healthy hu5.10    99865            C
7  healthy hu5.10    99865            C
8  healthy hu5.10    99866            T
9  healthy hu5.10    99866            T
10 healthy hu5.10    99867         TTTC
                               cpg_positions
1        99859,99860,99861,99862,99863,99864
2  99859,99860,99861,99862,99863,99864,99865
3                                      99865
4                                      99865
5                                      99865
6                                      99865
7                                      99865
8                                      99866
9                                      99866
10                   99867,99868,99869,99870
                                                  genom_positions num_meth
1           18895995,18896177,18896209,18896219,18896226,18896246        5
2  18895995,18896177,18896209,18896219,18896226,18896246,18896304        6
3                                                        18896304        1
4                                                        18896304        1
5                                                        18896304        0
6                                                        18896304        1
7                                                        18896304        1
8                                                        18896601        0
9                                                        18896601        0
10                            18896751,18896782,18896832,18896853        1
   total     alpha read_start read_end
1      6 0.8333333   18895773 18896292
2      7 0.8571429   18895877 18896536
3      1 1.0000000   18896258 18896556
4      1 1.0000000   18896269 18896453
5      1 0.0000000   18896282 18896536
6      1 1.0000000   18896293 18896589
7      1 1.0000000   18896293 18896495
8      1 0.0000000   18896372 18896738
9      1 0.0000000   18896473 18896647
10     4 0.2500000   18896652 18896935</code></pre>
<ul>
<li>maybe there’s not actually a problem and it’s just a messy
plot?</li>
<li>but there was something funny</li>
</ul>
</div>
<div id="something-weird-and-wrong" class="section level1">
<h1>something weird and wrong</h1>
<ul>
<li>only 2 reads overlapping this cpg site</li>
<li>on 1 read, the site is meth, the other the site is unmeth</li>
<li>beta should be 0.5, but it’s 1</li>
</ul>
<pre class="r"><code>bin_cpgs %&gt;% filter(bin_cpgs == 18895995, sample == &quot;hu5.10&quot;)</code></pre>
<pre><code>  bin_cpgs meth site_is_unmeth     alpha sample    type
1 18895995    T           TRUE 0.8333333 hu5.10 healthy
2 18895995    C          FALSE 0.8571429 hu5.10 healthy</code></pre>
<pre class="r"><code>bed_betas %&gt;% filter(start == 18895995, sample == &quot;hu5.10&quot;)</code></pre>
<pre><code>  seqnames    start      end width strand name sample    type beta bed_sample
1    chr22 18895995 18895995     1      *    1 hu5.10 healthy    1     hu5.10</code></pre>
<pre class="r"><code>all_samples %&gt;% filter(sample == &quot;hu5.10&quot;) %&gt;% .[grep(18895995, .$genom_positions),]</code></pre>
<pre><code>                                  read_id seqnames    start      end
3795 0d26cd7f-14c3-4877-9682-dd3f19d9a54c    chr22 18895773 18896292
3796 f57e628a-7b6d-4d3f-92ff-a7ea08d588ed    chr22 18895877 18896536
     read_length    type sample index_cg meth_pattern
3795         519 healthy hu5.10    99859       TCCCCC
3796         659 healthy hu5.10    99859      CCCCTCC
                                 cpg_positions
3795       99859,99860,99861,99862,99863,99864
3796 99859,99860,99861,99862,99863,99864,99865
                                                    genom_positions num_meth
3795          18895995,18896177,18896209,18896219,18896226,18896246        5
3796 18895995,18896177,18896209,18896219,18896226,18896246,18896304        6
     total     alpha read_start read_end
3795     6 0.8333333   18895773 18896292
3796     7 0.8571429   18895877 18896536</code></pre>
<pre class="r"><code>indiv_reads %&gt;% filter(ref_position == 18895995)</code></pre>
<pre><code>                                read_id forward_read_position ref_position
                                 &lt;char&gt;                 &lt;int&gt;        &lt;int&gt;
1: 0d26cd7f-14c3-4877-9682-dd3f19d9a54c                   296     18895995
    chrom mod_strand ref_strand ref_mod_strand fw_soft_clipped_start
   &lt;char&gt;     &lt;char&gt;     &lt;char&gt;         &lt;char&gt;                 &lt;int&gt;
1:  chr22          +          -              -                     0
   fw_soft_clipped_end alignment_start alignment_end read_length call_prob
                 &lt;int&gt;           &lt;int&gt;         &lt;int&gt;       &lt;int&gt;     &lt;num&gt;
1:                   0        18895773      18896292         519 0.7246094
   call_code base_qual ref_kmer query_kmer canonical_base modified_primary_base
      &lt;char&gt;     &lt;int&gt;   &lt;char&gt;     &lt;char&gt;         &lt;char&gt;                &lt;char&gt;
1:         -       255        .      TACGT              C                     C
     fail inferred within_alignment  flag seqnames    start      end
   &lt;lgcl&gt;   &lt;lgcl&gt;           &lt;lgcl&gt; &lt;int&gt;   &lt;char&gt;    &lt;int&gt;    &lt;int&gt;
1:   TRUE    FALSE             TRUE    16    chr22 18895773 18896292</code></pre>
<pre class="r"><code>indiv_reads %&gt;% filter(read_id %in% c(&quot;0d26cd7f-14c3-4877-9682-dd3f19d9a54c&quot;, &quot;f57e628a-7b6d-4d3f-92ff-a7ea08d588ed&quot;)) %&gt;% .[c(1,7),]</code></pre>
<pre><code>                                read_id forward_read_position ref_position
                                 &lt;char&gt;                 &lt;int&gt;        &lt;int&gt;
1: 0d26cd7f-14c3-4877-9682-dd3f19d9a54c                   296     18895995
2: f57e628a-7b6d-4d3f-92ff-a7ea08d588ed                   117     18895994
    chrom mod_strand ref_strand ref_mod_strand fw_soft_clipped_start
   &lt;char&gt;     &lt;char&gt;     &lt;char&gt;         &lt;char&gt;                 &lt;int&gt;
1:  chr22          +          -              -                     0
2:  chr22          +          +              +                     0
   fw_soft_clipped_end alignment_start alignment_end read_length call_prob
                 &lt;int&gt;           &lt;int&gt;         &lt;int&gt;       &lt;int&gt;     &lt;num&gt;
1:                   0        18895773      18896292         519 0.7246094
2:                   0        18895877      18896536         659 0.9433594
   call_code base_qual ref_kmer query_kmer canonical_base modified_primary_base
      &lt;char&gt;     &lt;int&gt;   &lt;char&gt;     &lt;char&gt;         &lt;char&gt;                &lt;char&gt;
1:         -       255        .      TACGT              C                     C
2:         m       255        .      GACGT              C                     C
     fail inferred within_alignment  flag seqnames    start      end
   &lt;lgcl&gt;   &lt;lgcl&gt;           &lt;lgcl&gt; &lt;int&gt;   &lt;char&gt;    &lt;int&gt;    &lt;int&gt;
1:   TRUE    FALSE             TRUE    16    chr22 18895773 18896292
2:   TRUE    FALSE             TRUE     0    chr22 18895877 18896536</code></pre>
<ul>
<li>so the unmeth read has prob &lt; 0.8, and fail = TRUE</li>
<li>but the other read also has fail = TRUE</li>
<li>I guess if you were to say that if it is too low prob it becomes the
other one that would fix this scenario</li>
</ul>
<p><a href="https://nanoporetech.github.io/modkit/faq.html"
class="uri">https://nanoporetech.github.io/modkit/faq.html</a> - * the
probability section for the calls</p>
</div>
<div id="check-out-the-bedgraph" class="section level1">
<h1>check out the bedgraph</h1>
<ul>
<li>could there be something wrong with the bedgraph??</li>
</ul>
<p>To extract (stranded) methylation information from the
mod_mapping.bam files, we used modbam2bed (<a
href="https://github.com/epi2me-labs/modbam2bed"
class="uri">https://github.com/epi2me-labs/modbam2bed</a>) v.0.4.5,
specifying a minimum probability threshold of 0.667, and filtering out
positions with 0 confident reads using awk. The full command line was
“modbam2bed –cpg -t 4 -a 0.333 -b 0.667 | awk ’($5&gt;0){print} &gt;
out.bed.” All coordinates are in GRCh38 and are 0-based. These files are
named “<em>.5mC.cut0.667.hg38.bed.gz.” Column 11 corresponds to the
percent of reads methylated. Modbam2bed does not provide a column for
the actual number of reads that this percentage is based on, but it can
be calculated from the other columns. readCount=(col5</em>col10)/1000.
We also provide a simple bedgraph with just the methylation fraction
(beta) values in files named “*cut0.667.hg38.sorted.bedgraph.gz.”</p>
<ul>
<li>could also be the bedgraph filtering with modbam2bed is different to
the modkit filtering</li>
<li>can modkit make bed files?</li>
<li>it does: run that /modkit/bed_modkit.sbatch</li>
<li>hopefully they will then match up as they will have used the same
parameters</li>
</ul>
</div>
<div id="bed-file-from-modkit" class="section level1">
<h1>bed file from modkit</h1>
<pre class="r"><code>list.files(&quot;/researchers/caitlin.page/cf_nano/modkit_output/bed&quot;)</code></pre>
<pre><code> [1] &quot;hu5.10.bed&quot;        &quot;hu5.10.bed.gz.tbi&quot; &quot;hu5.11.bed&quot;       
 [4] &quot;ispro.bc1.bed&quot;     &quot;ispro.bc10.bed&quot;    &quot;ispro.bc11.bed&quot;   
 [7] &quot;ispro.bc2.bed&quot;     &quot;ispro.bc3.bed&quot;     &quot;ispro.bc4.bed&quot;    
[10] &quot;ispro.bc5.bed&quot;     &quot;ispro.bc8.bed&quot;     &quot;ispro.bc9.bed&quot;    
[13] &quot;ispro.s1.bed&quot;      &quot;sort_hu5.10.bed&quot;  </code></pre>
<pre class="r"><code>modkit_bed_hu5.10 &lt;- read.table(&quot;/researchers/caitlin.page/cf_nano/modkit_output/bed/hu5.10.bed&quot;)</code></pre>
<p><a
href="https://nanoporetech.github.io/modkit/intro_pileup.html">colnames
of bed file</a> positions are 0 based</p>
<pre class="r"><code>colnames(modkit_bed_hu5.10) &lt;- c(&quot;seqnames&quot;, &quot;start&quot;, &quot;end&quot;, &quot;mod_base_code&quot;, &quot;score_same_as_coverage&quot;, &quot;strand&quot;, &quot;start_pos&quot;, &quot;end_pos&quot;, &quot;colour&quot;, &quot;coverage&quot;, &quot;beta&quot;, &quot;n_mod&quot;, &quot;n_canon&quot;, &quot;n_other_mod&quot;, &quot;n_delete&quot;, &quot;n_fail&quot;, &quot;n_diff&quot;, &quot;n_nocall&quot;)</code></pre>
<pre class="r"><code>unique(modkit_bed_hu5.10$mod_base_code)</code></pre>
<pre><code>[1] &quot;m&quot;</code></pre>
<pre class="r"><code>summary(modkit_bed_hu5.10$coverage)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  1.000   1.000   1.000   1.065   1.000 843.000 </code></pre>
<pre class="r"><code>summary(modkit_bed_hu5.10$n_mod)</code></pre>
<pre><code>     Min.   1st Qu.    Median      Mean   3rd Qu.      Max. 
  0.00000   0.00000   0.00000   0.02619   0.00000 104.00000 </code></pre>
<pre class="r"><code>summary(modkit_bed_hu5.10$n_canon)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  0.000   1.000   1.000   1.039   1.000 843.000 </code></pre>
<pre class="r"><code>modkit_bed_hu5.10_chr22 &lt;- modkit_bed_hu5.10 %&gt;% filter(seqnames == &quot;chr22&quot;)</code></pre>
<ul>
<li>here we go let’s find that funky position again 18895995</li>
</ul>
<pre class="r"><code>modkit_bed_hu5.10_chr22 %&gt;% filter(start &gt; 18895980) %&gt;% head()</code></pre>
<pre><code>  seqnames    start      end mod_base_code score_same_as_coverage strand
1    chr22 18895981 18895982             m                      1      -
2    chr22 18895992 18895993             m                      1      -
3    chr22 18896011 18896012             m                      1      -
4    chr22 18896022 18896023             m                      1      -
5    chr22 18896024 18896025             m                      1      +
6    chr22 18896041 18896042             m                      1      -
  start_pos  end_pos  colour coverage beta n_mod n_canon n_other_mod n_delete
1  18895981 18895982 255,0,0        1    0     0       1           0        0
2  18895992 18895993 255,0,0        1    0     0       1           0        0
3  18896011 18896012 255,0,0        1    0     0       1           0        0
4  18896022 18896023 255,0,0        1    0     0       1           0        0
5  18896024 18896025 255,0,0        1    0     0       1           0        0
6  18896041 18896042 255,0,0        1    0     0       1           0        0
  n_fail n_diff n_nocall
1      0      0        0
2      0      0        0
3      0      0        0
4      0      0        0
5      0      0        0
6      0      0        0</code></pre>
<ul>
<li><p>and of course here it doesn’t even exist as a position</p></li>
<li><p>because we need another problem</p></li>
<li><p>and the closest one to it is 2 bases off and is canon - beta is
0</p></li>
<li><p>so maybe the positions are off - but still means the answer is
wrong</p></li>
<li><p>oh wait it doesn’t automatically restrict to just cpgs (b/c I
didn’t do the cpg bit with the fasta - should try that again)</p></li>
</ul>
<pre class="r"><code>find_overlaps(as_granges(modkit_bed_hu5.10_chr22), as_granges(cg_sites)) %&gt;% data.frame() %&gt;% head()</code></pre>
<pre><code>  seqnames    start      end width strand mod_base_code score_same_as_coverage
1    chr22 10521180 10521181     2      -             m                      1
2    chr22 10522480 10522481     2      +             m                      1
3    chr22 10522482 10522483     2      +             m                      1
4    chr22 10522500 10522501     2      +             m                      1
5    chr22 10522648 10522649     2      +             m                      1
6    chr22 10527072 10527073     2      -             m                      1
  start_pos  end_pos  colour coverage beta n_mod n_canon n_other_mod n_delete
1  10521180 10521181 255,0,0        1    0     0       1           0        0
2  10522480 10522481 255,0,0        1  100     1       0           0        0
3  10522482 10522483 255,0,0        1    0     0       1           0        0
4  10522500 10522501 255,0,0        1  100     1       0           0        0
5  10522648 10522649 255,0,0        1    0     0       1           0        0
6  10527072 10527073 255,0,0        1    0     0       1           0        0
  n_fail n_diff n_nocall            pos seq index
1      0      0        0 chr22-10521180  CG    48
2      0      0        0 chr22-10522481  CG    53
3      0      0        0 chr22-10522481  CG    53
4      0      0        0 chr22-10522501  CG    54
5      0      0        0 chr22-10522649  CG    56
6      0      0        0 chr22-10527071  CG   299</code></pre>
<ul>
<li>so that’s what I need but still that position is missing</li>
<li>all the positions in extract calls should exist in the bed file I
don’t get why they don’t</li>
<li>extract calls does same filtering as pileup</li>
<li>unless i did extract full accidentally??</li>
</ul>
<pre class="r"><code>find_overlaps(as_granges(modkit_bed_hu5.10_chr22), as_granges(cg_sites)) %&gt;% data.frame() %&gt;%
  ggplot(aes(x = beta)) +
  geom_density()</code></pre>
<p><img src="figure/checkAlphaAndBetaMatch.Rmd/unnamed-chunk-32-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>indiv_reads %&gt;% filter(ref_position == 18895995)</code></pre>
<pre><code>                                read_id forward_read_position ref_position
                                 &lt;char&gt;                 &lt;int&gt;        &lt;int&gt;
1: 0d26cd7f-14c3-4877-9682-dd3f19d9a54c                   296     18895995
    chrom mod_strand ref_strand ref_mod_strand fw_soft_clipped_start
   &lt;char&gt;     &lt;char&gt;     &lt;char&gt;         &lt;char&gt;                 &lt;int&gt;
1:  chr22          +          -              -                     0
   fw_soft_clipped_end alignment_start alignment_end read_length call_prob
                 &lt;int&gt;           &lt;int&gt;         &lt;int&gt;       &lt;int&gt;     &lt;num&gt;
1:                   0        18895773      18896292         519 0.7246094
   call_code base_qual ref_kmer query_kmer canonical_base modified_primary_base
      &lt;char&gt;     &lt;int&gt;   &lt;char&gt;     &lt;char&gt;         &lt;char&gt;                &lt;char&gt;
1:         -       255        .      TACGT              C                     C
     fail inferred within_alignment  flag seqnames    start      end
   &lt;lgcl&gt;   &lt;lgcl&gt;           &lt;lgcl&gt; &lt;int&gt;   &lt;char&gt;    &lt;int&gt;    &lt;int&gt;
1:   TRUE    FALSE             TRUE    16    chr22 18895773 18896292</code></pre>
<pre class="r"><code>indiv_reads %&gt;% filter(ref_position &gt; 18895980, ref_position &lt; 18896100)</code></pre>
<pre><code>                                read_id forward_read_position ref_position
                                 &lt;char&gt;                 &lt;int&gt;        &lt;int&gt;
1: 0d26cd7f-14c3-4877-9682-dd3f19d9a54c                   296     18895995
2: f57e628a-7b6d-4d3f-92ff-a7ea08d588ed                   117     18895994
    chrom mod_strand ref_strand ref_mod_strand fw_soft_clipped_start
   &lt;char&gt;     &lt;char&gt;     &lt;char&gt;         &lt;char&gt;                 &lt;int&gt;
1:  chr22          +          -              -                     0
2:  chr22          +          +              +                     0
   fw_soft_clipped_end alignment_start alignment_end read_length call_prob
                 &lt;int&gt;           &lt;int&gt;         &lt;int&gt;       &lt;int&gt;     &lt;num&gt;
1:                   0        18895773      18896292         519 0.7246094
2:                   0        18895877      18896536         659 0.9433594
   call_code base_qual ref_kmer query_kmer canonical_base modified_primary_base
      &lt;char&gt;     &lt;int&gt;   &lt;char&gt;     &lt;char&gt;         &lt;char&gt;                &lt;char&gt;
1:         -       255        .      TACGT              C                     C
2:         m       255        .      GACGT              C                     C
     fail inferred within_alignment  flag seqnames    start      end
   &lt;lgcl&gt;   &lt;lgcl&gt;           &lt;lgcl&gt; &lt;int&gt;   &lt;char&gt;    &lt;int&gt;    &lt;int&gt;
1:   TRUE    FALSE             TRUE    16    chr22 18895773 18896292
2:   TRUE    FALSE             TRUE     0    chr22 18895877 18896536</code></pre>
<ul>
<li>oh so it should be 0 - but they should be the same position -
ref_position is 0 based</li>
</ul>
<pre class="r"><code>cg_sites %&gt;% filter(start &gt; 18895980) %&gt;% head()</code></pre>
<pre><code>             pos seqnames    start      end width seq index
1 chr22-18895995    chr22 18895995 18895996     2  CG 99859
2 chr22-18896177    chr22 18896177 18896178     2  CG 99860
3 chr22-18896209    chr22 18896209 18896210     2  CG 99861
4 chr22-18896219    chr22 18896219 18896220     2  CG 99862
5 chr22-18896226    chr22 18896226 18896227     2  CG 99863
6 chr22-18896246    chr22 18896246 18896247     2  CG 99864</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.1 (2024-06-14)
Platform: x86_64-pc-linux-gnu
Running under: Red Hat Enterprise Linux 9.5 (Plow)

Matrix products: default
BLAS/LAPACK: FlexiBLAS OPENBLAS-OPENMP;  LAPACK version 3.9.0

locale:
 [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
 [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8   
 [7] LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       

time zone: Australia/Melbourne
tzcode source: system (glibc)

attached base packages:
[1] stats4    stats     graphics  grDevices datasets  utils     methods  
[8] base     

other attached packages:
 [1] ggplot2_3.5.2        dplyr_1.1.4          tidyr_1.3.1         
 [4] plyranges_1.26.0     GenomicRanges_1.58.0 GenomeInfoDb_1.42.3 
 [7] IRanges_2.40.1       S4Vectors_0.44.0     BiocGenerics_0.52.0 
[10] data.table_1.17.8    workflowr_1.7.1     

loaded via a namespace (and not attached):
 [1] tidyselect_1.2.1            farver_2.1.2               
 [3] Biostrings_2.74.1           bitops_1.0-9               
 [5] fastmap_1.2.0               RCurl_1.98-1.17            
 [7] GenomicAlignments_1.42.0    promises_1.3.3             
 [9] XML_3.99-0.18               digest_0.6.37              
[11] lifecycle_1.0.4             processx_3.8.6             
[13] magrittr_2.0.3              compiler_4.4.1             
[15] rlang_1.1.6                 sass_0.4.10                
[17] tools_4.4.1                 utf8_1.2.6                 
[19] yaml_2.3.10                 rtracklayer_1.66.0         
[21] knitr_1.50                  labeling_0.4.3             
[23] S4Arrays_1.6.0              curl_6.4.0                 
[25] DelayedArray_0.32.0         RColorBrewer_1.1-3         
[27] abind_1.4-8                 BiocParallel_1.40.2        
[29] withr_3.0.2                 purrr_1.1.0                
[31] grid_4.4.1                  git2r_0.36.2               
[33] scales_1.4.0                SummarizedExperiment_1.36.0
[35] cli_3.6.5                   rmarkdown_2.29             
[37] crayon_1.5.3                generics_0.1.4             
[39] rstudioapi_0.17.1           httr_1.4.7                 
[41] rjson_0.2.23                cachem_1.1.0               
[43] stringr_1.5.1               zlibbioc_1.52.0            
[45] parallel_4.4.1              BiocManager_1.30.26        
[47] XVector_0.46.0              restfulr_0.0.16            
[49] matrixStats_1.5.0           vctrs_0.6.5                
[51] Matrix_1.7-0                jsonlite_2.0.0             
[53] callr_3.7.6                 jquerylib_0.1.4            
[55] glue_1.8.0                  codetools_0.2-20           
[57] ps_1.9.1                    stringi_1.8.7              
[59] gtable_0.3.6                later_1.4.2                
[61] BiocIO_1.16.0               UCSC.utils_1.2.0           
[63] tibble_3.3.0                pillar_1.11.0              
[65] htmltools_0.5.8.1           GenomeInfoDbData_1.2.13    
[67] R6_2.6.1                    rprojroot_2.1.0            
[69] evaluate_1.0.4              lattice_0.22-6             
[71] Biobase_2.66.0              Rsamtools_2.22.0           
[73] renv_1.1.5                  httpuv_1.6.16              
[75] bslib_0.9.0                 Rcpp_1.1.0                 
[77] SparseArray_1.6.2           whisker_0.4.1              
[79] xfun_0.52                   fs_1.6.6                   
[81] MatrixGenerics_1.18.1       getPass_0.2-4              
[83] pkgconfig_2.0.3            </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
