<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Caitlin Page" />

<meta name="date" content="2025-11-28" />

<title>correlationOfAlphas</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">3_cf_nano</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/caitlinpage/methyl_nano_cf">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">correlationOfAlphas</h1>
<h4 class="author">Caitlin Page</h4>
<h4 class="date">2025-11-28</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-12-02
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>3_cf_nano/analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250606code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250606)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250606code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250606)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomcaitlinpage3cfnanotreeefabbcf5ded5b595a08997a81b3831b4869b883dtargetblankefabbcfa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/caitlinpage/3_cf_nano/tree/efabbcf5ded5b595a08997a81b3831b4869b883d" target="_blank">efabbcf</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomcaitlinpage3cfnanotreeefabbcf5ded5b595a08997a81b3831b4869b883dtargetblankefabbcfa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/caitlinpage/3_cf_nano/tree/efabbcf5ded5b595a08997a81b3831b4869b883d" target="_blank">efabbcf</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    code/Fragmentomics_GenomBiol-main/
    Ignored:    code/dist_modkit_v0.5.0_5120ef7_candle/
    Ignored:    code/dist_modkit_v0.5.0_5120ef7_tch/
    Ignored:    code/modkit-0.5.0/
    Ignored:    code/modkit_v0.5.0_5120ef7/
    Ignored:    code/uses_biscuit/
    Ignored:    data/cfnano_paper/
    Ignored:    data/ncbi_dataset/
    Ignored:    data/rrbs/
    Ignored:    output/bedtools/
    Ignored:    output/biscuit/
    Ignored:    output/cluster_files/
    Ignored:    output/html docs - versions/
    Ignored:    output/modkit/
    Ignored:    output/nanoplot/
    Ignored:    output/research_sprint/
    Ignored:    output/samtools/
    Ignored:    output/schematics/
    Ignored:    renv.lock
    Ignored:    renv/

Unstaged changes:
    Modified:   .Rprofile
    Modified:   .gitignore
    Modified:   3_cf_nano.Rproj
    Modified:   _workflowr.yml
    Modified:   analysis/00_importantObjects.Rmd
    Modified:   analysis/01a_bedFileCorrelation.Rmd
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/correlationOfAlphas.Rmd</code>)
and HTML (<code>docs/correlationOfAlphas.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/caitlinpage/3_cf_nano/blob/efabbcf5ded5b595a08997a81b3831b4869b883d/analysis/correlationOfAlphas.Rmd" target="_blank">efabbcf</a>
</td>
<td>
Caitlin Page
</td>
<td>
2025-12-02
</td>
<td>
wflow_publish("correlationOfAlphas.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="packages" class="section level2">
<h2>Packages</h2>
<pre class="r"><code>library(BSgenome.Hsapiens.UCSC.hg38)</code></pre>
<pre><code>Loading required package: GenomeInfoDb</code></pre>
<pre><code>Loading required package: BiocGenerics</code></pre>
<pre><code>
Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    IQR, mad, sd, var, xtabs</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    anyDuplicated, aperm, append, as.data.frame, basename, cbind,
    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,
    get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,
    match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,
    Position, rank, rbind, Reduce, rownames, sapply, saveRDS, setdiff,
    table, tapply, union, unique, unsplit, which.max, which.min</code></pre>
<pre><code>Loading required package: S4Vectors</code></pre>
<pre><code>Loading required package: stats4</code></pre>
<pre><code>
Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:utils&#39;:

    findMatches</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    expand.grid, I, unname</code></pre>
<pre><code>Loading required package: IRanges</code></pre>
<pre><code>Loading required package: BSgenome</code></pre>
<pre><code>Loading required package: GenomicRanges</code></pre>
<pre><code>Loading required package: Biostrings</code></pre>
<pre><code>Loading required package: XVector</code></pre>
<pre><code>
Attaching package: &#39;Biostrings&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:base&#39;:

    strsplit</code></pre>
<pre><code>Loading required package: BiocIO</code></pre>
<pre><code>Loading required package: rtracklayer</code></pre>
<pre><code>
Attaching package: &#39;rtracklayer&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:BiocIO&#39;:

    FileForFormat</code></pre>
<pre class="r"><code>library(data.table)</code></pre>
<pre><code>
Attaching package: &#39;data.table&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:GenomicRanges&#39;:

    shift</code></pre>
<pre><code>The following object is masked from &#39;package:IRanges&#39;:

    shift</code></pre>
<pre><code>The following objects are masked from &#39;package:S4Vectors&#39;:

    first, second</code></pre>
<pre class="r"><code>library(plyranges)</code></pre>
<pre><code>
Attaching package: &#39;plyranges&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:data.table&#39;:

    between</code></pre>
<pre><code>The following object is masked from &#39;package:XVector&#39;:

    slice</code></pre>
<pre><code>The following object is masked from &#39;package:IRanges&#39;:

    slice</code></pre>
<pre><code>The following object is masked from &#39;package:stats&#39;:

    filter</code></pre>
<pre class="r"><code>library(tidyr)</code></pre>
<pre><code>
Attaching package: &#39;tidyr&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:S4Vectors&#39;:

    expand</code></pre>
<pre class="r"><code>library(stringr)
library(dplyr)</code></pre>
<pre><code>
Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:plyranges&#39;:

    between, n, n_distinct</code></pre>
<pre><code>The following objects are masked from &#39;package:data.table&#39;:

    between, first, last</code></pre>
<pre><code>The following objects are masked from &#39;package:Biostrings&#39;:

    collapse, intersect, setdiff, setequal, union</code></pre>
<pre><code>The following object is masked from &#39;package:XVector&#39;:

    slice</code></pre>
<pre><code>The following objects are masked from &#39;package:GenomicRanges&#39;:

    intersect, setdiff, union</code></pre>
<pre><code>The following object is masked from &#39;package:GenomeInfoDb&#39;:

    intersect</code></pre>
<pre><code>The following objects are masked from &#39;package:IRanges&#39;:

    collapse, desc, intersect, setdiff, slice, union</code></pre>
<pre><code>The following objects are masked from &#39;package:S4Vectors&#39;:

    first, intersect, rename, setdiff, setequal, union</code></pre>
<pre><code>The following objects are masked from &#39;package:BiocGenerics&#39;:

    combine, intersect, setdiff, union</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    filter, lag</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    intersect, setdiff, setequal, union</code></pre>
<pre class="r"><code>library(ggplot2)</code></pre>
</div>
<div id="data-objects" class="section level2">
<h2>Data Objects</h2>
<pre class="r"><code>cg_sites &lt;- cbind(data.frame(Biostrings::matchPattern(&quot;CG&quot;, BSgenome.Hsapiens.UCSC.hg38::BSgenome.Hsapiens.UCSC.hg38[[&quot;chr22&quot;]])), seqnames = &quot;chr22&quot;) %&gt;% 
  mutate(pos = paste0(seqnames, &quot;-&quot;, start)) %&gt;%
  relocate(pos, seqnames)</code></pre>
<pre><code>Warning in .local(x, row.names, optional, ...): &#39;optional&#39; argument was ignored</code></pre>
<pre class="r"><code>cg_sites$index &lt;- 1:nrow(cg_sites)</code></pre>
<pre class="r"><code>source(&quot;../code/processData.R&quot;)

for_beta &lt;- readRDS(&quot;../output/cluster_files/rrbs_alpha_beta.rds&quot;)
for_beta &lt;- for_beta %&gt;% filter(seqnames == &quot;chr22&quot;)
rrbs_for_beta_pat &lt;- pat_for_wgbs(for_beta)</code></pre>
<pre><code>`summarise()` has grouped output by &#39;seqnames&#39;, &#39;start_pos&#39;, &#39;meth_pattern&#39;,
&#39;alpha&#39;, &#39;end_pos&#39;. You can override using the `.groups` argument.</code></pre>
<pre class="r"><code>for_beta &lt;- for_beta %&gt;% group_by(start) %&gt;% mutate(coverage = n()) %&gt;% ungroup() %&gt;% data.frame()
just_beta &lt;- for_beta %&gt;% distinct(start, beta, coverage) %&gt;% .[order(.$start),]</code></pre>
<pre class="r"><code>for_beta %&gt;% 
  group_by(start) %&gt;% 
  mutate(mean_alpha = mean(alpha)) %&gt;%
  ungroup() %&gt;% 
  distinct(seqnames, start, mean_alpha, beta, coverage) %&gt;%
  ggplot(aes(x = beta, y = mean_alpha)) +
  geom_count(alpha = 0.3) +
  geom_abline(colour = &quot;red&quot;, linetype = &quot;dashed&quot;) +
  labs(title = &quot;Mean alpha and beta, size by geom_count&quot;)</code></pre>
<p><img src="figure/correlationOfAlphas.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>for_beta %&gt;% 
  group_by(start) %&gt;% 
  mutate(mean_alpha = mean(alpha)) %&gt;%
  ungroup() %&gt;% 
  distinct(seqnames, start, mean_alpha, beta, coverage) %&gt;%
  ggplot(aes(x = beta, y = mean_alpha, size = coverage)) +
  geom_point(alpha = 0.3) +
  geom_abline(colour = &quot;red&quot;, linetype = &quot;dashed&quot;) +
  labs(title = &quot;Mean alpha and beta, size by coverage at site&quot;)</code></pre>
<p><img src="figure/correlationOfAlphas.Rmd/unnamed-chunk-4-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>for_beta &lt;- for_beta %&gt;% 
  group_by(start) %&gt;% 
  mutate(mean_alpha = mean(alpha)) %&gt;%
  ungroup() %&gt;% 
  data.frame()</code></pre>
<pre class="r"><code>just_beta &lt;- just_beta %&gt;%
  mutate(mean_alpha = for_beta[match(.$start, for_beta$start), &quot;mean_alpha&quot;])

cor.test(just_beta$beta, just_beta$mean_alpha)</code></pre>
<pre><code>
    Pearson&#39;s product-moment correlation

data:  just_beta$beta and just_beta$mean_alpha
t = 470.64, df = 32609, p-value &lt; 2.2e-16
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 0.9322275 0.9350134
sample estimates:
      cor 
0.9336346 </code></pre>
<pre class="r"><code>just_beta %&gt;%
  ggplot(aes(x = beta, y = mean_alpha - beta)) +
  geom_count(alpha = 0.3)</code></pre>
<p><img src="figure/correlationOfAlphas.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>just_beta %&gt;%
  ggplot(aes(x = beta, y = mean_alpha - beta, size = coverage)) +
  geom_point(alpha = 0.3)</code></pre>
<p><img src="figure/correlationOfAlphas.Rmd/unnamed-chunk-7-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>for_beta %&gt;% 
  distinct(read_id, meth_pattern, alpha) %&gt;%
  ggplot(aes(x = alpha)) +
  geom_density()</code></pre>
<p><img src="figure/correlationOfAlphas.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<ul>
<li>I don’t actually have the read positions I only have the cpg
positions on the read</li>
</ul>
<pre class="r"><code>for_beta %&gt;% 
  distinct(read_id, meth_pattern, alpha, total, read_start, read_end) %&gt;%
  filter(total &gt; 2) %&gt;%
  mutate(read_length = read_end - read_start + 1) %&gt;% 
  group_by(read_length) %&gt;%
  summarise(n=n()) %&gt;% .[order(.$n, decreasing = TRUE),] %&gt;% .[1:10,]</code></pre>
<pre><code># A tibble: 10 × 2
   read_length     n
         &lt;dbl&gt; &lt;int&gt;
 1          35 12613
 2          36 11713
 3          34 10370
 4          33 10162
 5          32  7796
 6          31  7268
 7          28  6778
 8          30  6135
 9          27  6005
10          29  5593</code></pre>
</div>
<div id="read-length-dist-last---first-cpg-alpha"
class="section level1">
<h1>read length (dist last - first cpg &amp; alpha)</h1>
<pre class="r"><code>for_beta %&gt;% 
  distinct(read_id, meth_pattern, alpha, total, read_start, read_end) %&gt;%
  filter(total &gt; 2) %&gt;%
  mutate(read_length = read_end - read_start + 1) %&gt;%
  ggplot(aes(x = as.factor(read_length), y = alpha)) +
  geom_boxplot()</code></pre>
<p><img src="figure/correlationOfAlphas.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>for_beta %&gt;% 
  distinct(read_id, meth_pattern, alpha, total, read_start, read_end) %&gt;%
  filter(total &gt; 2) %&gt;%
  mutate(read_length = read_end - read_start + 1) %&gt;%
  ggplot(aes(x = read_length, y = alpha)) +
  geom_point(alpha = 0.3)</code></pre>
<p><img src="figure/correlationOfAlphas.Rmd/unnamed-chunk-10-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="read-length-last---first-cpg-total-amount-of-cpgs"
class="section level1">
<h1>read length (last - first cpg &amp; total amount of cpgs)</h1>
<pre class="r"><code>for_beta %&gt;% 
  distinct(read_id, meth_pattern, alpha, total, read_start, read_end) %&gt;%
  filter(total &gt; 2) %&gt;%
  mutate(read_length = read_end - read_start + 1) %&gt;%
  group_by(read_length, total) %&gt;%
  summarise(n=n()) %&gt;% .[1:10,]</code></pre>
<pre><code>`summarise()` has grouped output by &#39;read_length&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code># A tibble: 10 × 3
# Groups:   read_length [6]
   read_length total     n
         &lt;dbl&gt; &lt;int&gt; &lt;int&gt;
 1           5     3     5
 2           6     3    89
 3           7     3    86
 4           7     4     1
 5           8     3   188
 6           8     4     3
 7           9     3   264
 8           9     4     5
 9          10     3   499
10          10     4    93</code></pre>
<pre class="r"><code>for_beta %&gt;% 
  distinct(read_id, meth_pattern, alpha, total, read_start, read_end) %&gt;%
  filter(total &gt; 2) %&gt;%
  mutate(read_length = read_end - read_start + 1) %&gt;%
  ggplot(aes(x = read_length, y = total)) +
  geom_point(alpha = 0.3)</code></pre>
<p><img src="figure/correlationOfAlphas.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>for_beta %&gt;% 
  distinct(read_id, meth_pattern, alpha, total, read_start, read_end) %&gt;%
  filter(total &gt; 2) %&gt;%
  mutate(read_length = read_end - read_start + 1) %&gt;%
  ggplot(aes(x = as.factor(read_length), y = total)) +
  geom_boxplot()</code></pre>
<p><img src="figure/correlationOfAlphas.Rmd/unnamed-chunk-12-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>for_beta %&gt;% 
  distinct(read_id, meth_pattern, alpha, total, read_start, read_end) %&gt;%
  filter(total &gt; 2) %&gt;%
  mutate(read_length = read_end - read_start + 1) %&gt;%
  ggplot(aes(x = as.factor(total), y = read_length)) +
  geom_boxplot()</code></pre>
<p><img src="figure/correlationOfAlphas.Rmd/unnamed-chunk-12-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="total-number-of-cpgs-and-alpha" class="section level1">
<h1>total number of cpgs and alpha</h1>
<pre class="r"><code>for_beta %&gt;% 
  distinct(read_id, meth_pattern, alpha, total, read_start, read_end) %&gt;%
  filter(total &gt; 2) %&gt;%
  mutate(read_length = read_end - read_start + 1) %&gt;%
  group_by(total, alpha) %&gt;%
  summarise(n=n()) %&gt;% .[1:10,]</code></pre>
<pre><code>`summarise()` has grouped output by &#39;total&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code># A tibble: 10 × 3
# Groups:   total [3]
   total alpha     n
   &lt;int&gt; &lt;dbl&gt; &lt;int&gt;
 1     3 0     15411
 2     3 0.333  3700
 3     3 0.667  4922
 4     3 1     11967
 5     4 0     15825
 6     4 0.25   2152
 7     4 0.5    1370
 8     4 0.75   1949
 9     4 1      4000
10     5 0     13863</code></pre>
<pre class="r"><code>for_beta %&gt;% 
  distinct(read_id, meth_pattern, alpha, total, read_start, read_end) %&gt;%
  filter(total &gt; 2) %&gt;%
  mutate(read_length = read_end - read_start + 1) %&gt;%
  ggplot(aes(x = as.factor(total), y = alpha)) +
  geom_boxplot()</code></pre>
<p><img src="figure/correlationOfAlphas.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>for_beta %&gt;% 
  distinct(read_id, meth_pattern, alpha, total, read_start, read_end) %&gt;%
  filter(total &gt; 2) %&gt;%
  mutate(read_length = read_end - read_start + 1) %&gt;%
  ggplot(aes(x = total, y = alpha)) +
  geom_point(alpha = 0.3)</code></pre>
<p><img src="figure/correlationOfAlphas.Rmd/unnamed-chunk-14-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>for_beta %&gt;% 
  distinct(read_id, meth_pattern, alpha, total, read_start, read_end) %&gt;%
  filter(total &gt; 2) %&gt;%
  mutate(read_length = read_end - read_start + 1) %&gt;%
  ggplot(aes(x = alpha, colour = as.factor(total))) +
  geom_density()</code></pre>
<p><img src="figure/correlationOfAlphas.Rmd/unnamed-chunk-14-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>for_beta %&gt;% 
  distinct(read_id, meth_pattern, alpha, total, read_start, read_end) %&gt;%
  filter(total &gt; 2) %&gt;%
  mutate(read_length = read_end - read_start + 1) %&gt;%
  group_by(meth_pattern, total, alpha) %&gt;%
  summarise(n=n()) %&gt;% .[order(.$total),] %&gt;%
  ungroup() %&gt;% group_by(total) %&gt;% 
  mutate(n_pat = n(), sum_pat = sum(n), prop = n/sum_pat) %&gt;%
  ggplot(aes(x = alpha, y = prop, colour = meth_pattern)) +
  geom_point() +
  facet_wrap(~as.factor(total)) +
  theme(legend.position = &quot;none&quot;)</code></pre>
<pre><code>`summarise()` has grouped output by &#39;meth_pattern&#39;, &#39;total&#39;. You can override
using the `.groups` argument.</code></pre>
<p><img src="figure/correlationOfAlphas.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" />
* ok actually this is semi something (possibly better than below plots)
* each colour is a different meth pattern * and prop is how often it
shows up * it says that bulk of data is alpha = 0 (all same) * this
should probably be bar charts but I can’t be bothered</p>
<pre class="r"><code>for_beta %&gt;% 
  distinct(read_id, meth_pattern, alpha, total, read_start, read_end) %&gt;%
  filter(total &gt; 2) %&gt;%
  mutate(read_length = read_end - read_start + 1) %&gt;%
  group_by(meth_pattern, total, alpha) %&gt;%
  summarise(n=n()) %&gt;% .[order(.$total),] %&gt;%
  ungroup() %&gt;% group_by(total) %&gt;% 
  mutate(n_pat = n(), sum_pat = sum(n), prop = n/sum_pat) %&gt;%
  filter(alpha != 0, alpha != 1) %&gt;%
  ggplot(aes(x = alpha, y = prop, colour = meth_pattern)) +
  geom_point() +
  facet_wrap(~as.factor(total)) +
  theme(legend.position = &quot;none&quot;)</code></pre>
<pre><code>`summarise()` has grouped output by &#39;meth_pattern&#39;, &#39;total&#39;. You can override
using the `.groups` argument.</code></pre>
<p><img src="figure/correlationOfAlphas.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>for_beta %&gt;% 
  distinct(read_id, meth_pattern, alpha, total, read_start, read_end) %&gt;%
  filter(total &gt; 2) %&gt;%
  mutate(read_length = read_end - read_start + 1) %&gt;%
  group_by(meth_pattern, total, alpha) %&gt;%
  summarise(n=n()) %&gt;% .[order(.$total),] %&gt;%
  ungroup() %&gt;% group_by(total) %&gt;% 
  mutate(n_pat = n(), sum_pat = sum(n), prop = n/sum_pat) %&gt;%
  filter(alpha != 0, alpha != 1, total == 3) %&gt;%
  ggplot(aes(x = alpha, y = prop, colour = meth_pattern)) +
  geom_point() </code></pre>
<pre><code>`summarise()` has grouped output by &#39;meth_pattern&#39;, &#39;total&#39;. You can override
using the `.groups` argument.</code></pre>
<p><img src="figure/correlationOfAlphas.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" />
* the lowest prop here is the umu and mum * that’s actually nice to see
- following logic of cpgs sharing meth states</p>
<div id="section" class="section level11">
<p class="heading"></p>
<ul>
<li><p>is any of the above interesting? I don’t know</p></li>
<li><p>I want to know if there is something about the distance between
cpgs on a read that impacts their relationship (same status)</p></li>
<li><p>because we know cpgs are spatially correlated</p></li>
<li><p>would this be useful to know?</p></li>
<li><p>maybe in context of identifying a window size to look
along</p></li>
<li><p>can the relationship between cpg distance and methylation state
be modelled??</p></li>
<li><p>like if after a certain distance - cpg was way less likely to
have same methylation state - that’s interesting</p></li>
<li><p>first step - distance to previous cpg on read, and is the meth
state shared</p></li>
<li><p>work with only reads min 3 cpgs</p></li>
</ul>
<pre class="r"><code>for_beta %&gt;% filter(total &gt; 2) %&gt;%
  group_by(read_id) %&gt;%
  mutate(gap_to_previous = start - lag(start),
         meth_state_same_previous = ifelse(meth_status == lag(meth_status), TRUE, FALSE)) %&gt;%
  ungroup() %&gt;%
  filter(!is.na(gap_to_previous)) %&gt;%
  ggplot(aes(x = meth_state_same_previous, y = gap_to_previous)) +
  geom_boxplot() </code></pre>
<p><img src="figure/correlationOfAlphas.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>for_beta %&gt;% filter(total &gt; 2) %&gt;%
  group_by(read_id) %&gt;%
  mutate(gap_to_previous = start - lag(start),
         meth_state_same_previous = ifelse(meth_status == lag(meth_status), TRUE, FALSE)) %&gt;%
  ungroup() %&gt;%
  filter(!is.na(gap_to_previous)) %&gt;%
  ggplot(aes(x = meth_state_same_previous, y = gap_to_previous)) +
  geom_violin()</code></pre>
<p><img src="figure/correlationOfAlphas.Rmd/unnamed-chunk-18-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>for_beta %&gt;% filter(total &gt; 2) %&gt;%
  group_by(read_id) %&gt;%
  mutate(gap_to_previous = start - lag(start),
         meth_state_same_previous = ifelse(meth_status == lag(meth_status), TRUE, FALSE)) %&gt;%
  ungroup() %&gt;%
  filter(!is.na(gap_to_previous)) %&gt;%
  ggplot(aes(x = gap_to_previous, colour = meth_state_same_previous)) +
  geom_density()</code></pre>
<p><img src="figure/correlationOfAlphas.Rmd/unnamed-chunk-18-3.png" width="672" style="display: block; margin: auto;" />
* so it is lower in true which is expected * but the shape of the
distributions are similar * also the TRUE is a weird looking
distribution</p>
<pre class="r"><code>for_beta %&gt;% distinct(read_id, read_start, read_end) %&gt;% mutate(read_length = read_end - read_start + 1) %&gt;% .[order(.$read_length, decreasing = TRUE),] %&gt;% .[1:10,]</code></pre>
<pre><code>                               read_id read_start read_end read_length
8    HWI-EAS149_7:1:100:1015:197:0:1:1   36506967 36507002          36
13  HWI-EAS149_7:1:100:1021:1167:0:1:1   18111247 18111282          36
57    HWI-EAS149_7:1:100:111:147:0:1:1   31084850 31084885          36
76   HWI-EAS149_7:1:100:1156:103:0:1:1   42719993 42720028          36
81   HWI-EAS149_7:1:100:1163:666:0:1:1   36387370 36387405          36
104 HWI-EAS149_7:1:100:1194:1114:0:1:1   40396188 40396223          36
122  HWI-EAS149_7:1:100:1227:850:0:1:1   19447913 19447948          36
152 HWI-EAS149_7:1:100:1273:1547:0:1:1   45284795 45284830          36
154 HWI-EAS149_7:1:100:1277:1013:0:1:1   42331210 42331245          36
165  HWI-EAS149_7:1:100:1290:462:0:1:0   20016365 20016400          36</code></pre>
<ul>
<li><p>why are the reads 36 bp????</p></li>
<li><p>or is it a cpg thing and they don’t actually start and end on cpg
sites???</p></li>
<li><p>ooh could I plot like I did the tp in gst? - if same line is
same, if changes it goes up a step?</p></li>
<li><p>just as a way to visualise the reads</p></li>
</ul>
<pre class="r"><code>step_plot_alpha &lt;- for_beta %&gt;% filter(total &gt; 2) %&gt;%
  group_by(read_id) %&gt;%
  mutate(gap_to_previous = start - lag(start),
         meth_state_same_previous = ifelse(meth_status == lag(meth_status), TRUE, FALSE)) %&gt;%
  ungroup()</code></pre>
<pre class="r"><code>step_plot_alpha %&gt;%
  group_by(read_id) %&gt;%
  mutate(gap_to_previous = ifelse(is.na(gap_to_previous), 0, gap_to_previous), 
         new_length = cumsum(gap_to_previous),
         step_level = consecutive_id(meth_state_same_previous)) %&gt;%
  ungroup() %&gt;%
  ggplot(aes(x = new_length, y = step_level)) +
  geom_line()</code></pre>
<p><img src="figure/correlationOfAlphas.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>step_plot_alpha[1:500,] %&gt;%
  group_by(read_id) %&gt;%
  mutate(gap_to_previous = ifelse(is.na(gap_to_previous), 0, gap_to_previous), 
         new_length = cumsum(gap_to_previous),
         step_level = consecutive_id(meth_state_same_previous)) %&gt;%
  ggplot(aes(x = new_length, y = step_level, colour = read_id)) +
  geom_line() +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/correlationOfAlphas.Rmd/unnamed-chunk-22-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>what if similar but just with m and u at 1 and 0??</p>
<pre class="r"><code>step_plot_alpha[1:50,] %&gt;%
  group_by(read_id) %&gt;%
  mutate(gap_to_previous = ifelse(is.na(gap_to_previous), 0, gap_to_previous), 
         new_length = cumsum(gap_to_previous),
         step_meth = ifelse(meth_status == &quot;M&quot;, 1, 0)) %&gt;%
  ggplot(aes(x = new_length, y = step_meth, colour = read_id)) +
  geom_line() +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/correlationOfAlphas.Rmd/unnamed-chunk-23-1.png" width="672" style="display: block; margin: auto;" /></p>
<ul>
<li>yeah this is impossible to interpret</li>
<li>also I feel like this is a plot I made agesss ago</li>
</ul>
<pre class="r"><code>step_plot_alpha %&gt;% distinct(read_id) %&gt;% nrow()</code></pre>
<pre><code>[1] 115403</code></pre>
</div>
</div>
<div id="maybe-easier-as-beta" class="section level1">
<h1>maybe easier as beta?</h1>
<pre class="r"><code>just_beta %&gt;%
  mutate(gap_previous_cpg = start - lag(start),
         diff_previous_beta = beta - lag(beta)) %&gt;%
  filter(gap_previous_cpg &lt; 10000) %&gt;%
  ggplot(aes(x = gap_previous_cpg, y = diff_previous_beta)) +
  geom_point(alpha = 0.3)</code></pre>
<p><img src="figure/correlationOfAlphas.Rmd/unnamed-chunk-25-1.png" width="672" style="display: block; margin: auto;" />
* well this doesn’t look like anything</p>
<pre class="r"><code>just_beta %&gt;%
  mutate(beta_range = case_when(beta &lt; 0.3 ~ &quot;low&quot;,
                                beta &gt; 0.7 ~ &quot;high&quot;,
                                TRUE ~ &quot;medium&quot;),
          gap_previous_cpg = start - lag(start),
         diff_previous_beta = ifelse(beta_range == lag(beta_range), TRUE, FALSE)) %&gt;%
  filter(gap_previous_cpg &lt; 5000) %&gt;%
  ggplot(aes(x = diff_previous_beta, y = log2(gap_previous_cpg))) +
  geom_boxplot()</code></pre>
<p><img src="figure/correlationOfAlphas.Rmd/unnamed-chunk-26-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>just_beta %&gt;%
  mutate(beta_range = case_when(beta &lt; 0.3 ~ &quot;low&quot;,
                                beta &gt; 0.7 ~ &quot;high&quot;,
                                TRUE ~ &quot;medium&quot;),
          gap_previous_cpg = start - lag(start),
         diff_previous_beta = ifelse(beta_range == lag(beta_range), TRUE, FALSE)) %&gt;%
  filter(gap_previous_cpg &lt; 5000) %&gt;%
  ggplot(aes(x = log2(gap_previous_cpg), colour = diff_previous_beta)) +
  geom_density()</code></pre>
<p><img src="figure/correlationOfAlphas.Rmd/unnamed-chunk-26-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># just gap &lt; 500
just_beta %&gt;%
  mutate(beta_range = case_when(beta &lt; 0.3 ~ &quot;low&quot;,
                                beta &gt; 0.7 ~ &quot;high&quot;,
                                TRUE ~ &quot;medium&quot;),
          gap_previous_cpg = start - lag(start),
         diff_previous_beta = ifelse(beta_range == lag(beta_range), TRUE, FALSE)) %&gt;%
  filter(gap_previous_cpg &lt; 500) %&gt;%
  ggplot(aes(x = diff_previous_beta, y = log2(gap_previous_cpg))) +
  geom_boxplot()</code></pre>
<p><img src="figure/correlationOfAlphas.Rmd/unnamed-chunk-27-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>just_beta %&gt;%
  mutate(beta_range = case_when(beta &lt; 0.3 ~ &quot;low&quot;,
                                beta &gt; 0.7 ~ &quot;high&quot;,
                                TRUE ~ &quot;medium&quot;),
          gap_previous_cpg = start - lag(start),
         diff_previous_beta = ifelse(beta_range == lag(beta_range), TRUE, FALSE)) %&gt;%
  filter(gap_previous_cpg &lt; 500) %&gt;%
  ggplot(aes(x = log2(gap_previous_cpg), colour = diff_previous_beta)) +
  geom_density()</code></pre>
<p><img src="figure/correlationOfAlphas.Rmd/unnamed-chunk-27-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code> just_beta %&gt;%
  mutate(beta_range = case_when(beta &lt; 0.3 ~ &quot;low&quot;,
                                beta &gt; 0.7 ~ &quot;high&quot;,
                                TRUE ~ &quot;medium&quot;),
          gap_previous_cpg = start - lag(start),
         diff_previous_beta = ifelse(beta_range == lag(beta_range), TRUE, FALSE)) %&gt;%
#  filter(gap_previous_cpg &lt; 500) %&gt;%
  mutate(gap_previous_cpg = ifelse(is.na(gap_previous_cpg), 0, gap_previous_cpg), 
         new_length = cumsum(gap_previous_cpg),
         step_level = consecutive_id(diff_previous_beta)) %&gt;%
  ggplot(aes(x = new_length, y = step_level)) +
  geom_line()</code></pre>
<p><img src="figure/correlationOfAlphas.Rmd/unnamed-chunk-28-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.2 (2024-10-31)
Platform: aarch64-apple-darwin20
Running under: macOS Sequoia 15.2

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Australia/Melbourne
tzcode source: internal

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] ggplot2_3.5.2                     dplyr_1.1.4                      
 [3] stringr_1.5.1                     tidyr_1.3.1                      
 [5] plyranges_1.26.0                  data.table_1.17.4                
 [7] BSgenome.Hsapiens.UCSC.hg38_1.4.5 BSgenome_1.74.0                  
 [9] rtracklayer_1.66.0                BiocIO_1.16.0                    
[11] Biostrings_2.74.1                 XVector_0.46.0                   
[13] GenomicRanges_1.58.0              GenomeInfoDb_1.42.3              
[15] IRanges_2.40.1                    S4Vectors_0.44.0                 
[17] BiocGenerics_0.52.0              

loaded via a namespace (and not attached):
 [1] SummarizedExperiment_1.36.0 gtable_0.3.6               
 [3] rjson_0.2.23                xfun_0.52                  
 [5] bslib_0.9.0                 Biobase_2.66.0             
 [7] lattice_0.22-6              generics_0.1.4             
 [9] vctrs_0.6.5                 tools_4.4.2                
[11] bitops_1.0-9                curl_6.2.3                 
[13] parallel_4.4.2              tibble_3.2.1               
[15] pkgconfig_2.0.3             Matrix_1.7-1               
[17] RColorBrewer_1.1-3          lifecycle_1.0.4            
[19] GenomeInfoDbData_1.2.13     farver_2.1.2               
[21] compiler_4.4.2              git2r_0.36.2               
[23] Rsamtools_2.22.0            codetools_0.2-20           
[25] httpuv_1.6.16               htmltools_0.5.8.1          
[27] sass_0.4.10                 RCurl_1.98-1.17            
[29] yaml_2.3.10                 later_1.4.2                
[31] pillar_1.10.2               crayon_1.5.3               
[33] jquerylib_0.1.4             whisker_0.4.1              
[35] BiocParallel_1.40.2         DelayedArray_0.32.0        
[37] cachem_1.1.0                abind_1.4-8                
[39] tidyselect_1.2.1            digest_0.6.37              
[41] stringi_1.8.7               purrr_1.0.4                
[43] restfulr_0.0.15             labeling_0.4.3             
[45] rprojroot_2.0.4             fastmap_1.2.0              
[47] grid_4.4.2                  SparseArray_1.6.2          
[49] cli_3.6.5                   magrittr_2.0.3             
[51] S4Arrays_1.6.0              XML_3.99-0.18              
[53] withr_3.0.2                 scales_1.4.0               
[55] UCSC.utils_1.2.0            promises_1.3.3             
[57] rmarkdown_2.29              httr_1.4.7                 
[59] matrixStats_1.5.0           workflowr_1.7.1            
[61] evaluate_1.0.3              knitr_1.50                 
[63] rlang_1.1.6                 Rcpp_1.0.14                
[65] glue_1.8.0                  rstudioapi_0.17.1          
[67] jsonlite_2.0.0              R6_2.6.1                   
[69] MatrixGenerics_1.18.1       GenomicAlignments_1.42.0   
[71] fs_1.6.6                    zlibbioc_1.52.0            </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
