<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Caitlin Page" />

<meta name="date" content="2025-12-11" />

<title>distNextCpG</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">3_cf_nano</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/caitlinpage/methyl_nano_cf">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">distNextCpG</h1>
<h4 class="author">Caitlin Page</h4>
<h4 class="date">2025-12-11</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-12-17
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>3_cf_nano/analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250606code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250606)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250606code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250606)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomcaitlinpage3cfnanotree38f6e2208cc6d3e83635e5e760a6505499f8fac0targetblank38f6e22a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/caitlinpage/3_cf_nano/tree/38f6e2208cc6d3e83635e5e760a6505499f8fac0" target="_blank">38f6e22</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomcaitlinpage3cfnanotree38f6e2208cc6d3e83635e5e760a6505499f8fac0targetblank38f6e22a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/caitlinpage/3_cf_nano/tree/38f6e2208cc6d3e83635e5e760a6505499f8fac0" target="_blank">38f6e22</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    code/Fragmentomics_GenomBiol-main/
    Ignored:    code/dist_modkit_v0.5.0_5120ef7_candle/
    Ignored:    code/dist_modkit_v0.5.0_5120ef7_tch/
    Ignored:    code/modkit-0.5.0/
    Ignored:    code/modkit_v0.5.0_5120ef7/
    Ignored:    code/uses_biscuit/
    Ignored:    data/cfnano_paper/
    Ignored:    data/ncbi_dataset/
    Ignored:    data/rrbs/
    Ignored:    output/bedtools/
    Ignored:    output/biscuit/
    Ignored:    output/cluster_files/
    Ignored:    output/html docs - versions/
    Ignored:    output/modkit/
    Ignored:    output/nanoplot/
    Ignored:    output/research_sprint/
    Ignored:    output/samtools/
    Ignored:    output/schematics/
    Ignored:    renv.lock
    Ignored:    renv/

Unstaged changes:
    Modified:   .Rprofile
    Modified:   .gitignore
    Modified:   3_cf_nano.Rproj
    Modified:   _workflowr.yml
    Modified:   analysis/01a_bedFileCorrelation.Rmd
    Modified:   analysis/cpgFeatures.Rmd
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/distNextCpG.Rmd</code>) and HTML
(<code>docs/distNextCpG.html</code>) files. If you’ve configured a
remote Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/caitlinpage/3_cf_nano/blob/38f6e2208cc6d3e83635e5e760a6505499f8fac0/analysis/distNextCpG.Rmd" target="_blank">38f6e22</a>
</td>
<td>
Caitlin Page
</td>
<td>
2025-12-17
</td>
<td>
wflow_publish("distNextCpg.Rmd")
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/caitlinpage/3_cf_nano/blob/cfd31c10ab437ba9089ac4a075029fba9513358b/analysis/distNextCpG.Rmd" target="_blank">cfd31c1</a>
</td>
<td>
Caitlin Page
</td>
<td>
2025-12-17
</td>
<td>
plots look weird and i dont know how i feel about them
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/caitlinpage/3_cf_nano/blob/6d57fa61d6a203abaef19f4fda46c4831d7971fe/analysis/distNextCpG.Rmd" target="_blank">6d57fa6</a>
</td>
<td>
Caitlin Page
</td>
<td>
2025-12-17
</td>
<td>
tidied fn and removed working out code
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/caitlinpage/3_cf_nano/blob/7cd4287b0821d750def44bde18665cf943cebfa2/analysis/distNextCpG.Rmd" target="_blank">7cd4287</a>
</td>
<td>
Caitlin Page
</td>
<td>
2025-12-17
</td>
<td>
function and figured out how to deal with missing patterns and inf
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/caitlinpage/3_cf_nano/blob/4f9981d2ab7fa46ae6ac3c70d020dbcbe38fd723/analysis/distNextCpG.Rmd" target="_blank">4f9981d</a>
</td>
<td>
Caitlin Page
</td>
<td>
2025-12-17
</td>
<td>
fixed the p-val and tidied up
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/caitlinpage/3_cf_nano/blob/1c552bbc880eb2a144974239ffecc0c7b0da6434/analysis/distNextCpG.Rmd" target="_blank">1c552bb</a>
</td>
<td>
Caitlin Page
</td>
<td>
2025-12-17
</td>
<td>
go from captured image to df
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/caitlinpage/3_cf_nano/blob/9135adb8393018a95a263962aa01a39760607a3f/analysis/distNextCpG.Rmd" target="_blank">9135adb</a>
</td>
<td>
Caitlin Page
</td>
<td>
2025-12-11
</td>
<td>
init commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="packages" class="section level2">
<h2>Packages</h2>
<pre class="r"><code>library(annotatr)
library(data.table)
library(plyranges)</code></pre>
<pre><code>Loading required package: BiocGenerics</code></pre>
<pre><code>
Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    IQR, mad, sd, var, xtabs</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    anyDuplicated, aperm, append, as.data.frame, basename, cbind,
    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,
    get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,
    match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,
    Position, rank, rbind, Reduce, rownames, sapply, saveRDS, setdiff,
    table, tapply, union, unique, unsplit, which.max, which.min</code></pre>
<pre><code>Loading required package: IRanges</code></pre>
<pre><code>Loading required package: S4Vectors</code></pre>
<pre><code>Loading required package: stats4</code></pre>
<pre><code>
Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:data.table&#39;:

    first, second</code></pre>
<pre><code>The following object is masked from &#39;package:utils&#39;:

    findMatches</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    expand.grid, I, unname</code></pre>
<pre><code>
Attaching package: &#39;IRanges&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:data.table&#39;:

    shift</code></pre>
<pre><code>Loading required package: GenomicRanges</code></pre>
<pre><code>Loading required package: GenomeInfoDb</code></pre>
<pre><code>
Attaching package: &#39;plyranges&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:IRanges&#39;:

    slice</code></pre>
<pre><code>The following object is masked from &#39;package:data.table&#39;:

    between</code></pre>
<pre><code>The following object is masked from &#39;package:stats&#39;:

    filter</code></pre>
<pre class="r"><code>library(tidyr)</code></pre>
<pre><code>
Attaching package: &#39;tidyr&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:S4Vectors&#39;:

    expand</code></pre>
<pre class="r"><code>library(stringr)
library(dplyr)</code></pre>
<pre><code>
Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:plyranges&#39;:

    between, n, n_distinct</code></pre>
<pre><code>The following objects are masked from &#39;package:GenomicRanges&#39;:

    intersect, setdiff, union</code></pre>
<pre><code>The following object is masked from &#39;package:GenomeInfoDb&#39;:

    intersect</code></pre>
<pre><code>The following objects are masked from &#39;package:IRanges&#39;:

    collapse, desc, intersect, setdiff, slice, union</code></pre>
<pre><code>The following objects are masked from &#39;package:S4Vectors&#39;:

    first, intersect, rename, setdiff, setequal, union</code></pre>
<pre><code>The following objects are masked from &#39;package:BiocGenerics&#39;:

    combine, intersect, setdiff, union</code></pre>
<pre><code>The following objects are masked from &#39;package:data.table&#39;:

    between, first, last</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    filter, lag</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    intersect, setdiff, setequal, union</code></pre>
<pre class="r"><code>library(ggplot2)</code></pre>
</div>
<div id="data-objects" class="section level2">
<h2>Data Objects</h2>
<pre class="r"><code>cg_sites &lt;- cbind(data.frame(Biostrings::matchPattern(&quot;CG&quot;, BSgenome.Hsapiens.UCSC.hg38::BSgenome.Hsapiens.UCSC.hg38[[&quot;chr22&quot;]])), seqnames = &quot;chr22&quot;) %&gt;% 
  mutate(pos = paste0(seqnames, &quot;-&quot;, start)) %&gt;%
  relocate(pos, seqnames)</code></pre>
<pre><code>Warning in .local(x, row.names, optional, ...): &#39;optional&#39; argument was ignored</code></pre>
<pre class="r"><code>cg_sites$index &lt;- 1:nrow(cg_sites)</code></pre>
<pre class="r"><code>source(&quot;../code/processData.R&quot;)

for_beta &lt;- readRDS(&quot;../output/cluster_files/rrbs_alpha_beta.rds&quot;)
for_beta &lt;- for_beta %&gt;% filter(seqnames == &quot;chr22&quot;)
rrbs_for_beta_pat &lt;- pat_for_wgbs(for_beta)</code></pre>
<pre><code>`summarise()` has grouped output by &#39;seqnames&#39;, &#39;start_pos&#39;, &#39;meth_pattern&#39;,
&#39;alpha&#39;, &#39;end_pos&#39;. You can override using the `.groups` argument.</code></pre>
<pre class="r"><code>for_beta &lt;- for_beta %&gt;% group_by(start) %&gt;% mutate(coverage = n(), mean_alpha = mean(alpha)) %&gt;% ungroup() %&gt;% data.frame()
just_beta &lt;- for_beta %&gt;% distinct(start, beta, coverage, mean_alpha) %&gt;% .[order(.$start),]</code></pre>
<pre class="r"><code>for_beta %&gt;%
  group_by(read_id) %&gt;%
  mutate(dist_to_next = lead(start) - start) %&gt;%
  ungroup() %&gt;%
  ggplot(aes(x = dist_to_next)) +
  geom_density()</code></pre>
<pre><code>Warning: Removed 204769 rows containing non-finite outside the scale range
(`stat_density()`).</code></pre>
<p><img src="figure/distNextCpG.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="pete-hickey-methylationtuples" class="section level1">
<h1>Pete Hickey: MethylationTuples</h1>
<ul>
<li><p>I couldn’t download the package</p></li>
<li><p>but the math is Mantel-Haenszel odds ratio estimator</p></li>
<li><p>in a 2 by 2 chi square table for neighbouring cpgs</p></li>
<li><p>and distance between cpgs is also involved</p></li>
<li><p>2 x 2 x K table for each IPD (distance between 2 cpgs)</p></li>
<li><p>I want table like</p></li>
</ul>
<p>observ (X, Y)</p>
<pre><code>  Y
1   2.  total</code></pre>
<p>1 n11 n12 X 2 n21 n22 total and 1 and 2 are U and M</p>
<ul>
<li>so it’s a separate table for each distance</li>
<li>maybe should try it separately for different total number of cpgs on
read</li>
<li>try for 2 cpgs on read</li>
</ul>
<pre class="r"><code>for_beta %&gt;%
  filter(total == 2) %&gt;%
  group_by(read_id) %&gt;%
  mutate(dist_to_next = lead(start) - start) %&gt;%
  ungroup() %&gt;%
  fill(dist_to_next, .direction = &quot;down&quot;) %&gt;%
  distinct(read_id, meth_pattern, dist_to_next) %&gt;%
  group_by(dist_to_next, meth_pattern) %&gt;%
  summarise(count = n()) %&gt;% .[1:5,]</code></pre>
<pre><code>`summarise()` has grouped output by &#39;dist_to_next&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code># A tibble: 5 × 3
# Groups:   dist_to_next [2]
  dist_to_next meth_pattern count
         &lt;int&gt; &lt;chr&gt;        &lt;int&gt;
1            2 MM              61
2            2 MU               7
3            2 UM              11
4            2 UU              27
5            3 MM             270</code></pre>
<pre class="r"><code>tab &lt;- for_beta %&gt;%
  filter(total == 2) %&gt;%
  group_by(read_id) %&gt;%
  mutate(dist_to_next = lead(start) - start) %&gt;%
  ungroup() %&gt;%
  fill(dist_to_next, .direction = &quot;down&quot;) %&gt;%
  distinct(read_id, meth_pattern, dist_to_next) %&gt;%
  group_by(dist_to_next, meth_pattern) %&gt;%
  summarise(count = n()) %&gt;%
  filter(dist_to_next == 2) %&gt;% 
  ungroup() %&gt;%
  mutate(row_factor = substr(meth_pattern, 1, 1),
         col_factor = substr(meth_pattern, 2, 2))</code></pre>
<pre><code>`summarise()` has grouped output by &#39;dist_to_next&#39;. You can override using the
`.groups` argument.</code></pre>
<pre class="r"><code>tab</code></pre>
<pre><code># A tibble: 4 × 5
  dist_to_next meth_pattern count row_factor col_factor
         &lt;int&gt; &lt;chr&gt;        &lt;int&gt; &lt;chr&gt;      &lt;chr&gt;     
1            2 MM              61 M          M         
2            2 MU               7 M          U         
3            2 UM              11 U          M         
4            2 UU              27 U          U         </code></pre>
<pre class="r"><code>with(tab, tapply(count, list(row_factor, col_factor), sum))</code></pre>
<pre><code>   M  U
M 61  7
U 11 27</code></pre>
<pre class="r"><code>chisq_table &lt;- for_beta %&gt;%
  filter(total == 2) %&gt;%
  group_by(read_id) %&gt;%
  mutate(dist_to_next = lead(start) - start) %&gt;%
  ungroup() %&gt;%
  fill(dist_to_next, .direction = &quot;down&quot;) %&gt;%
  distinct(read_id, meth_pattern, dist_to_next) %&gt;%
  group_by(dist_to_next, meth_pattern) %&gt;%
  summarise(count = n()) %&gt;%
  ungroup() %&gt;%
  mutate(cpg_1 = substr(meth_pattern, 1, 1),
         cpg_2 = substr(meth_pattern, 2, 2)) %&gt;%
  group_by(dist_to_next) %&gt;%
  reframe(tab = list(xtabs(count ~ cpg_1 + cpg_2, data = cur_data())))</code></pre>
<pre><code>`summarise()` has grouped output by &#39;dist_to_next&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code>Warning: There was 1 warning in `reframe()`.
ℹ In argument: `tab = list(xtabs(count ~ cpg_1 + cpg_2, data = cur_data()))`.
ℹ In group 1: `dist_to_next = 2`.
Caused by warning:
! `cur_data()` was deprecated in dplyr 1.1.0.
ℹ Please use `pick()` instead.</code></pre>
<pre class="r"><code>chisq_table &lt;- chisq_table$tab 
names(chisq_table) &lt;- for_beta %&gt;%
  filter(total == 2) %&gt;%
  group_by(read_id) %&gt;%
  mutate(dist_to_next = lead(start) - start) %&gt;%
  ungroup() %&gt;%
  fill(dist_to_next, .direction = &quot;down&quot;) %&gt;%
  distinct(read_id, meth_pattern, dist_to_next) %&gt;%
  group_by(dist_to_next, meth_pattern) %&gt;%
  summarise(count = n()) %&gt;%
  mutate(dist_to_next = paste0(&quot;dist_next_&quot;, dist_to_next)) %&gt;%
  .$dist_to_next %&gt;% unique()</code></pre>
<pre><code>`summarise()` has grouped output by &#39;dist_to_next&#39;. You can override using the
`.groups` argument.</code></pre>
<pre class="r"><code>chisq_table[1:2]</code></pre>
<pre><code>$dist_next_2
     cpg_2
cpg_1  M  U
    M 61  7
    U 11 27

$dist_next_3
     cpg_2
cpg_1   M   U
    M 270 100
    U  30 519</code></pre>
<ul>
<li>do list and then convert to array I think</li>
</ul>
<pre class="r"><code>chisq_table &lt;- simplify2array(chisq_table)</code></pre>
<pre class="r"><code>mh_test &lt;- mantelhaen.test(chisq_table)
mh_test</code></pre>
<pre><code>
    Mantel-Haenszel chi-squared test with continuity correction

data:  chisq_table
Mantel-Haenszel X-squared = 13898, df = 1, p-value &lt; 2.2e-16
alternative hypothesis: true common odds ratio is not equal to 1
95 percent confidence interval:
 12.39991 13.61250
sample estimates:
common odds ratio 
         12.99206 </code></pre>
<pre class="r"><code>mh_test$estimate</code></pre>
<pre><code>common odds ratio 
         12.99206 </code></pre>
<ul>
<li>this version of the test I only get the summarised odds ratio</li>
<li>need another package to get individual if I want to plot like
Petes</li>
</ul>
<p><a
href="https://rdrr.io/cran/epiDisplay/man/mhor.html">epiDisplay</a></p>
<pre class="r"><code>epiDisplay::mhor(mhtable = chisq_table)</code></pre>
<pre><code>
Stratified analysis by  Var3 
                      OR lower lim. upper lim.   P value
Var3 dist_next_2   20.49       6.76      71.07  1.92e-10
Var3 dist_next_3   46.46      29.77      74.30 2.07e-109
Var3 dist_next_4    7.93       6.14      10.28  2.02e-67
Var3 dist_next_5   42.32      30.30      59.74 4.33e-166
Var3 dist_next_6   32.81      23.66      46.02 1.78e-143
Var3 dist_next_7   34.35      27.04      44.00 2.20e-273
Var3 dist_next_8   17.44      13.28      23.05 2.30e-123
Var3 dist_next_9   24.83      17.94      34.73 1.52e-114
Var3 dist_next_10  36.09      27.83      46.98 8.26e-242
Var3 dist_next_11  30.44      21.44      43.72 3.08e-118
Var3 dist_next_12  16.03      11.98      21.60 6.99e-103
Var3 dist_next_13  21.37      15.77      29.22 3.16e-122
Var3 dist_next_14  20.11      13.75      29.88  5.56e-75
Var3 dist_next_15  14.01      10.80      18.30 2.08e-114
Var3 dist_next_16   9.01       7.18      11.34  7.72e-99
Var3 dist_next_17  14.70      10.82      20.20  6.52e-93
Var3 dist_next_18  11.47       8.41      15.78  1.31e-67
Var3 dist_next_19  12.13       8.82      16.87  4.86e-72
Var3 dist_next_20  16.44      11.87      23.01  1.39e-84
Var3 dist_next_21  14.30      11.05      18.60 6.46e-116
Var3 dist_next_22  11.48       9.17      14.45 5.78e-125
Var3 dist_next_23  17.11      13.20      22.31 6.29e-133
Var3 dist_next_24  10.86       8.37      14.17  3.91e-89
Var3 dist_next_25  10.10       7.83      13.10  1.05e-86
Var3 dist_next_26   6.58       5.21       8.33  8.67e-64
Var3 dist_next_27  13.97      10.29      19.11  5.45e-83
Var3 dist_next_28  14.26       9.71      21.28  2.31e-56
Var3 dist_next_29   9.39       7.01      12.65  7.67e-62
Var3 dist_next_30   6.33       4.43       9.12  7.05e-28
Var3 dist_next_31   3.73       2.90       4.82  1.62e-26
Var3 dist_next_32   8.40       6.81      10.38 1.72e-104
Var3 dist_next_33  10.26       7.83      13.51  1.41e-77
Var3 dist_next_34   6.85       5.37       8.77  5.54e-63
Var3 dist_next_35   6.62       5.11       8.60  4.04e-54
M-H combined       12.99      12.40      13.61  0.00e+00

M-H Chi2(1) = 13900.44 , P value = 0 
Homogeneity test, chi-squared 33 d.f. = 578.11 , P value = 0 </code></pre>
<p><img src="figure/distNextCpG.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>try &lt;- capture.output(epiDisplay::mhor(mhtable = chisq_table, graph = FALSE))

try &lt;- try %&gt;% data.frame() %&gt;% .[4:39,] %&gt;% data.frame()
colnames(try) &lt;- &quot;all&quot;


# sep the words and the numbers
try &lt;- try %&gt;% separate_wider_delim(cols = all, delim = &quot; &quot;, names = c(&quot;type&quot;, &quot;dist&quot;, &quot;numbers&quot;), too_few = &quot;align_start&quot;, too_many = &quot;merge&quot;)

# sep the numbers
numeric_lines &lt;- grep(&quot;-?\\d+\\.\\d{2,}.*-?\\d+\\.\\d{2,}.*-?\\d+\\.\\d{2,}&quot;, 
                      try, value = TRUE)
# to maintain the p-val
pattern &lt;- &quot;-?\\d+\\.\\d{2,}|[-+]?\\d+\\.?\\d*(?:[eE][-+]?\\d+)&quot;

vals &lt;- regmatches(
  numeric_lines,
  gregexpr(pattern, numeric_lines))

m &lt;- do.call(rbind, vals)
m &lt;- as.vector(m)
names(m) &lt;- rep(c(&quot;OR&quot;, &quot;lower_lim&quot;, &quot;upper_lim&quot;, &quot;p_value&quot;), times = 35)
split_vals &lt;- split(m, names(m))
df &lt;- cbind.data.frame(split_vals)
df &lt;- lapply(df, as.double) %&gt;% data.frame()

# merge
try &lt;- cbind(try[1:35,1:2], df[,c(2,1,4,3)])
try[1:5,]</code></pre>
<pre><code>  type        dist    OR lower_lim upper_lim   p_value
1 Var3 dist_next_2 20.49      6.76     71.07  1.92e-10
2 Var3 dist_next_3 46.46     29.77     74.30 2.07e-109
3 Var3 dist_next_4  7.93      6.14     10.28  2.02e-67
4 Var3 dist_next_5 42.32     30.30     59.74 4.33e-166
5 Var3 dist_next_6 32.81     23.66     46.02 1.78e-143</code></pre>
<pre class="r"><code>try &lt;- try %&gt;% mutate(IPD = as.numeric(sub(&quot;.*_&quot;, &quot;&quot;, dist)))</code></pre>
<pre><code>Warning: There was 1 warning in `mutate()`.
ℹ In argument: `IPD = as.numeric(sub(&quot;.*_&quot;, &quot;&quot;, dist))`.
Caused by warning:
! NAs introduced by coercion</code></pre>
<pre class="r"><code>try %&gt;%
  ggplot(aes(x = IPD, y = OR)) +
  geom_point()</code></pre>
<pre><code>Warning: Removed 1 row containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
<p><img src="figure/distNextCpG.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" />
* this is basically one of Pete’s plots * comethylation is strongest for
smaller IPD’s, steadies out at IPD = 20 * this is consistent with Pete’s
work</p>
<ul>
<li><p>this was for reads with only 2 cpgs</p></li>
<li><p>try for more</p></li>
<li><p>need to split the meth pattern into groups of 2</p></li>
</ul>
<pre class="r"><code>chisq_table &lt;- for_beta %&gt;%
  filter(total == 3) %&gt;%
  group_by(read_id) %&gt;%
  mutate(dist_to_next = lead(start) - start,
         meth_pattern = paste0(meth_status, lead(meth_status))) %&gt;%
  ungroup() %&gt;%
  fill(dist_to_next, .direction = &quot;down&quot;) %&gt;%
  distinct(read_id, meth_pattern, dist_to_next) %&gt;%
  group_by(dist_to_next, meth_pattern) %&gt;%
  summarise(count = n()) %&gt;%
  ungroup() %&gt;%
  mutate(cpg_1 = substr(meth_pattern, 1, 1),
         cpg_2 = substr(meth_pattern, 2, 2)) %&gt;% 
  filter(!grepl(&quot;A&quot;, .$meth_pattern)) %&gt;%
  group_by(dist_to_next) %&gt;%
  reframe(tab = list(xtabs(count ~ cpg_1 + cpg_2, data = cur_data())))</code></pre>
<pre><code>`summarise()` has grouped output by &#39;dist_to_next&#39;. You can override using the
`.groups` argument.</code></pre>
<pre class="r"><code>chisq_table &lt;- chisq_table$tab 
names(chisq_table) &lt;- for_beta %&gt;%
  filter(total == 3) %&gt;%
  group_by(read_id) %&gt;%
  mutate(dist_to_next = lead(start) - start,
         meth_pattern = paste0(meth_status, lead(meth_status))) %&gt;%
  ungroup() %&gt;%
  fill(dist_to_next, .direction = &quot;down&quot;) %&gt;%
  distinct(read_id, meth_pattern, dist_to_next) %&gt;%
  filter(!grepl(&quot;A&quot;, .$meth_pattern)) %&gt;%
  group_by(dist_to_next, meth_pattern) %&gt;%
  summarise(count = n()) %&gt;% 
  mutate(dist_to_next = paste0(&quot;dist_next_&quot;, dist_to_next)) %&gt;%
  .$dist_to_next %&gt;% unique()</code></pre>
<pre><code>`summarise()` has grouped output by &#39;dist_to_next&#39;. You can override using the
`.groups` argument.</code></pre>
<pre class="r"><code>chisq_table[1:2]</code></pre>
<pre><code>$dist_next_2
     cpg_2
cpg_1    M    U
    M 1324   99
    U  124 1353

$dist_next_3
     cpg_2
cpg_1    M    U
    M 1146  281
    U  200 1857</code></pre>
<pre class="r"><code>chisq_table &lt;- simplify2array(chisq_table)</code></pre>
<pre class="r"><code>try3 &lt;- capture.output(epiDisplay::mhor(mhtable = chisq_table, graph = FALSE))

try3 &lt;- try3 %&gt;% data.frame() %&gt;% .[4:36,] %&gt;% data.frame()
colnames(try3) &lt;- &quot;all&quot;


# sep the words and the numbers
try3 &lt;- try3 %&gt;% separate_wider_delim(cols = all, delim = &quot; &quot;, names = c(&quot;type&quot;, &quot;dist&quot;, &quot;numbers&quot;), too_few = &quot;align_start&quot;, too_many = &quot;merge&quot;)

# sep the numbers
numeric_lines &lt;- grep(&quot;-?\\d+\\.\\d{2,}.*-?\\d+\\.\\d{2,}.*-?\\d+\\.\\d{2,}&quot;, 
                      try3, value = TRUE)
# to maintain the p-val
pattern &lt;- &quot;-?\\d+\\.\\d{2,}|[-+]?\\d+\\.?\\d*(?:[eE][-+]?\\d+)&quot;

vals &lt;- regmatches(
  numeric_lines,
  gregexpr(pattern, numeric_lines))

m &lt;- do.call(rbind, vals)
m &lt;- as.vector(m)
names(m) &lt;- rep(c(&quot;OR&quot;, &quot;lower_lim&quot;, &quot;upper_lim&quot;, &quot;p_value&quot;), times = 33)
split_vals &lt;- split(m, names(m))
df &lt;- cbind.data.frame(split_vals)
df &lt;- lapply(df, as.double) %&gt;% data.frame()

# merge
try3 &lt;- cbind(try3[,1:2], df[,c(2,1,4,3)])
try3 &lt;- try3 %&gt;% mutate(IPD = as.numeric(sub(&quot;.*_&quot;, &quot;&quot;, dist)))</code></pre>
<pre><code>Warning: There was 1 warning in `mutate()`.
ℹ In argument: `IPD = as.numeric(sub(&quot;.*_&quot;, &quot;&quot;, dist))`.
Caused by warning:
! NAs introduced by coercion</code></pre>
<pre class="r"><code>try3[1:5,]</code></pre>
<pre><code>  type        dist     OR lower_lim upper_lim p_value IPD
1 Var3 dist_next_2 144.99    109.62    194.44       0   2
2 Var3 dist_next_3  37.81     30.97     46.25       0   3
3 Var3 dist_next_4  35.22     29.70     41.93       0   4
4 Var3 dist_next_5  37.85     31.46     45.70       0   5
5 Var3 dist_next_6  36.03     30.16     43.16       0   6</code></pre>
<pre class="r"><code>try3 %&gt;%
  ggplot(aes(x = IPD, y = OR)) +
  geom_point()</code></pre>
<pre><code>Warning: Removed 1 row containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
<p><img src="figure/distNextCpG.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" />
* why is it so different when the read has 3 cpgs? - I’m doing it the
same way</p>
<pre class="r"><code>mantel_haen_dist_next_cpg &lt;- function(long_data, num_cpg_read = NULL) {
  if(!is.null(num_cpg_read)) {
    format_data &lt;- long_data %&gt;%
        filter(total == num_cpg_read) 
  } else {format_data &lt;- long_data}
  format_data &lt;- format_data %&gt;%
      group_by(read_id) %&gt;%
      mutate(dist_to_next = lead(start) - start,
            meth_pattern = paste0(meth_status, lead(meth_status))) %&gt;%
      ungroup() %&gt;%
      fill(dist_to_next, .direction = &quot;down&quot;) %&gt;%
      distinct(read_id, meth_pattern, dist_to_next) %&gt;%
      filter(!grepl(&quot;A&quot;, .$meth_pattern)) %&gt;%
      group_by(dist_to_next, meth_pattern) %&gt;%
      summarise(count = n()) %&gt;%
      ungroup()
  
  chisq_table &lt;- format_data %&gt;%
      mutate(cpg_1 = substr(meth_pattern, 1, 1),
             cpg_2 = substr(meth_pattern, 2, 2)) 
  # check for missing patterns (it screws up the chi test) and replace with 0
  missing_mm &lt;- chisq_table %&gt;% group_by(dist_to_next) %&gt;% filter(!any(grepl(&quot;MM&quot;, meth_pattern))) %&gt;% distinct(dist_to_next) %&gt;% ungroup() %&gt;% .$dist_to_next
  for(i in 1:length(missing_mm)) {
    chisq_table &lt;- rbind(chisq_table, c(missing_mm[i], &quot;MM&quot;, 0, &quot;M&quot;, &quot;M&quot;))
  }
  missing_mu &lt;- chisq_table %&gt;% group_by(dist_to_next) %&gt;% filter(!any(grepl(&quot;MU&quot;, meth_pattern))) %&gt;% distinct(dist_to_next) %&gt;% ungroup() %&gt;% .$dist_to_next
  for(i in 1:length(missing_mu)) {
    chisq_table &lt;- rbind(chisq_table, c(missing_mu[i], &quot;MU&quot;, 0, &quot;M&quot;, &quot;U&quot;))
  }
  missing_um &lt;- chisq_table %&gt;% group_by(dist_to_next) %&gt;% filter(!any(grepl(&quot;UM&quot;, meth_pattern))) %&gt;% distinct(dist_to_next) %&gt;% ungroup() %&gt;% .$dist_to_next
  for(i in 1:length(missing_um)) {
    chisq_table &lt;- rbind(chisq_table, c(missing_um[i], &quot;UM&quot;, 0, &quot;U&quot;, &quot;M&quot;))
  }
  missing_uu &lt;- chisq_table %&gt;% group_by(dist_to_next) %&gt;% filter(!any(grepl(&quot;UU&quot;, meth_pattern))) %&gt;% distinct(dist_to_next) %&gt;% ungroup() %&gt;% .$dist_to_next
  for(i in 1:length(missing_uu)) {
    chisq_table &lt;- rbind(chisq_table, c(missing_uu[i], &quot;UU&quot;, 0, &quot;U&quot;, &quot;U&quot;))
  }

  chisq_table$count &lt;- as.integer(chisq_table$count)
  chisq_table$dist_to_next &lt;- as.integer(chisq_table$dist_to_next)
  chisq_table &lt;- chisq_table %&gt;% filter(!is.na(dist_to_next))
  
  chisq_table &lt;- chisq_table %&gt;% 
      group_by(dist_to_next) %&gt;%
      reframe(tab = list(xtabs(count ~ cpg_1 + cpg_2, data = cur_data())))
  chisq_table &lt;- chisq_table$tab
  
  names(chisq_table) &lt;- format_data %&gt;%
      mutate(dist_to_next = paste0(&quot;dist_next_&quot;, dist_to_next)) %&gt;%
      .$dist_to_next %&gt;% 
    unique()
  distinct_rows &lt;- format_data$dist_to_next %&gt;% 
    unique() %&gt;%
    length()
  chisq_table &lt;- simplify2array(chisq_table)
  
  # the testing
  try &lt;- capture.output(epiDisplay::mhor(mhtable = chisq_table, graph = FALSE))

  try &lt;- try %&gt;% data.frame() %&gt;% .[4:(distinct_rows + 4),] %&gt;% data.frame()
  colnames(try) &lt;- &quot;all&quot;

  try &lt;- try %&gt;% separate_wider_delim(cols = all, delim = &quot; &quot;, names = c(&quot;type&quot;, &quot;dist&quot;, &quot;numbers&quot;), too_few = &quot;align_start&quot;, too_many = &quot;merge&quot;)

  numeric_lines &lt;- try$numbers
  # get Inf and d separate
  pattern &lt;- &quot;Inf|-Inf|[-+]?\\d+\\.?\\d*(?:[eE][-+]?\\d+)?&quot;
  vals &lt;- regmatches(
    numeric_lines,
    gregexpr(pattern, numeric_lines))

  m &lt;- do.call(rbind, vals)
  m &lt;- as.vector(m)
  names(m) &lt;- rep(c(&quot;OR&quot;, &quot;lower_lim&quot;, &quot;upper_lim&quot;, &quot;p_value&quot;), times = (distinct_rows + 1))
  split_vals &lt;- split(m, names(m))
  df &lt;- cbind.data.frame(split_vals)
  df &lt;- lapply(df, as.double) %&gt;% data.frame()

  # merge
  try &lt;- cbind(try[,1:2], df[,c(2,1,4,3)])
  try &lt;- try %&gt;% mutate(IPD = as.numeric(sub(&quot;.*_&quot;, &quot;&quot;, dist)))
  try
}
mantel_haen_dist_next_cpg(for_beta, 7) %&gt;% .[1:5,]</code></pre>
<pre><code>`summarise()` has grouped output by &#39;dist_to_next&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code>Warning in rbind(deparse.level, ...): number of columns of result, 5, is not a
multiple of vector length 4 of arg 2</code></pre>
<pre><code>Warning in mantel_haen_dist_next_cpg(for_beta, 7): NAs introduced by coercion
Warning in mantel_haen_dist_next_cpg(for_beta, 7): NAs introduced by coercion</code></pre>
<pre><code>Warning: There was 1 warning in `mutate()`.
ℹ In argument: `IPD = as.numeric(sub(&quot;.*_&quot;, &quot;&quot;, dist))`.
Caused by warning:
! NAs introduced by coercion</code></pre>
<pre><code>  type        dist   OR lower_lim upper_lim p_value IPD
1 Var3 dist_next_2 49.1      41.0      19.7  45.900   2
2 Var3 dist_next_3 25.7      59.5      41.1  50.700   3
3 Var3 dist_next_4 16.6      24.9     148.6  31.700   4
4 Var3 dist_next_5 36.9       4.5       0.0     Inf   5
5 Var3 dist_next_6  0.0       0.0      35.9  41.064   6</code></pre>
<pre class="r"><code>cpgs_read &lt;- for_beta %&gt;% distinct(total) %&gt;% .[order(.$total),]</code></pre>
<pre class="r"><code>mantel_haen_dist_next_cpg(for_beta, cpgs_read[2]) %&gt;%
    ggplot(aes(x = IPD, y = OR)) +
    geom_point() +
    labs(title = paste0(&quot;total cpgs on read = &quot;, cpgs_read[2]))</code></pre>
<pre><code>`summarise()` has grouped output by &#39;dist_to_next&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code>Warning in rbind(deparse.level, ...): number of columns of result, 5, is not a
multiple of vector length 4 of arg 2</code></pre>
<pre><code>Warning in mantel_haen_dist_next_cpg(for_beta, cpgs_read[2]): NAs introduced by
coercion
Warning in mantel_haen_dist_next_cpg(for_beta, cpgs_read[2]): NAs introduced by
coercion</code></pre>
<pre><code>Warning: There was 1 warning in `mutate()`.
ℹ In argument: `IPD = as.numeric(sub(&quot;.*_&quot;, &quot;&quot;, dist))`.
Caused by warning:
! NAs introduced by coercion</code></pre>
<pre><code>Warning: Removed 1 row containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
<p><img src="figure/distNextCpG.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot_vec &lt;- list()
for(x in 2:length(cpgs_read)) {
  plot &lt;- mantel_haen_dist_next_cpg(for_beta, cpgs_read[x]) %&gt;%
    ggplot(aes(x = IPD, y = OR)) +
    geom_point() +
    labs(title = paste0(&quot;total cpgs on read = &quot;, cpgs_read[x]))
  plot_vec &lt;- list(plot_vec, plot)
}</code></pre>
<pre><code>`summarise()` has grouped output by &#39;dist_to_next&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code>Warning in rbind(deparse.level, ...): number of columns of result, 5, is not a
multiple of vector length 4 of arg 2</code></pre>
<pre><code>Warning in mantel_haen_dist_next_cpg(for_beta, cpgs_read[x]): NAs introduced by
coercion
Warning in mantel_haen_dist_next_cpg(for_beta, cpgs_read[x]): NAs introduced by
coercion</code></pre>
<pre><code>Warning: There was 1 warning in `mutate()`.
ℹ In argument: `IPD = as.numeric(sub(&quot;.*_&quot;, &quot;&quot;, dist))`.
Caused by warning:
! NAs introduced by coercion</code></pre>
<pre><code>`summarise()` has grouped output by &#39;dist_to_next&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code>Warning in rbind(deparse.level, ...): number of columns of result, 5, is not a
multiple of vector length 4 of arg 2</code></pre>
<pre><code>Warning in mantel_haen_dist_next_cpg(for_beta, cpgs_read[x]): NAs introduced by
coercion
Warning in mantel_haen_dist_next_cpg(for_beta, cpgs_read[x]): NAs introduced by
coercion</code></pre>
<pre><code>Warning: There was 1 warning in `mutate()`.
ℹ In argument: `IPD = as.numeric(sub(&quot;.*_&quot;, &quot;&quot;, dist))`.
Caused by warning:
! NAs introduced by coercion</code></pre>
<pre><code>`summarise()` has grouped output by &#39;dist_to_next&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code>Warning in rbind(deparse.level, ...): number of columns of result, 5, is not a
multiple of vector length 4 of arg 2</code></pre>
<pre><code>Warning in mantel_haen_dist_next_cpg(for_beta, cpgs_read[x]): NAs introduced by
coercion
Warning in mantel_haen_dist_next_cpg(for_beta, cpgs_read[x]): NAs introduced by
coercion</code></pre>
<pre><code>Warning: There was 1 warning in `mutate()`.
ℹ In argument: `IPD = as.numeric(sub(&quot;.*_&quot;, &quot;&quot;, dist))`.
Caused by warning:
! NAs introduced by coercion</code></pre>
<pre><code>`summarise()` has grouped output by &#39;dist_to_next&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code>Warning: There was 1 warning in `mutate()`.
ℹ In argument: `IPD = as.numeric(sub(&quot;.*_&quot;, &quot;&quot;, dist))`.
Caused by warning:
! NAs introduced by coercion</code></pre>
<pre><code>`summarise()` has grouped output by &#39;dist_to_next&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code>Warning in rbind(deparse.level, ...): number of columns of result, 5, is not a
multiple of vector length 4 of arg 2</code></pre>
<pre><code>Warning in mantel_haen_dist_next_cpg(for_beta, cpgs_read[x]): NAs introduced by
coercion
Warning in mantel_haen_dist_next_cpg(for_beta, cpgs_read[x]): NAs introduced by
coercion</code></pre>
<pre><code>Warning: There was 1 warning in `mutate()`.
ℹ In argument: `IPD = as.numeric(sub(&quot;.*_&quot;, &quot;&quot;, dist))`.
Caused by warning:
! NAs introduced by coercion</code></pre>
<pre><code>`summarise()` has grouped output by &#39;dist_to_next&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code>Warning in rbind(deparse.level, ...): number of columns of result, 5, is not a
multiple of vector length 4 of arg 2</code></pre>
<pre><code>Warning in mantel_haen_dist_next_cpg(for_beta, cpgs_read[x]): NAs introduced by
coercion
Warning in mantel_haen_dist_next_cpg(for_beta, cpgs_read[x]): NAs introduced by
coercion</code></pre>
<pre><code>Warning: There was 1 warning in `mutate()`.
ℹ In argument: `IPD = as.numeric(sub(&quot;.*_&quot;, &quot;&quot;, dist))`.
Caused by warning:
! NAs introduced by coercion</code></pre>
<pre><code>`summarise()` has grouped output by &#39;dist_to_next&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code>Warning in rbind(deparse.level, ...): number of columns of result, 5, is not a
multiple of vector length 4 of arg 2</code></pre>
<pre><code>Warning in mantel_haen_dist_next_cpg(for_beta, cpgs_read[x]): NAs introduced by
coercion
Warning in mantel_haen_dist_next_cpg(for_beta, cpgs_read[x]): NAs introduced by
coercion</code></pre>
<pre><code>Warning: There was 1 warning in `mutate()`.
ℹ In argument: `IPD = as.numeric(sub(&quot;.*_&quot;, &quot;&quot;, dist))`.
Caused by warning:
! NAs introduced by coercion</code></pre>
<pre><code>`summarise()` has grouped output by &#39;dist_to_next&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code>Warning in rbind(deparse.level, ...): number of columns of result, 5, is not a
multiple of vector length 4 of arg 2</code></pre>
<pre><code>Warning in mantel_haen_dist_next_cpg(for_beta, cpgs_read[x]): NAs introduced by
coercion
Warning in mantel_haen_dist_next_cpg(for_beta, cpgs_read[x]): NAs introduced by
coercion</code></pre>
<pre><code>Warning in rbind(...): number of columns of result is not a multiple of vector
length (arg 14)</code></pre>
<pre><code>Warning: There was 1 warning in `mutate()`.
ℹ In argument: `IPD = as.numeric(sub(&quot;.*_&quot;, &quot;&quot;, dist))`.
Caused by warning:
! NAs introduced by coercion</code></pre>
<pre><code>`summarise()` has grouped output by &#39;dist_to_next&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code>Warning in rbind(deparse.level, ...): number of columns of result, 5, is not a
multiple of vector length 4 of arg 2</code></pre>
<pre><code>Warning in mantel_haen_dist_next_cpg(for_beta, cpgs_read[x]): NAs introduced by
coercion
Warning in mantel_haen_dist_next_cpg(for_beta, cpgs_read[x]): NAs introduced by
coercion</code></pre>
<pre><code>Warning in rbind(...): number of columns of result is not a multiple of vector
length (arg 12)</code></pre>
<pre><code>Warning: There was 1 warning in `mutate()`.
ℹ In argument: `IPD = as.numeric(sub(&quot;.*_&quot;, &quot;&quot;, dist))`.
Caused by warning:
! NAs introduced by coercion</code></pre>
<pre><code>`summarise()` has grouped output by &#39;dist_to_next&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code>Warning in rbind(deparse.level, ...): number of columns of result, 5, is not a
multiple of vector length 4 of arg 2</code></pre>
<pre><code>Warning in mantel_haen_dist_next_cpg(for_beta, cpgs_read[x]): NAs introduced by
coercion
Warning in mantel_haen_dist_next_cpg(for_beta, cpgs_read[x]): NAs introduced by
coercion</code></pre>
<pre><code>Warning: There was 1 warning in `mutate()`.
ℹ In argument: `IPD = as.numeric(sub(&quot;.*_&quot;, &quot;&quot;, dist))`.
Caused by warning:
! NAs introduced by coercion</code></pre>
<pre class="r"><code>plot_vec</code></pre>
<pre><code>[[1]]
[[1]][[1]]
[[1]][[1]][[1]]
[[1]][[1]][[1]][[1]]
[[1]][[1]][[1]][[1]][[1]]
[[1]][[1]][[1]][[1]][[1]][[1]]
[[1]][[1]][[1]][[1]][[1]][[1]][[1]]
[[1]][[1]][[1]][[1]][[1]][[1]][[1]][[1]]
[[1]][[1]][[1]][[1]][[1]][[1]][[1]][[1]][[1]]
[[1]][[1]][[1]][[1]][[1]][[1]][[1]][[1]][[1]][[1]]
list()

[[1]][[1]][[1]][[1]][[1]][[1]][[1]][[1]][[1]][[2]]</code></pre>
<pre><code>Warning: Removed 1 row containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
<p><img src="figure/distNextCpG.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>

[[1]][[1]][[1]][[1]][[1]][[1]][[1]][[1]][[2]]</code></pre>
<pre><code>Warning: Removed 1 row containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
<p><img src="figure/distNextCpG.Rmd/unnamed-chunk-20-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>

[[1]][[1]][[1]][[1]][[1]][[1]][[1]][[2]]</code></pre>
<pre><code>Warning: Removed 1 row containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
<p><img src="figure/distNextCpG.Rmd/unnamed-chunk-20-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>

[[1]][[1]][[1]][[1]][[1]][[1]][[2]]</code></pre>
<pre><code>Warning: Removed 1 row containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
<p><img src="figure/distNextCpG.Rmd/unnamed-chunk-20-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>

[[1]][[1]][[1]][[1]][[1]][[2]]</code></pre>
<pre><code>Warning: Removed 1 row containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
<p><img src="figure/distNextCpG.Rmd/unnamed-chunk-20-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>

[[1]][[1]][[1]][[1]][[2]]</code></pre>
<pre><code>Warning: Removed 1 row containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
<p><img src="figure/distNextCpG.Rmd/unnamed-chunk-20-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>

[[1]][[1]][[1]][[2]]</code></pre>
<pre><code>Warning: Removed 1 row containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
<p><img src="figure/distNextCpG.Rmd/unnamed-chunk-20-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>

[[1]][[1]][[2]]</code></pre>
<pre><code>Warning: Removed 1 row containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
<p><img src="figure/distNextCpG.Rmd/unnamed-chunk-20-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>

[[1]][[2]]</code></pre>
<pre><code>Warning: Removed 1 row containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
<p><img src="figure/distNextCpG.Rmd/unnamed-chunk-20-9.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>

[[2]]</code></pre>
<pre><code>Warning: Removed 1 row containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
<p><img src="figure/distNextCpG.Rmd/unnamed-chunk-20-10.png" width="672" style="display: block; margin: auto;" /></p>
<ul>
<li><p>the plot looks different for 2 and 3 now</p></li>
<li><p>I don’t know why the values have changed</p></li>
<li><p>log the plots</p></li>
</ul>
<pre class="r"><code>plot_vec &lt;- list()
for(x in 2:length(cpgs_read)) {
  plot &lt;- mantel_haen_dist_next_cpg(for_beta, cpgs_read[x]) %&gt;%
    ggplot(aes(x = IPD, y = log2(OR))) +
    geom_point() +
    labs(title = paste0(&quot;total cpgs on read = &quot;, cpgs_read[x]))
  plot_vec &lt;- list(plot_vec, plot)
}</code></pre>
<pre><code>`summarise()` has grouped output by &#39;dist_to_next&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code>Warning in rbind(deparse.level, ...): number of columns of result, 5, is not a
multiple of vector length 4 of arg 2</code></pre>
<pre><code>Warning in mantel_haen_dist_next_cpg(for_beta, cpgs_read[x]): NAs introduced by
coercion
Warning in mantel_haen_dist_next_cpg(for_beta, cpgs_read[x]): NAs introduced by
coercion</code></pre>
<pre><code>Warning: There was 1 warning in `mutate()`.
ℹ In argument: `IPD = as.numeric(sub(&quot;.*_&quot;, &quot;&quot;, dist))`.
Caused by warning:
! NAs introduced by coercion</code></pre>
<pre><code>`summarise()` has grouped output by &#39;dist_to_next&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code>Warning in rbind(deparse.level, ...): number of columns of result, 5, is not a
multiple of vector length 4 of arg 2</code></pre>
<pre><code>Warning in mantel_haen_dist_next_cpg(for_beta, cpgs_read[x]): NAs introduced by
coercion
Warning in mantel_haen_dist_next_cpg(for_beta, cpgs_read[x]): NAs introduced by
coercion</code></pre>
<pre><code>Warning: There was 1 warning in `mutate()`.
ℹ In argument: `IPD = as.numeric(sub(&quot;.*_&quot;, &quot;&quot;, dist))`.
Caused by warning:
! NAs introduced by coercion</code></pre>
<pre><code>`summarise()` has grouped output by &#39;dist_to_next&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code>Warning in rbind(deparse.level, ...): number of columns of result, 5, is not a
multiple of vector length 4 of arg 2</code></pre>
<pre><code>Warning in mantel_haen_dist_next_cpg(for_beta, cpgs_read[x]): NAs introduced by
coercion
Warning in mantel_haen_dist_next_cpg(for_beta, cpgs_read[x]): NAs introduced by
coercion</code></pre>
<pre><code>Warning: There was 1 warning in `mutate()`.
ℹ In argument: `IPD = as.numeric(sub(&quot;.*_&quot;, &quot;&quot;, dist))`.
Caused by warning:
! NAs introduced by coercion</code></pre>
<pre><code>`summarise()` has grouped output by &#39;dist_to_next&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code>Warning: There was 1 warning in `mutate()`.
ℹ In argument: `IPD = as.numeric(sub(&quot;.*_&quot;, &quot;&quot;, dist))`.
Caused by warning:
! NAs introduced by coercion</code></pre>
<pre><code>`summarise()` has grouped output by &#39;dist_to_next&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code>Warning in rbind(deparse.level, ...): number of columns of result, 5, is not a
multiple of vector length 4 of arg 2</code></pre>
<pre><code>Warning in mantel_haen_dist_next_cpg(for_beta, cpgs_read[x]): NAs introduced by
coercion
Warning in mantel_haen_dist_next_cpg(for_beta, cpgs_read[x]): NAs introduced by
coercion</code></pre>
<pre><code>Warning: There was 1 warning in `mutate()`.
ℹ In argument: `IPD = as.numeric(sub(&quot;.*_&quot;, &quot;&quot;, dist))`.
Caused by warning:
! NAs introduced by coercion</code></pre>
<pre><code>`summarise()` has grouped output by &#39;dist_to_next&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code>Warning in rbind(deparse.level, ...): number of columns of result, 5, is not a
multiple of vector length 4 of arg 2</code></pre>
<pre><code>Warning in mantel_haen_dist_next_cpg(for_beta, cpgs_read[x]): NAs introduced by
coercion
Warning in mantel_haen_dist_next_cpg(for_beta, cpgs_read[x]): NAs introduced by
coercion</code></pre>
<pre><code>Warning: There was 1 warning in `mutate()`.
ℹ In argument: `IPD = as.numeric(sub(&quot;.*_&quot;, &quot;&quot;, dist))`.
Caused by warning:
! NAs introduced by coercion</code></pre>
<pre><code>`summarise()` has grouped output by &#39;dist_to_next&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code>Warning in rbind(deparse.level, ...): number of columns of result, 5, is not a
multiple of vector length 4 of arg 2</code></pre>
<pre><code>Warning in mantel_haen_dist_next_cpg(for_beta, cpgs_read[x]): NAs introduced by
coercion
Warning in mantel_haen_dist_next_cpg(for_beta, cpgs_read[x]): NAs introduced by
coercion</code></pre>
<pre><code>Warning: There was 1 warning in `mutate()`.
ℹ In argument: `IPD = as.numeric(sub(&quot;.*_&quot;, &quot;&quot;, dist))`.
Caused by warning:
! NAs introduced by coercion</code></pre>
<pre><code>`summarise()` has grouped output by &#39;dist_to_next&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code>Warning in rbind(deparse.level, ...): number of columns of result, 5, is not a
multiple of vector length 4 of arg 2</code></pre>
<pre><code>Warning in mantel_haen_dist_next_cpg(for_beta, cpgs_read[x]): NAs introduced by
coercion
Warning in mantel_haen_dist_next_cpg(for_beta, cpgs_read[x]): NAs introduced by
coercion</code></pre>
<pre><code>Warning in rbind(...): number of columns of result is not a multiple of vector
length (arg 14)</code></pre>
<pre><code>Warning: There was 1 warning in `mutate()`.
ℹ In argument: `IPD = as.numeric(sub(&quot;.*_&quot;, &quot;&quot;, dist))`.
Caused by warning:
! NAs introduced by coercion</code></pre>
<pre><code>`summarise()` has grouped output by &#39;dist_to_next&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code>Warning in rbind(deparse.level, ...): number of columns of result, 5, is not a
multiple of vector length 4 of arg 2</code></pre>
<pre><code>Warning in mantel_haen_dist_next_cpg(for_beta, cpgs_read[x]): NAs introduced by
coercion
Warning in mantel_haen_dist_next_cpg(for_beta, cpgs_read[x]): NAs introduced by
coercion</code></pre>
<pre><code>Warning in rbind(...): number of columns of result is not a multiple of vector
length (arg 12)</code></pre>
<pre><code>Warning: There was 1 warning in `mutate()`.
ℹ In argument: `IPD = as.numeric(sub(&quot;.*_&quot;, &quot;&quot;, dist))`.
Caused by warning:
! NAs introduced by coercion</code></pre>
<pre><code>`summarise()` has grouped output by &#39;dist_to_next&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code>Warning in rbind(deparse.level, ...): number of columns of result, 5, is not a
multiple of vector length 4 of arg 2</code></pre>
<pre><code>Warning in mantel_haen_dist_next_cpg(for_beta, cpgs_read[x]): NAs introduced by
coercion
Warning in mantel_haen_dist_next_cpg(for_beta, cpgs_read[x]): NAs introduced by
coercion</code></pre>
<pre><code>Warning: There was 1 warning in `mutate()`.
ℹ In argument: `IPD = as.numeric(sub(&quot;.*_&quot;, &quot;&quot;, dist))`.
Caused by warning:
! NAs introduced by coercion</code></pre>
<pre class="r"><code>plot_vec</code></pre>
<pre><code>[[1]]
[[1]][[1]]
[[1]][[1]][[1]]
[[1]][[1]][[1]][[1]]
[[1]][[1]][[1]][[1]][[1]]
[[1]][[1]][[1]][[1]][[1]][[1]]
[[1]][[1]][[1]][[1]][[1]][[1]][[1]]
[[1]][[1]][[1]][[1]][[1]][[1]][[1]][[1]]
[[1]][[1]][[1]][[1]][[1]][[1]][[1]][[1]][[1]]
[[1]][[1]][[1]][[1]][[1]][[1]][[1]][[1]][[1]][[1]]
list()

[[1]][[1]][[1]][[1]][[1]][[1]][[1]][[1]][[1]][[2]]</code></pre>
<pre><code>Warning: Removed 1 row containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
<p><img src="figure/distNextCpG.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>

[[1]][[1]][[1]][[1]][[1]][[1]][[1]][[1]][[2]]</code></pre>
<pre><code>Warning: Removed 1 row containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
<p><img src="figure/distNextCpG.Rmd/unnamed-chunk-21-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>

[[1]][[1]][[1]][[1]][[1]][[1]][[1]][[2]]</code></pre>
<pre><code>Warning: Removed 1 row containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
<p><img src="figure/distNextCpG.Rmd/unnamed-chunk-21-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>

[[1]][[1]][[1]][[1]][[1]][[1]][[2]]</code></pre>
<pre><code>Warning: Removed 1 row containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
<p><img src="figure/distNextCpG.Rmd/unnamed-chunk-21-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>

[[1]][[1]][[1]][[1]][[1]][[2]]</code></pre>
<pre><code>Warning: Removed 1 row containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
<p><img src="figure/distNextCpG.Rmd/unnamed-chunk-21-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>

[[1]][[1]][[1]][[1]][[2]]</code></pre>
<pre><code>Warning: Removed 1 row containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
<p><img src="figure/distNextCpG.Rmd/unnamed-chunk-21-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>

[[1]][[1]][[1]][[2]]</code></pre>
<pre><code>Warning: Removed 1 row containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
<p><img src="figure/distNextCpG.Rmd/unnamed-chunk-21-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>

[[1]][[1]][[2]]</code></pre>
<pre><code>Warning: Removed 1 row containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
<p><img src="figure/distNextCpG.Rmd/unnamed-chunk-21-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>

[[1]][[2]]</code></pre>
<pre><code>Warning: Removed 1 row containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
<p><img src="figure/distNextCpG.Rmd/unnamed-chunk-21-9.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>

[[2]]</code></pre>
<pre><code>Warning: Removed 1 row containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
<p><img src="figure/distNextCpG.Rmd/unnamed-chunk-21-10.png" width="672" style="display: block; margin: auto;" /></p>
<ul>
<li><p>I think Pete’s plots are on the log axis but they are very
different to mine</p></li>
<li><p>they’re positive and small log numbers</p></li>
<li><p>all reads at once</p></li>
</ul>
<pre class="r"><code>mantel_haen_dist_next_cpg(for_beta) %&gt;%
    ggplot(aes(x = IPD, y = OR)) +
    geom_point() +
    labs(title = paste0(&quot;all reads together&quot;))</code></pre>
<pre><code>`summarise()` has grouped output by &#39;dist_to_next&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code>Warning in rbind(deparse.level, ...): number of columns of result, 5, is not a
multiple of vector length 4 of arg 2</code></pre>
<pre><code>Warning in mantel_haen_dist_next_cpg(for_beta): NAs introduced by coercion
Warning in mantel_haen_dist_next_cpg(for_beta): NAs introduced by coercion</code></pre>
<pre><code>Warning: There was 1 warning in `mutate()`.
ℹ In argument: `IPD = as.numeric(sub(&quot;.*_&quot;, &quot;&quot;, dist))`.
Caused by warning:
! NAs introduced by coercion</code></pre>
<pre><code>Warning: Removed 1 row containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
<p><img src="figure/distNextCpG.Rmd/unnamed-chunk-22-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.2 (2024-10-31)
Platform: aarch64-apple-darwin20
Running under: macOS Sequoia 15.2

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Australia/Melbourne
tzcode source: internal

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] ggplot2_3.5.2        dplyr_1.1.4          stringr_1.5.1       
 [4] tidyr_1.3.1          plyranges_1.26.0     GenomicRanges_1.58.0
 [7] GenomeInfoDb_1.42.3  IRanges_2.40.1       S4Vectors_0.44.0    
[10] BiocGenerics_0.52.0  data.table_1.17.4    annotatr_1.32.0     

loaded via a namespace (and not attached):
  [1] DBI_1.2.3                         bitops_1.0-9                     
  [3] rlang_1.1.6                       magrittr_2.0.3                   
  [5] git2r_0.36.2                      matrixStats_1.5.0                
  [7] compiler_4.4.2                    RSQLite_2.4.0                    
  [9] GenomicFeatures_1.58.0            png_0.1-8                        
 [11] vctrs_0.6.5                       reshape2_1.4.4                   
 [13] pkgconfig_2.0.3                   crayon_1.5.3                     
 [15] fastmap_1.2.0                     dbplyr_2.5.1                     
 [17] XVector_0.46.0                    labeling_0.4.3                   
 [19] utf8_1.2.5                        Rsamtools_2.22.0                 
 [21] promises_1.3.3                    rmarkdown_2.29                   
 [23] tzdb_0.5.0                        UCSC.utils_1.2.0                 
 [25] purrr_1.0.4                       bit_4.6.0                        
 [27] BSgenome.Hsapiens.UCSC.hg38_1.4.5 xfun_0.52                        
 [29] zlibbioc_1.52.0                   cachem_1.1.0                     
 [31] jsonlite_2.0.0                    blob_1.2.4                       
 [33] later_1.4.2                       DelayedArray_0.32.0              
 [35] BiocParallel_1.40.2               parallel_4.4.2                   
 [37] R6_2.6.1                          bslib_0.9.0                      
 [39] stringi_1.8.7                     RColorBrewer_1.1-3               
 [41] rtracklayer_1.66.0                jquerylib_0.1.4                  
 [43] Rcpp_1.0.14                       SummarizedExperiment_1.36.0      
 [45] knitr_1.50                        readr_2.1.5                      
 [47] nnet_7.3-19                       splines_4.4.2                    
 [49] httpuv_1.6.16                     Matrix_1.7-1                     
 [51] tidyselect_1.2.1                  rstudioapi_0.17.1                
 [53] abind_1.4-8                       yaml_2.3.10                      
 [55] codetools_0.2-20                  curl_6.2.3                       
 [57] lattice_0.22-6                    tibble_3.2.1                     
 [59] regioneR_1.38.0                   plyr_1.8.9                       
 [61] withr_3.0.2                       Biobase_2.66.0                   
 [63] KEGGREST_1.46.0                   evaluate_1.0.3                   
 [65] foreign_0.8-87                    survival_3.7-0                   
 [67] BiocFileCache_2.14.0              Biostrings_2.74.1                
 [69] pillar_1.10.2                     BiocManager_1.30.25              
 [71] filelock_1.0.3                    MatrixGenerics_1.18.1            
 [73] whisker_0.4.1                     generics_0.1.4                   
 [75] rprojroot_2.0.4                   RCurl_1.98-1.17                  
 [77] BiocVersion_3.20.0                hms_1.1.3                        
 [79] scales_1.4.0                      glue_1.8.0                       
 [81] tools_4.4.2                       AnnotationHub_3.14.0             
 [83] BiocIO_1.16.0                     BSgenome_1.74.0                  
 [85] GenomicAlignments_1.42.0          fs_1.6.6                         
 [87] XML_3.99-0.18                     grid_4.4.2                       
 [89] AnnotationDbi_1.68.0              GenomeInfoDbData_1.2.13          
 [91] restfulr_0.0.15                   cli_3.6.5                        
 [93] rappdirs_0.3.3                    workflowr_1.7.1                  
 [95] S4Arrays_1.6.0                    epiDisplay_3.5.0.2               
 [97] gtable_0.3.6                      sass_0.4.10                      
 [99] digest_0.6.37                     SparseArray_1.6.2                
[101] rjson_0.2.23                      farver_2.1.2                     
[103] memoise_2.0.1                     htmltools_0.5.8.1                
[105] lifecycle_1.0.4                   httr_1.4.7                       
[107] MASS_7.3-61                       bit64_4.6.0-1                    </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
