<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="caitlinpage" />

<meta name="date" content="2026-01-23" />

<title>distAlphaByDistance</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">3_cf_nano</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/caitlinpage/methyl_nano_cf">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">distAlphaByDistance</h1>
<h4 class="author">caitlinpage</h4>
<h4 class="date">2026-01-23</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2026-01-27
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>methyl_nano_cf/analysis/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250606code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250606)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250606code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250606)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomcaitlinpagemethylnanocftree94342b6bcca309b64825804e96b7fb641cd18312targetblank94342b6a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/caitlinpage/methyl_nano_cf/tree/94342b6bcca309b64825804e96b7fb641cd18312" target="_blank">94342b6</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomcaitlinpagemethylnanocftree94342b6bcca309b64825804e96b7fb641cd18312targetblank94342b6a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/caitlinpage/methyl_nano_cf/tree/94342b6bcca309b64825804e96b7fb641cd18312" target="_blank">94342b6</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  renv.lock
    Untracked:  renv/

Unstaged changes:
    Modified:   .Rprofile
    Modified:   _workflowr.yml
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/distAlphaByDistance.Rmd</code>)
and HTML (<code>docs/distAlphaByDistance.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/caitlinpage/methyl_nano_cf/blob/94342b6bcca309b64825804e96b7fb641cd18312/analysis/distAlphaByDistance.Rmd" target="_blank">94342b6</a>
</td>
<td>
caitlinpage
</td>
<td>
2026-01-23
</td>
<td>
building on cometh
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<ul>
<li>Building off comethylation mantel-haensz work from <a
href="distNextCpG.html">distNextCpg</a></li>
</ul>
<p>GOAL: Distribution of alpha by distance: 2 cpgs close to each other
should be 1 or 0, 2 cpgs far away should be 0.5 Density of cpgs on
read</p>
<pre class="r"><code>library(annotatr)</code></pre>
<pre><code>Warning: replacing previous import &#39;S4Arrays::makeNindexFromArrayViewport&#39; by
&#39;DelayedArray::makeNindexFromArrayViewport&#39; when loading &#39;SummarizedExperiment&#39;</code></pre>
<pre class="r"><code>library(data.table)
library(plyranges)</code></pre>
<pre><code>Loading required package: BiocGenerics</code></pre>
<pre><code>
Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    IQR, mad, sd, var, xtabs</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    anyDuplicated, aperm, append, as.data.frame, basename, cbind,
    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,
    get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,
    match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,
    Position, rank, rbind, Reduce, rownames, sapply, saveRDS, setdiff,
    table, tapply, union, unique, unsplit, which.max, which.min</code></pre>
<pre><code>Loading required package: IRanges</code></pre>
<pre><code>Loading required package: S4Vectors</code></pre>
<pre><code>Loading required package: stats4</code></pre>
<pre><code>
Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:data.table&#39;:

    first, second</code></pre>
<pre><code>The following object is masked from &#39;package:utils&#39;:

    findMatches</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    expand.grid, I, unname</code></pre>
<pre><code>
Attaching package: &#39;IRanges&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:data.table&#39;:

    shift</code></pre>
<pre><code>Loading required package: GenomicRanges</code></pre>
<pre><code>Loading required package: GenomeInfoDb</code></pre>
<pre><code>
Attaching package: &#39;plyranges&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:IRanges&#39;:

    slice</code></pre>
<pre><code>The following object is masked from &#39;package:data.table&#39;:

    between</code></pre>
<pre><code>The following object is masked from &#39;package:stats&#39;:

    filter</code></pre>
<pre class="r"><code>library(tidyr)</code></pre>
<pre><code>
Attaching package: &#39;tidyr&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:S4Vectors&#39;:

    expand</code></pre>
<pre class="r"><code>library(stringr)
library(dplyr)</code></pre>
<pre><code>
Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:plyranges&#39;:

    between, n, n_distinct</code></pre>
<pre><code>The following objects are masked from &#39;package:GenomicRanges&#39;:

    intersect, setdiff, union</code></pre>
<pre><code>The following object is masked from &#39;package:GenomeInfoDb&#39;:

    intersect</code></pre>
<pre><code>The following objects are masked from &#39;package:IRanges&#39;:

    collapse, desc, intersect, setdiff, slice, union</code></pre>
<pre><code>The following objects are masked from &#39;package:S4Vectors&#39;:

    first, intersect, rename, setdiff, setequal, union</code></pre>
<pre><code>The following objects are masked from &#39;package:BiocGenerics&#39;:

    combine, intersect, setdiff, union</code></pre>
<pre><code>The following objects are masked from &#39;package:data.table&#39;:

    between, first, last</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    filter, lag</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    intersect, setdiff, setequal, union</code></pre>
<pre class="r"><code>library(ggplot2)</code></pre>
</div>
<div id="data-objects" class="section level2">
<h2>Data Objects</h2>
<pre class="r"><code>cg_sites &lt;- cbind(data.frame(Biostrings::matchPattern(&quot;CG&quot;, BSgenome.Hsapiens.UCSC.hg38::BSgenome.Hsapiens.UCSC.hg38[[&quot;chr22&quot;]])), seqnames = &quot;chr22&quot;) %&gt;% 
  mutate(pos = paste0(seqnames, &quot;-&quot;, start)) %&gt;%
  relocate(pos, seqnames)</code></pre>
<pre><code>Warning in .local(x, row.names, optional, ...): &#39;optional&#39; argument was ignored</code></pre>
<pre class="r"><code>cg_sites$index &lt;- 1:nrow(cg_sites)</code></pre>
<pre class="r"><code>source(&quot;../code/processData.R&quot;)
for_beta &lt;- readRDS(&quot;/researchers/caitlin.page/cf_nano/r_output/rrbs_alpha_beta.rds&quot;)
#for_beta &lt;- readRDS(&quot;../output/cluster_files/rrbs_alpha_beta.rds&quot;)
for_beta &lt;- for_beta %&gt;% filter(seqnames == &quot;chr22&quot;)
rrbs_for_beta_pat &lt;- pat_for_wgbs(for_beta)</code></pre>
<pre><code>`summarise()` has grouped output by &#39;seqnames&#39;, &#39;start_pos&#39;, &#39;meth_pattern&#39;,
&#39;alpha&#39;, &#39;end_pos&#39;. You can override using the `.groups` argument.</code></pre>
<pre class="r"><code>for_beta &lt;- for_beta %&gt;% group_by(start) %&gt;% mutate(coverage = n(), mean_alpha = mean(alpha)) %&gt;% ungroup() %&gt;% data.frame()
just_beta &lt;- for_beta %&gt;% distinct(start, beta, coverage, mean_alpha) %&gt;% .[order(.$start),]</code></pre>
<pre class="r"><code>for_beta[1:10,]</code></pre>
<pre><code>                              read_id strand seqnames    start meth_status
1  HWI-EAS149_7:1:100:1003:1410:0:1:1      +    chr22 39994598           M
2  HWI-EAS149_7:1:100:1003:1410:0:1:1      -    chr22 39994607           U
3  HWI-EAS149_7:1:100:1003:1410:0:1:1      -    chr22 39994611           U
4  HWI-EAS149_7:1:100:1003:1410:0:1:1      -    chr22 39994613           U
5  HWI-EAS149_7:1:100:1003:1410:0:1:1      +    chr22 39994626           M
6  HWI-EAS149_7:1:100:1003:1410:0:1:1      -    chr22 39994632           U
7   HWI-EAS149_7:1:100:1003:514:0:1:1      +    chr22 17968401           M
8   HWI-EAS149_7:1:100:1003:514:0:1:1      +    chr22 17968428           M
9   HWI-EAS149_7:1:100:1003:934:0:1:0      +    chr22 20423913           M
10  HWI-EAS149_7:1:100:1003:934:0:1:0      +    chr22 20423925           M
        end meth_pattern num_meth total     alpha
1  39994598       MUUUMU        2     6 0.3333333
2  39994607       MUUUMU        2     6 0.3333333
3  39994611       MUUUMU        2     6 0.3333333
4  39994613       MUUUMU        2     6 0.3333333
5  39994626       MUUUMU        2     6 0.3333333
6  39994632       MUUUMU        2     6 0.3333333
7  17968401           MM        2     2 1.0000000
8  17968428           MM        2     2 1.0000000
9  20423913         MMUU        2     4 0.5000000
10 20423925         MMUU        2     4 0.5000000
                                         genom_positions read_start read_end
1  39994598,39994607,39994611,39994613,39994626,39994632   39994598 39994632
2  39994598,39994607,39994611,39994613,39994626,39994632   39994598 39994632
3  39994598,39994607,39994611,39994613,39994626,39994632   39994598 39994632
4  39994598,39994607,39994611,39994613,39994626,39994632   39994598 39994632
5  39994598,39994607,39994611,39994613,39994626,39994632   39994598 39994632
6  39994598,39994607,39994611,39994613,39994626,39994632   39994598 39994632
7                                      17968401,17968428   17968401 17968428
8                                      17968401,17968428   17968401 17968428
9                    20423913,20423925,20423939,20423945   20423913 20423945
10                   20423913,20423925,20423939,20423945   20423913 20423945
        beta coverage mean_alpha
1  0.7826087       23  0.6144928
2  0.6363636       22  0.6060606
3  0.7391304       23  0.6144928
4  0.8260870       23  0.6144928
5  0.3913043       23  0.6144928
6  0.3043478       23  0.6144928
7  0.9326923      104  0.9519231
8  0.9716981      106  0.9528302
9  0.5252525       99  0.3779461
10 0.4747475       99  0.3779461</code></pre>
<pre class="r"><code>mantel_haen_dist_next_cpg &lt;- function(long_data, num_cpg_read = NULL) {
  if(!is.null(num_cpg_read)) {
    format_data &lt;- long_data %&gt;%
        filter(total == num_cpg_read) 
  } else {format_data &lt;- long_data}
  format_data &lt;- format_data %&gt;%
      group_by(read_id) %&gt;%
      mutate(dist_to_next = lead(start) - start,
            meth_pattern = paste0(meth_status, lead(meth_status))) %&gt;%
      ungroup() %&gt;%
      fill(dist_to_next, .direction = &quot;down&quot;) %&gt;%
      distinct(read_id, meth_pattern, dist_to_next) %&gt;%
      filter(!grepl(&quot;A&quot;, .$meth_pattern)) %&gt;%
      group_by(dist_to_next, meth_pattern) %&gt;%
      summarise(count = n()) %&gt;%
      ungroup()
  
  chisq_table &lt;- format_data %&gt;%
      mutate(cpg_1 = substr(meth_pattern, 1, 1),
             cpg_2 = substr(meth_pattern, 2, 2)) 
  # check for missing patterns (it screws up the chi test) and replace with 0
  missing_mm &lt;- chisq_table %&gt;% group_by(dist_to_next) %&gt;% filter(!any(grepl(&quot;MM&quot;, meth_pattern))) %&gt;% distinct(dist_to_next) %&gt;% ungroup() %&gt;% .$dist_to_next
  for(i in 1:length(missing_mm)) {
    chisq_table &lt;- rbind(chisq_table, c(missing_mm[i], &quot;MM&quot;, 0, &quot;M&quot;, &quot;M&quot;))
  }
  missing_mu &lt;- chisq_table %&gt;% group_by(dist_to_next) %&gt;% filter(!any(grepl(&quot;MU&quot;, meth_pattern))) %&gt;% distinct(dist_to_next) %&gt;% ungroup() %&gt;% .$dist_to_next
  for(i in 1:length(missing_mu)) {
    chisq_table &lt;- rbind(chisq_table, c(missing_mu[i], &quot;MU&quot;, 0, &quot;M&quot;, &quot;U&quot;))
  }
  missing_um &lt;- chisq_table %&gt;% group_by(dist_to_next) %&gt;% filter(!any(grepl(&quot;UM&quot;, meth_pattern))) %&gt;% distinct(dist_to_next) %&gt;% ungroup() %&gt;% .$dist_to_next
  for(i in 1:length(missing_um)) {
    chisq_table &lt;- rbind(chisq_table, c(missing_um[i], &quot;UM&quot;, 0, &quot;U&quot;, &quot;M&quot;))
  }
  missing_uu &lt;- chisq_table %&gt;% group_by(dist_to_next) %&gt;% filter(!any(grepl(&quot;UU&quot;, meth_pattern))) %&gt;% distinct(dist_to_next) %&gt;% ungroup() %&gt;% .$dist_to_next
  for(i in 1:length(missing_uu)) {
    chisq_table &lt;- rbind(chisq_table, c(missing_uu[i], &quot;UU&quot;, 0, &quot;U&quot;, &quot;U&quot;))
  }

  chisq_table$count &lt;- as.integer(chisq_table$count)
  chisq_table$dist_to_next &lt;- as.integer(chisq_table$dist_to_next)
  chisq_table &lt;- chisq_table %&gt;% filter(!is.na(dist_to_next))
  
  chisq_table &lt;- chisq_table %&gt;% 
      group_by(dist_to_next) %&gt;%
      reframe(tab = list(xtabs(count ~ cpg_1 + cpg_2, data = cur_data())))
  chisq_table &lt;- chisq_table$tab
  
  names(chisq_table) &lt;- format_data %&gt;%
      mutate(dist_to_next = paste0(&quot;dist_next_&quot;, dist_to_next)) %&gt;%
      .$dist_to_next %&gt;% 
    unique()
  distinct_rows &lt;- format_data$dist_to_next %&gt;% 
    unique() %&gt;%
    length()
  chisq_table &lt;- simplify2array(chisq_table)
  
  # the testing
  try &lt;- capture.output(epiDisplay::mhor(mhtable = chisq_table, graph = FALSE))

  try &lt;- try %&gt;% data.frame() %&gt;% .[4:(distinct_rows + 4),] %&gt;% data.frame()
  colnames(try) &lt;- &quot;all&quot;

  try &lt;- try %&gt;% separate_wider_delim(cols = all, delim = &quot; &quot;, names = c(&quot;type&quot;, &quot;dist&quot;, &quot;numbers&quot;), too_few = &quot;align_start&quot;, too_many = &quot;merge&quot;)

  numeric_lines &lt;- try$numbers
  # get Inf and d separate
  pattern &lt;- &quot;Inf|-Inf|[-+]?\\d+\\.?\\d*(?:[eE][-+]?\\d+)?&quot;
  vals &lt;- regmatches(
    numeric_lines,
    gregexpr(pattern, numeric_lines))

  m &lt;- do.call(rbind, vals)
  m &lt;- as.vector(m)
  names(m) &lt;- rep(c(&quot;OR&quot;, &quot;lower_lim&quot;, &quot;upper_lim&quot;, &quot;p_value&quot;), times = (distinct_rows + 1))
  split_vals &lt;- split(m, names(m))
  df &lt;- cbind.data.frame(split_vals)
  df &lt;- lapply(df, as.double) %&gt;% data.frame()

  # merge
  try &lt;- cbind(try[,1:2], df[,c(2,1,4,3)])
  try &lt;- try %&gt;% mutate(IPD = as.numeric(sub(&quot;.*_&quot;, &quot;&quot;, dist)))
  try
}</code></pre>
<pre class="r"><code>try_2 &lt;- mantel_haen_dist_next_cpg(for_beta, 2)</code></pre>
<pre><code>`summarise()` has grouped output by &#39;dist_to_next&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code>Warning in rbind(deparse.level, ...): number of columns of result, 5, is not a
multiple of vector length 4 of arg 2</code></pre>
<pre><code>Warning in mantel_haen_dist_next_cpg(for_beta, 2): NAs introduced by coercion
Warning in mantel_haen_dist_next_cpg(for_beta, 2): NAs introduced by coercion</code></pre>
<pre><code>Warning: There was 1 warning in `reframe()`.
ℹ In argument: `tab = list(xtabs(count ~ cpg_1 + cpg_2, data = cur_data()))`.
ℹ In group 1: `dist_to_next = 2`.
Caused by warning:
! `cur_data()` was deprecated in dplyr 1.1.0.
ℹ Please use `pick()` instead.</code></pre>
<pre><code>Warning: There was 1 warning in `mutate()`.
ℹ In argument: `IPD = as.numeric(sub(&quot;.*_&quot;, &quot;&quot;, dist))`.
Caused by warning:
! NAs introduced by coercion</code></pre>
</div>
<div id="density-of-cpgs-on-read" class="section level1">
<h1>density of cpgs on read</h1>
<p>density = num_cpgs on read / length of read * 100</p>
<pre class="r"><code>for_beta %&gt;% distinct(read_id, total, read_start, read_end) %&gt;%
  mutate(width = read_end - read_start + 1,
         cpg_density = (total/width)*100) %&gt;%
  ggplot(aes(x = cpg_density)) +
  geom_density()</code></pre>
<p><img src="figure/distAlphaByDistance.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>for_beta %&gt;% distinct(read_id, total, read_start, read_end) %&gt;% 
  mutate(width = read_end - read_start + 1,
         cpg_density = (total/width)*100,
         a = &quot;a&quot;) %&gt;%
  ggplot(aes(x = a, y = cpg_density)) +
  geom_boxplot() +
  geom_violin()</code></pre>
<p><img src="figure/distAlphaByDistance.Rmd/unnamed-chunk-7-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="distribution-of-alpha-by-distance" class="section level1">
<h1>Distribution of alpha by distance</h1>
<pre class="r"><code>alpha_distance &lt;- for_beta %&gt;%
      group_by(read_id) %&gt;%
      mutate(dist_to_next = lead(start) - start,
            meth_pattern = paste0(meth_status, lead(meth_status))) %&gt;%
      ungroup() %&gt;%
      fill(dist_to_next, .direction = &quot;down&quot;) %&gt;%
      distinct(read_id, meth_pattern, dist_to_next, alpha, total) %&gt;%
      filter(!grepl(&quot;A&quot;, .$meth_pattern))
nrow(alpha_distance)</code></pre>
<pre><code>[1] 416830</code></pre>
<pre class="r"><code>alpha_distance[1:10,]</code></pre>
<pre><code># A tibble: 10 × 5
   read_id                            meth_pattern dist_to_next alpha total
   &lt;chr&gt;                              &lt;chr&gt;               &lt;int&gt; &lt;dbl&gt; &lt;int&gt;
 1 HWI-EAS149_7:1:100:1003:1410:0:1:1 MU                      9 0.333     6
 2 HWI-EAS149_7:1:100:1003:1410:0:1:1 UU                      4 0.333     6
 3 HWI-EAS149_7:1:100:1003:1410:0:1:1 UU                      2 0.333     6
 4 HWI-EAS149_7:1:100:1003:1410:0:1:1 UM                     13 0.333     6
 5 HWI-EAS149_7:1:100:1003:1410:0:1:1 MU                      6 0.333     6
 6 HWI-EAS149_7:1:100:1003:514:0:1:1  MM                     27 1         2
 7 HWI-EAS149_7:1:100:1003:934:0:1:0  MM                     12 0.5       4
 8 HWI-EAS149_7:1:100:1003:934:0:1:0  MU                     14 0.5       4
 9 HWI-EAS149_7:1:100:1003:934:0:1:0  UU                      6 0.5       4
10 HWI-EAS149_7:1:100:1011:2006:0:1:1 UU                     28 0         3</code></pre>
<ul>
<li>2 cpgs close to each other 1 or 0, far away = 0.5</li>
<li>so filter to total = 2</li>
<li>alpha is the factor because only 3 options</li>
</ul>
<pre class="r"><code>alpha_distance %&gt;%
  filter(total == 2) %&gt;%
  ggplot(aes(x = as.factor(alpha), y = dist_to_next)) +
  geom_boxplot()</code></pre>
<p><img src="figure/distAlphaByDistance.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>alpha_distance %&gt;%
  filter(total == 2) %&gt;%
  ggplot(aes(x = as.factor(alpha), y = dist_to_next)) +
  geom_violin()</code></pre>
<p><img src="figure/distAlphaByDistance.Rmd/unnamed-chunk-9-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>alpha_distance %&gt;%
  filter(total == 2) %&gt;%
  ggplot(aes(fill = as.factor(alpha), x = dist_to_next)) +
  geom_density(alpha = 0.3)</code></pre>
<p><img src="figure/distAlphaByDistance.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<ul>
<li><p>doesn’t look how I thought</p></li>
<li><p>maybe with cpg features</p></li>
</ul>
</div>
<div id="split-by-in-cgi-or-not" class="section level1">
<h1>Split by in CGI or not</h1>
<pre class="r"><code>annots &lt;- c(&quot;hg38_cpgs&quot;)

annotations &lt;- annotatr::build_annotations(genome = &#39;hg38&#39;, annotations = annots)</code></pre>
<pre><code>Building CpG islands...</code></pre>
<pre><code>Building CpG shores...</code></pre>
<pre><code>Building CpG shelves...</code></pre>
<pre><code>Building inter-CpG-islands...</code></pre>
<pre class="r"><code>seqs &lt;- annotations %&gt;% data.frame() %&gt;% distinct(seqnames) %&gt;% .[1:22,]
annotations &lt;- annotations %&gt;% data.frame() %&gt;% filter(seqnames %in% seqs)</code></pre>
<pre class="r"><code>for_beta &lt;- find_overlaps(as_granges(for_beta), as_granges(annotations)) %&gt;% data.frame() %&gt;%
  mutate(cgi = ifelse(type == &quot;hg38_cpg_islands&quot;, &quot;island&quot;, &quot;not_island&quot;))</code></pre>
<pre class="r"><code>alpha_distance &lt;- for_beta %&gt;%
      group_by(cgi, read_id) %&gt;%
      mutate(dist_to_next = lead(start) - start,
            meth_pattern = paste0(meth_status, lead(meth_status))) %&gt;%
      ungroup() %&gt;%
      fill(dist_to_next, .direction = &quot;down&quot;) %&gt;%
      distinct(read_id, cgi, meth_pattern, dist_to_next, alpha, total) %&gt;%
      filter(!grepl(&quot;A&quot;, .$meth_pattern)) %&gt;%
  .[1:10,]</code></pre>
<pre class="r"><code>alpha_distance %&gt;%
  filter(total == 2) %&gt;%
  ggplot(aes(x = as.factor(alpha), y = dist_to_next, fill = cgi)) +
  geom_boxplot()</code></pre>
<p><img src="figure/distAlphaByDistance.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.1 (2024-06-14)
Platform: x86_64-pc-linux-gnu
Running under: Rocky Linux 9.6 (Blue Onyx)

Matrix products: default
BLAS/LAPACK: FlexiBLAS OPENBLAS-OPENMP;  LAPACK version 3.9.0

locale:
 [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
 [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8   
 [7] LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       

time zone: Australia/Melbourne
tzcode source: system (glibc)

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] ggplot2_3.5.2        dplyr_1.1.4          stringr_1.5.1       
 [4] tidyr_1.3.1          plyranges_1.26.0     GenomicRanges_1.58.0
 [7] GenomeInfoDb_1.42.3  IRanges_2.40.1       S4Vectors_0.44.0    
[10] BiocGenerics_0.52.0  data.table_1.17.8    annotatr_1.32.0     

loaded via a namespace (and not attached):
  [1] DBI_1.2.3                         bitops_1.0-9                     
  [3] rlang_1.1.6                       magrittr_2.0.3                   
  [5] git2r_0.36.2                      matrixStats_1.5.0                
  [7] compiler_4.4.1                    RSQLite_2.4.2                    
  [9] GenomicFeatures_1.58.0            png_0.1-8                        
 [11] vctrs_0.6.5                       reshape2_1.4.4                   
 [13] pkgconfig_2.0.3                   crayon_1.5.3                     
 [15] fastmap_1.2.0                     dbplyr_2.5.0                     
 [17] XVector_0.46.0                    labeling_0.4.3                   
 [19] utf8_1.2.6                        Rsamtools_2.22.0                 
 [21] promises_1.3.3                    rmarkdown_2.29                   
 [23] tzdb_0.5.0                        UCSC.utils_1.2.0                 
 [25] purrr_1.1.0                       bit_4.6.0                        
 [27] BSgenome.Hsapiens.UCSC.hg38_1.4.5 xfun_0.52                        
 [29] zlibbioc_1.52.0                   cachem_1.1.0                     
 [31] jsonlite_2.0.0                    blob_1.2.4                       
 [33] later_1.4.2                       DelayedArray_0.32.0              
 [35] BiocParallel_1.40.2               parallel_4.4.1                   
 [37] R6_2.6.1                          bslib_0.9.0                      
 [39] stringi_1.8.7                     RColorBrewer_1.1-3               
 [41] rtracklayer_1.66.0                jquerylib_0.1.4                  
 [43] Rcpp_1.1.0                        SummarizedExperiment_1.36.0      
 [45] knitr_1.50                        readr_2.1.5                      
 [47] nnet_7.3-19                       splines_4.4.1                    
 [49] httpuv_1.6.16                     Matrix_1.7-0                     
 [51] tidyselect_1.2.1                  rstudioapi_0.17.1                
 [53] abind_1.4-8                       yaml_2.3.10                      
 [55] codetools_0.2-20                  curl_6.4.0                       
 [57] lattice_0.22-6                    tibble_3.3.0                     
 [59] regioneR_1.38.0                   plyr_1.8.9                       
 [61] withr_3.0.2                       Biobase_2.66.0                   
 [63] KEGGREST_1.46.0                   evaluate_1.0.4                   
 [65] foreign_0.8-86                    survival_3.6-4                   
 [67] BiocFileCache_2.14.0              Biostrings_2.74.1                
 [69] pillar_1.11.0                     BiocManager_1.30.26              
 [71] filelock_1.0.3                    MatrixGenerics_1.18.1            
 [73] whisker_0.4.1                     generics_0.1.4                   
 [75] vroom_1.6.5                       rprojroot_2.1.0                  
 [77] RCurl_1.98-1.17                   BiocVersion_3.20.0               
 [79] hms_1.1.3                         scales_1.4.0                     
 [81] glue_1.8.0                        tools_4.4.1                      
 [83] AnnotationHub_3.14.0              BiocIO_1.16.0                    
 [85] BSgenome_1.74.0                   GenomicAlignments_1.42.0         
 [87] fs_1.6.6                          XML_3.99-0.18                    
 [89] grid_4.4.1                        AnnotationDbi_1.68.0             
 [91] GenomeInfoDbData_1.2.13           restfulr_0.0.16                  
 [93] cli_3.6.5                         rappdirs_0.3.3                   
 [95] workflowr_1.7.1                   S4Arrays_1.6.0                   
 [97] epiDisplay_3.5.0.2                gtable_0.3.6                     
 [99] sass_0.4.10                       digest_0.6.37                    
[101] SparseArray_1.6.2                 rjson_0.2.23                     
[103] farver_2.1.2                      memoise_2.0.1                    
[105] htmltools_0.5.8.1                 lifecycle_1.0.4                  
[107] httr_1.4.7                        MASS_7.3-60.2                    
[109] bit64_4.6.0-1                    </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
