<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2025-08-08" />

<title>exploratory_plots_genes</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">3_cf_nano</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/caitlinpage/methyl_nano_cf">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">exploratory_plots_genes</h1>
<h4 class="date">2025-08-08</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-08-15
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>methyl_nano_cf/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250606code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250606)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250606code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250606)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomcaitlinpagemethylnanocftree54baa35c8a0abd3cbbaeb5fc65f456596b03782ftargetblank54baa35a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/caitlinpage/methyl_nano_cf/tree/54baa35c8a0abd3cbbaeb5fc65f456596b03782f" target="_blank">54baa35</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomcaitlinpagemethylnanocftree54baa35c8a0abd3cbbaeb5fc65f456596b03782ftargetblank54baa35a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/caitlinpage/methyl_nano_cf/tree/54baa35c8a0abd3cbbaeb5fc65f456596b03782f" target="_blank">54baa35</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    renv.lock
    Ignored:    renv/

Unstaged changes:
    Modified:   .gitignore
    Modified:   analysis/00_ObtainData.Rmd
    Modified:   analysis/healthyVsLuad.Rmd
    Modified:   analysis/modkitOutputToReadFormat.Rmd
    Modified:   analysis/other_files.Rmd
    Modified:   analysis/testingKnowncfDNAFeatures.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/exploratory_plots_genes.Rmd</code>) and HTML
(<code>docs/exploratory_plots_genes.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/caitlinpage/methyl_nano_cf/blob/54baa35c8a0abd3cbbaeb5fc65f456596b03782f/analysis/exploratory_plots_genes.Rmd" target="_blank">54baa35</a>
</td>
<td>
caitlinpage
</td>
<td>
2025-08-15
</td>
<td>
wflow_publish("analysis/exploratory_plots_genes.Rmd")
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/caitlinpage/methyl_nano_cf/blob/7b05846e09ab3750d9b17d7e8999260664aa1602/analysis/exploratory_plots_genes.Rmd" target="_blank">7b05846</a>
</td>
<td>
caitlinpage
</td>
<td>
2025-08-14
</td>
<td>
moved file
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(plyranges)</code></pre>
<pre><code>Loading required package: BiocGenerics</code></pre>
<pre><code>
Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    IQR, mad, sd, var, xtabs</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    anyDuplicated, aperm, append, as.data.frame, basename, cbind,
    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,
    get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,
    match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,
    Position, rank, rbind, Reduce, rownames, sapply, saveRDS, setdiff,
    table, tapply, union, unique, unsplit, which.max, which.min</code></pre>
<pre><code>Loading required package: IRanges</code></pre>
<pre><code>Loading required package: S4Vectors</code></pre>
<pre><code>Loading required package: stats4</code></pre>
<pre><code>
Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:utils&#39;:

    findMatches</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    expand.grid, I, unname</code></pre>
<pre><code>Loading required package: GenomicRanges</code></pre>
<pre><code>Loading required package: GenomeInfoDb</code></pre>
<pre><code>Warning: replacing previous import &#39;S4Arrays::makeNindexFromArrayViewport&#39; by
&#39;DelayedArray::makeNindexFromArrayViewport&#39; when loading &#39;SummarizedExperiment&#39;</code></pre>
<pre><code>
Attaching package: &#39;plyranges&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:IRanges&#39;:

    slice</code></pre>
<pre><code>The following object is masked from &#39;package:stats&#39;:

    filter</code></pre>
<pre class="r"><code>library(dplyr)</code></pre>
<pre><code>
Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:plyranges&#39;:

    between, n, n_distinct</code></pre>
<pre><code>The following objects are masked from &#39;package:GenomicRanges&#39;:

    intersect, setdiff, union</code></pre>
<pre><code>The following object is masked from &#39;package:GenomeInfoDb&#39;:

    intersect</code></pre>
<pre><code>The following objects are masked from &#39;package:IRanges&#39;:

    collapse, desc, intersect, setdiff, slice, union</code></pre>
<pre><code>The following objects are masked from &#39;package:S4Vectors&#39;:

    first, intersect, rename, setdiff, setequal, union</code></pre>
<pre><code>The following objects are masked from &#39;package:BiocGenerics&#39;:

    combine, intersect, setdiff, union</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    filter, lag</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    intersect, setdiff, setequal, union</code></pre>
<pre class="r"><code>library(ggplot2)

library(&quot;TxDb.Hsapiens.UCSC.hg38.knownGene&quot;)</code></pre>
<pre><code>Loading required package: GenomicFeatures</code></pre>
<pre><code>Loading required package: AnnotationDbi</code></pre>
<pre><code>Loading required package: Biobase</code></pre>
<pre><code>Welcome to Bioconductor

    Vignettes contain introductory material; view with
    &#39;browseVignettes()&#39;. To cite Bioconductor, see
    &#39;citation(&quot;Biobase&quot;)&#39;, and for packages &#39;citation(&quot;pkgname&quot;)&#39;.</code></pre>
<pre><code>
Attaching package: &#39;AnnotationDbi&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:dplyr&#39;:

    select</code></pre>
<pre><code>The following object is masked from &#39;package:plyranges&#39;:

    select</code></pre>
<pre class="r"><code>epiread_format &lt;- readRDS(&quot;/researchers/caitlin.page/cf_nano/r_output/epiread_format.rds&quot;)
epiread_format$chrom &lt;- epiread_format$seqnames
head(epiread_format)</code></pre>
<pre><code>                               read_id chrom alignment_start alignment_end
1 7f095597-d2c5-47a2-93aa-d84b7f9abc94 chr22        10521100      10521455
2 e9abf215-8192-48f4-a50f-ef92241e5de3 chr22        10522473      10522658
3 4203dd9d-9bff-4d14-b046-7716315e9699 chr22        10524645      10524786
4 baf45276-b37b-43fd-8a33-6727de6255fa chr22        10527072      10527210
5 a75ce5a8-42f2-4fe2-b62b-35aba109507d chr22        10529269      10529453
6 b5e822b7-e149-4ae2-8937-fe4d570a274e chr22        10534032      10534204
  read_length num_cg      meth_pattern seqnames    start      end width
1         355      1                 T    chr22 10521100 10521455   356
2         185      4              CCTT    chr22 10522473 10522658   186
3         141      1                 T    chr22 10524645 10524786   142
4         138     17 CCCCCCCCCCCCCCCCC    chr22 10527072 10527210   139
5         184      2                TT    chr22 10529269 10529453   185
6         172      2                TT    chr22 10534032 10534204   173
  index_cg                                                       cpg_positions
1       48                                                                  48
2       53                                                         53,54,55,56
3       68                                                                  68
4      299 299,300,301,302,303,304,305,306,307,308,309,310,311,312,313,314,315
5      374                                                             374,375
6      408                                                             408,409
                                                                                                                                           genom_positions
1                                                                                                                                                 10521180
2                                                                                                                      10522481,10522501,10522591,10522649
3                                                                                                                                                 10524771
4 10527071,10527087,10527090,10527093,10527108,10527114,10527116,10527119,10527130,10527136,10527138,10527153,10527156,10527172,10527189,10527191,10527194
5                                                                                                                                        10529326,10529356
6                                                                                                                                        10534125,10534167
  num_meth total alpha
1        0     1   0.0
2        2     4   0.5
3        0     1   0.0
4       17    17   1.0
5        0     2   0.0
6        0     2   0.0</code></pre>
<div id="where-in-the-genome-are-reads-with-alpha-1-vs-alpha-0"
class="section level1">
<h1>where in the genome are reads with alpha = 1 vs alpha = 0?</h1>
<ul>
<li>promoters, intergenic, exons etc??</li>
</ul>
<pre class="r"><code>genes &lt;- genes(TxDb.Hsapiens.UCSC.hg38.knownGene)</code></pre>
<pre><code>  2162 genes were dropped because they have exons located on both strands
  of the same reference sequence or on more than one reference sequence,
  so cannot be represented by a single genomic range.
  Use &#39;single.strand.genes.only=FALSE&#39; to get all the genes in a
  GRangesList object, or use suppressMessages() to suppress this message.</code></pre>
<pre class="r"><code>genes %&gt;% data.frame() %&gt;% filter(seqnames == &quot;chr22&quot;) %&gt;% nrow()</code></pre>
<pre><code>[1] 680</code></pre>
<pre class="r"><code>#colnames(epiread_format)[2:4] &lt;- c(&quot;seqnames&quot;, &quot;start&quot;, &quot;end&quot;)
epiread_format$read_start &lt;- epiread_format$start
epiread_format$read_end &lt;- epiread_format$end</code></pre>
<ul>
<li>ok for some silly reason I can’t make this granges a data.frame - it
doesn’t like the duplicate row names or something - never had any issues
before</li>
<li>but find_overlaps still works</li>
</ul>
<pre class="r"><code>epiread_and_genes &lt;- plyranges::find_overlaps(genes, plyranges::as_granges(epiread_format)) %&gt;% data.frame()
colnames(epiread_and_genes)[2:5] &lt;- c(&quot;gene_start&quot;, &quot;gene_end&quot;, &quot;gene_width&quot;, &quot;gene_strand&quot;)

epiread_and_genes &lt;- epiread_and_genes[,c(1,19:20,2:18)]</code></pre>
<pre class="r"><code>epiread_and_genes %&gt;% group_by(gene_id) %&gt;% summarise(n=n()) %&gt;%
  ggplot(aes(x = n)) +
  geom_bar() +
  labs(title = &quot;Frequency of a certain number of reads overlapping a gene&quot;,
       x = &quot;Number of reads overlapping a gene&quot;, 
       y = &quot;How many genes have the same number of overlaps&quot;)</code></pre>
<p><img src="figure/exploratory_plots_genes.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>epiread_and_genes %&gt;%
  ggplot(aes(x = gene_width, y = read_length)) +
  geom_point(alpha = 0.3)</code></pre>
<p><img src="figure/exploratory_plots_genes.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>head(epiread_and_genes)</code></pre>
<pre><code>  seqnames alpha read_start gene_start gene_end gene_width gene_strand
1    chr22     1   29333114   29333158 29333267        110           -
2    chr22     1   25104964   25102433 25112692      10260           -
3    chr22     1   25105488   25102433 25112692      10260           -
4    chr22     1   25108649   25102433 25112692      10260           -
5    chr22     0   25109660   25102433 25112692      10260           -
6    chr22     1   25110258   25102433 25112692      10260           -
    gene_id                              read_id chrom alignment_start
1 100113380 fa8c31c6-71d4-4767-a55f-c63cfd2b2796 chr22        29333114
2 100128531 38c9b90a-a4ce-4a6a-8e9a-17851e7f6ae6 chr22        25104964
3 100128531 6299f86a-6767-4a43-b859-4180338d5e5a chr22        25105488
4 100128531 1ac644c9-d328-4f43-a7b3-930c2e9da7ca chr22        25108649
5 100128531 621314b5-60fa-4939-b31a-fbd4056ac2cf chr22        25109660
6 100128531 db1fbbcd-47fe-43bb-ace0-843c25c67dfd chr22        25110258
  alignment_end read_length num_cg meth_pattern index_cg        cpg_positions
1      29333295         181      2           CC   258624        258624,258625
2      25105125         161      3          CCC   207131 207131,207132,207133
3      25105611         123      1            C   207140               207140
4      25109034         385      3          CCC   207175 207175,207176,207177
5      25109995         335      2           TT   207194        207194,207195
6      25110431         173      2           CC   207198        207198,207199
             genom_positions num_meth total
1          29333119,29333217        2     2
2 25104984,25105054,25105114        3     3
3                   25105588        1     1
4 25108841,25108870,25109023        3     3
5          25109897,25109915        0     2
6          25110342,25110373        2     2</code></pre>
<ul>
<li>609 out of 680 genes: 90%</li>
<li>I don’t really like this plot</li>
</ul>
<pre class="r"><code>c(overlap_genes = length(unique(epiread_and_genes$gene_id)), 
  total_genes = length(unique(filter(genes, seqnames == &quot;chr22&quot;)$gene_id)), 
  prop_genes = length(unique(epiread_and_genes$gene_id))/length(unique(filter(genes, seqnames == &quot;chr22&quot;)$gene_id)))</code></pre>
<pre><code>overlap_genes   total_genes    prop_genes 
  609.0000000   680.0000000     0.8955882 </code></pre>
<pre class="r"><code>c(overlap_reads = length(unique(epiread_and_genes$read_id)), 
  total_reads = length(unique(epiread_format$read_id)), 
  prop_reads = length(unique(epiread_and_genes$read_id))/length(unique(epiread_format$read_id)))</code></pre>
<pre><code>overlap_reads   total_reads    prop_reads 
 1.605000e+04  2.666500e+04  6.019126e-01 </code></pre>
</div>
<div id="where-in-the-gene" class="section level1">
<h1>where in the gene?</h1>
<ul>
<li><p>I only have start and end gene coords from the TxDb file</p></li>
<li><p>Let’s fake some promoter coordinates from this</p></li>
<li><p>1kb in front of start, to 500bp after start</p></li>
<li><p>strand dependent: so if strand is negative it’s based around the
end coordinate</p></li>
</ul>
<pre class="r"><code>genes_22 &lt;- genes %&gt;% data.frame() %&gt;% filter(seqnames == &quot;chr22&quot;)

genes_22_promoter &lt;- genes_22 %&gt;% mutate(start = ifelse(strand == &quot;+&quot;, start - 1000, end - 500),
                                         end = ifelse(strand == &quot;+&quot;, start + 1500, end + 1000),
                                         width = end - start + 1)</code></pre>
<pre class="r"><code>epireads_and_promoters_1500 &lt;- plyranges::find_overlaps(plyranges::as_granges(epiread_format), plyranges::as_granges(genes_22_promoter)) %&gt;% data.frame()</code></pre>
<pre class="r"><code>epiread_format %&gt;% mutate(in_promoter = ifelse(read_id %in% epireads_and_promoters_1500$read_id, TRUE, FALSE)) %&gt;%
  ggplot(aes(x = in_promoter, y = alpha, fill = in_promoter)) +
  geom_boxplot() +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/exploratory_plots_genes.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>epiread_format %&gt;% mutate(in_gene = ifelse(read_id %in% epiread_and_genes$read_id, TRUE, FALSE)) %&gt;%
  ggplot(aes(x = in_gene, y = alpha, fill = in_gene)) +
  geom_boxplot() +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/exploratory_plots_genes.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" />
* this is useful * lower methylation in overlap than in gene body - good
b/c we know that gene bodies should be methylated</p>
<pre class="r"><code>epiread_format %&gt;% mutate(where = ifelse(read_id %in% epiread_and_genes$read_id, &quot;Gene&quot;, &quot;Random&quot;),
                          where = ifelse(read_id %in% epireads_and_promoters_1500$read_id, &quot;Promoter&quot;, where)) %&gt;%
  ggplot(aes(x = where, y = alpha, fill = where)) +
  geom_boxplot() +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/exploratory_plots_genes.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.1 (2024-06-14)
Platform: x86_64-pc-linux-gnu
Running under: Red Hat Enterprise Linux 9.5 (Plow)

Matrix products: default
BLAS/LAPACK: FlexiBLAS OPENBLAS-OPENMP;  LAPACK version 3.9.0

locale:
 [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
 [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8   
 [7] LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       

time zone: Australia/Melbourne
tzcode source: system (glibc)

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] TxDb.Hsapiens.UCSC.hg38.knownGene_3.20.0
 [2] GenomicFeatures_1.58.0                  
 [3] AnnotationDbi_1.68.0                    
 [4] Biobase_2.66.0                          
 [5] ggplot2_3.5.2                           
 [6] dplyr_1.1.4                             
 [7] plyranges_1.26.0                        
 [8] GenomicRanges_1.58.0                    
 [9] GenomeInfoDb_1.42.3                     
[10] IRanges_2.40.1                          
[11] S4Vectors_0.44.0                        
[12] BiocGenerics_0.52.0                     
[13] workflowr_1.7.1                         

loaded via a namespace (and not attached):
 [1] tidyselect_1.2.1            blob_1.2.4                 
 [3] farver_2.1.2                Biostrings_2.74.1          
 [5] bitops_1.0-9                fastmap_1.2.0              
 [7] RCurl_1.98-1.17             GenomicAlignments_1.42.0   
 [9] promises_1.3.3              XML_3.99-0.18              
[11] digest_0.6.37               lifecycle_1.0.4            
[13] KEGGREST_1.46.0             processx_3.8.6             
[15] RSQLite_2.4.2               magrittr_2.0.3             
[17] compiler_4.4.1              rlang_1.1.6                
[19] sass_0.4.10                 tools_4.4.1                
[21] yaml_2.3.10                 rtracklayer_1.66.0         
[23] knitr_1.50                  labeling_0.4.3             
[25] S4Arrays_1.6.0              bit_4.6.0                  
[27] curl_6.4.0                  DelayedArray_0.32.0        
[29] RColorBrewer_1.1-3          abind_1.4-8                
[31] BiocParallel_1.40.2         withr_3.0.2                
[33] grid_4.4.1                  git2r_0.36.2               
[35] scales_1.4.0                SummarizedExperiment_1.36.0
[37] cli_3.6.5                   rmarkdown_2.29             
[39] crayon_1.5.3                generics_0.1.4             
[41] rstudioapi_0.17.1           httr_1.4.7                 
[43] rjson_0.2.23                DBI_1.2.3                  
[45] cachem_1.1.0                stringr_1.5.1              
[47] zlibbioc_1.52.0             parallel_4.4.1             
[49] XVector_0.46.0              restfulr_0.0.16            
[51] matrixStats_1.5.0           vctrs_0.6.5                
[53] Matrix_1.7-0                jsonlite_2.0.0             
[55] callr_3.7.6                 bit64_4.6.0-1              
[57] jquerylib_0.1.4             glue_1.8.0                 
[59] codetools_0.2-20            ps_1.9.1                   
[61] stringi_1.8.7               gtable_0.3.6               
[63] later_1.4.2                 BiocIO_1.16.0              
[65] UCSC.utils_1.2.0            tibble_3.3.0               
[67] pillar_1.11.0               htmltools_0.5.8.1          
[69] GenomeInfoDbData_1.2.13     R6_2.6.1                   
[71] rprojroot_2.1.0             evaluate_1.0.4             
[73] lattice_0.22-6              png_0.1-8                  
[75] Rsamtools_2.22.0            memoise_2.0.1              
[77] httpuv_1.6.16               bslib_0.9.0                
[79] Rcpp_1.1.0                  SparseArray_1.6.2          
[81] whisker_0.4.1               xfun_0.52                  
[83] fs_1.6.6                    MatrixGenerics_1.18.1      
[85] getPass_0.2-4               pkgconfig_2.0.3            </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
