---
title: "00_ObtainData"
author: "Caitlin Page"
date: "2025-06-23"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: inline
---

## Nanopore

1. Nanopore cell-free Methylation
Paper: [Detecting cell-of-origin and cancer-specific methylation features of cell-free DNA from Nanopore sequencing](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-022-02710-1#availability-of-data-and-materials)
Data:
Data located:
* [Zenodo](https://zenodo.org/records/6642503)
* [GEO](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE185307)
Zenodo is the method recommended by the authors
Data accessed by:

Samples:
13 files:
(main one - epiread format) hu5.10

Healthy samples
* hu5.10
* hu5.11
* hu5.12
* ispro.bc2
* ispro.bc3
* ispro.bc4
* ispro.bc5

LUAD cancer samples
* ispro.s1
* ispro.bc1
* ispro.bc8
* ispro.bc9
* ispro.bc10
* ispro.bc11

bc8 and bc9 have the lowest tumour fraction of the samples
bc10 and bc11 have the highest tumour fraction



## RRBS
2. Encode RRBS cell-line

[encode rrbs fastqs avail](https://www.encodeproject.org/search/?type=Experiment&control_type%21=%2A&related_series.%40type=ReferenceEpigenome&replicates.library.biosample.donor.organism.scientific_name=Homo+sapiens&status=released&assay_title=RRBS)
* has 13 rrbs data sets

[GM12878 cell line](https://www.encodeproject.org/experiments/ENCSR000DFT/)
* it's a lymphoblastoid cell line
* picked it because it has no cancer or anything else - a few had cancer
* also immune cells - nice and have used before
* also it's blood - which ties in with my cell free analysis

* has 2 technical replicates
* single end this time

* processed fastq with bismark

** USING THIS ONE
ENCFF000LYO
** OOH ITS HG19

2a. [K562 cell line](https://www.encodeproject.org/experiments/ENCSR000DFK/)
* hg19
* erythroleukaemia cell line
* patient with CML - chronic myeloid leukaemia 
* bcr-abl1 positive

## WGBS
3. BSseq pancreatic cancer
Paper: [Epigenomic reprogramming during pancreatic cancer progression links anabolic glucose metabolism to distant metastasis](https://www.nature.com/articles/ng.3753)
[GEO](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE63123)
has 19 samples
* 4 individuals, diff replicates and situations
* processed with hg19

with bams available to download through SRA toolkit

downloading 4 samples via tmux and file transfer server
all from sample A_38
Lg is lung metastasis
Per is peritoneal metastasis

* SRR1646792: Lg rep 1
* SRR1646793: Lg rep 2
* SRR1646794: Per rep 1
* SRR1646795: Per rep 2

* oops there's 2 more for this sample: 2 liver metastasis: SRR3427332, SRR3427334

* also all of these are massive files
2 fastqs each, 93 is 51.8gb per fastq

Steps to process FastQs
[processing method](https://www.nature.com/articles/ng.3753#Sec8)
* 100 base pair seq
* BSmooth for alignment
* Bowtie2 for alignment (says this immediately after saying using BSmooth for alignment)
** BSmooth offers 2 options for alignment - 1 of them uses bowtie (so that's what the paper used)
* and the lambda phage genome (hold on this is all supposed to be human)
* got meth measurement for each CpG
* filtering: mapping qual < 20, nuc base qual at cyt position < 10
* filtering: removed meas from 5' most 10nt of both paired reads
* DMRs: bsseq
* DMRs: >= 3 reads per CpG
* DMRs: small: smooth window 20 CpGs, 1kb used, t stat 4.6, meth diff > 20%
* DMRs: big: smooth window 200 CpGs, 10kbp, t stat 2, meth diff > 10%, length dmr > 5kb


WHAT I DID:
* bismark
[Bismark steps](https://felixkrueger.github.io/Bismark/quick_reference/)
[steps](code/seqFastqToReadMeth.txt)

Z: methylation
z: unmethylated
[source](https://toolshed.g2.bx.psu.edu/repository/display_tool?repository_id=0df2df1072d600d7&tool_config=%2Fsrv%2Ftoolshed-repos%2Fmain%2F000%2Frepo_463%2Fbismark_methylation_extractor.xml&changeset_revision=8c191acde702&render_repository_actions_for=tool_shed)


## maybe use
[roadmap epigenome](https://egg2.wustl.edu/roadmap/web_portal/)
* still need to do some digging to find the files
* the paper data is used in the BSmooth paper
[GEO](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE16256)
[paper](https://www.nature.com/articles/nature08514#accession-codes) 2009

# cancer cell line encyclopedia = come back to this one
https://sites.broadinstitute.org/ccle/datasets
https://depmap.org/portal/data_page/?tab=allData
promoter data 

[raw data](https://www.ncbi.nlm.nih.gov/sra/?term=PRJNA523380+rrbs)
* fastq files

## Identified but unused data sets

# breast cancer
[paper](https://www.nature.com/articles/s41467-021-25661-w#data-availability)

just has a link to download processed methylation values so we'll see
also has fastq upon request

# healthy indiv neutrophils rrbs
Genome-scale DNA methylome and transcriptome profiling of human neutrophils
[paper](https://www.nature.com/articles/sdata201619#Sec12)
[GEO](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE59163)
* potentially
13 samples
* GSE table files
```{r}
#read.table("../data/rrbs/neutrophil_paper_2016/GSE59163_RAW/GSM1429638_X9006_ad3tr65.fastq_bismark.sam_methpcNF2t10.txt.gz")
```
* no column names
* guess: v4 = num meth, v5 = num unmeth, v7 = beta as %
```{r}
(92)/(92+8) * 100
52/(52+3)*100
```
* nope they don't match

# [rrbsdata (bioconductor package)](https://bioconductor.org/packages/release/data/experiment/html/RRBSdata.html)
has 12 samples
this is the paper they got the data from: https://ashpublications.org/blood/article/121/1/178/31062/DNA-methylation-changes-are-a-late-event-in-acute
* data was not available at the read level
