---
title: "00_ObtainData"
author: "Caitlin Page"
date: "2025-06-23"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: inline
---

## Introduction

1. Nanopore cell-free Methylation
Paper: [Detecting cell-of-origin and cancer-specific methylation features of cell-free DNA from Nanopore sequencing](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-022-02710-1#availability-of-data-and-materials)
Data:
Data located:
* [Zenodo](https://zenodo.org/records/6642503)
* [GEO](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE185307)
Zenodo is the method recommended by the authors
Data accessed by:

Samples:
13 files:
(main one - epiread format) hu5.10

Healthy samples
* hu5.10
* hu5.11
* hu5.12
* ispro.bc2
* ispro.bc3
* ispro.bc4
* ispro.bc5

LUAD cancer samples
* ispro.s1
* ispro.bc1
* ispro.bc8
* ispro.bc9
* ispro.bc10
* ispro.bc11

bc8 and bc9 have the lowest tumour fraction of the samples
bc10 and bc11 have the highest tumour fraction
