---
title: "distAlphaByDistance"
author: "caitlinpage"
date: "2026-01-23"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: inline
---

## Introduction
* Building off comethylation mantel-haensz work from [distNextCpg](distNextCpG.html)

GOAL:
Distribution of alpha by distance: 2 cpgs close to each other should be 1 or 0, 2 cpgs far away should be 0.5 
Density of cpgs on read


```{r}
library(annotatr)
library(data.table)
library(plyranges)
library(tidyr)
library(stringr)
library(dplyr)
library(ggplot2)
```

## Data Objects
```{r}
cg_sites <- cbind(data.frame(Biostrings::matchPattern("CG", BSgenome.Hsapiens.UCSC.hg38::BSgenome.Hsapiens.UCSC.hg38[["chr22"]])), seqnames = "chr22") %>% 
  mutate(pos = paste0(seqnames, "-", start)) %>%
  relocate(pos, seqnames)
cg_sites$index <- 1:nrow(cg_sites)
```


```{r}
source("../code/processData.R")
for_beta_all <- readRDS("/researchers/caitlin.page/cf_nano/r_output/rrbs_alpha_beta.rds")
#for_beta <- readRDS("../output/cluster_files/rrbs_alpha_beta.rds")
for_beta <- for_beta_all %>% filter(seqnames == "chr22")
rrbs_for_beta_pat <- pat_for_wgbs(for_beta)

for_beta <- for_beta %>% group_by(start) %>% mutate(coverage = n(), mean_alpha = mean(alpha)) %>% ungroup() %>% data.frame()
just_beta <- for_beta %>% distinct(start, beta, coverage, mean_alpha) %>% .[order(.$start),]

```

```{r}
for_beta[1:10,]
```


```{r}
mantel_haen_dist_next_cpg <- function(long_data, num_cpg_read = NULL) {
  if(!is.null(num_cpg_read)) {
    format_data <- long_data %>%
        filter(total == num_cpg_read) 
  } else {format_data <- long_data}
  format_data <- format_data %>%
      group_by(read_id) %>%
      mutate(dist_to_next = lead(start) - start,
            meth_pattern = paste0(meth_status, lead(meth_status))) %>%
      ungroup() %>%
      fill(dist_to_next, .direction = "down") %>%
      distinct(read_id, meth_pattern, dist_to_next) %>%
      filter(!grepl("A", .$meth_pattern)) %>%
      group_by(dist_to_next, meth_pattern) %>%
      summarise(count = n()) %>%
      ungroup()
  
  chisq_table <- format_data %>%
      mutate(cpg_1 = substr(meth_pattern, 1, 1),
             cpg_2 = substr(meth_pattern, 2, 2)) 
  # check for missing patterns (it screws up the chi test) and replace with 0
  missing_mm <- chisq_table %>% group_by(dist_to_next) %>% filter(!any(grepl("MM", meth_pattern))) %>% distinct(dist_to_next) %>% ungroup() %>% .$dist_to_next
  for(i in 1:length(missing_mm)) {
    chisq_table <- rbind(chisq_table, c(missing_mm[i], "MM", 0, "M", "M"))
  }
  missing_mu <- chisq_table %>% group_by(dist_to_next) %>% filter(!any(grepl("MU", meth_pattern))) %>% distinct(dist_to_next) %>% ungroup() %>% .$dist_to_next
  for(i in 1:length(missing_mu)) {
    chisq_table <- rbind(chisq_table, c(missing_mu[i], "MU", 0, "M", "U"))
  }
  missing_um <- chisq_table %>% group_by(dist_to_next) %>% filter(!any(grepl("UM", meth_pattern))) %>% distinct(dist_to_next) %>% ungroup() %>% .$dist_to_next
  for(i in 1:length(missing_um)) {
    chisq_table <- rbind(chisq_table, c(missing_um[i], "UM", 0, "U", "M"))
  }
  missing_uu <- chisq_table %>% group_by(dist_to_next) %>% filter(!any(grepl("UU", meth_pattern))) %>% distinct(dist_to_next) %>% ungroup() %>% .$dist_to_next
  for(i in 1:length(missing_uu)) {
    chisq_table <- rbind(chisq_table, c(missing_uu[i], "UU", 0, "U", "U"))
  }

  chisq_table$count <- as.integer(chisq_table$count)
  chisq_table$dist_to_next <- as.integer(chisq_table$dist_to_next)
  chisq_table <- chisq_table %>% filter(!is.na(dist_to_next))
  
  chisq_table <- chisq_table %>% 
      group_by(dist_to_next) %>%
      reframe(tab = list(xtabs(count ~ cpg_1 + cpg_2, data = cur_data())))
  chisq_table <- chisq_table$tab
  
  names(chisq_table) <- format_data %>%
      mutate(dist_to_next = paste0("dist_next_", dist_to_next)) %>%
      .$dist_to_next %>% 
    unique()
  distinct_rows <- format_data$dist_to_next %>% 
    unique() %>%
    length()
  chisq_table <- simplify2array(chisq_table)
  
  # the testing
  try <- capture.output(epiDisplay::mhor(mhtable = chisq_table, graph = FALSE))

  try <- try %>% data.frame() %>% .[4:(distinct_rows + 4),] %>% data.frame()
  colnames(try) <- "all"

  try <- try %>% separate_wider_delim(cols = all, delim = " ", names = c("type", "dist", "numbers"), too_few = "align_start", too_many = "merge")

  numeric_lines <- try$numbers
  # get Inf and d separate
  pattern <- "Inf|-Inf|[-+]?\\d+\\.?\\d*(?:[eE][-+]?\\d+)?"
  vals <- regmatches(
    numeric_lines,
    gregexpr(pattern, numeric_lines))

  m <- do.call(rbind, vals)
  m <- as.vector(m)
  names(m) <- rep(c("OR", "lower_lim", "upper_lim", "p_value"), times = (distinct_rows + 1))
  split_vals <- split(m, names(m))
  df <- cbind.data.frame(split_vals)
  df <- lapply(df, as.double) %>% data.frame()

  # merge
  try <- cbind(try[,1:2], df[,c(2,1,4,3)])
  try <- try %>% mutate(IPD = as.numeric(sub(".*_", "", dist)))
  try
}
```

```{r}
try_2 <- mantel_haen_dist_next_cpg(for_beta, 2)
try <- mantel_haen_dist_next_cpg(for_beta)
```

# density of cpgs on read

density = num_cpgs on read / length of read * 100
```{r}
for_beta %>% distinct(read_id, total, read_start, read_end) %>%
  mutate(width = read_end - read_start + 1,
         cpg_density = (total/width)*100) %>%
  ggplot(aes(x = cpg_density)) +
  geom_density()

for_beta %>% distinct(read_id, total, read_start, read_end) %>% 
  mutate(width = read_end - read_start + 1,
         cpg_density = (total/width)*100,
         a = "a") %>%
  ggplot(aes(x = a, y = cpg_density)) +
  geom_boxplot() +
  geom_violin()
```


# Distribution of alpha by distance

```{r}
alpha_distance <- for_beta %>%
      group_by(read_id) %>%
      mutate(dist_to_next = lead(start) - start,
            meth_pattern = paste0(meth_status, lead(meth_status))) %>%
      ungroup() %>%
      fill(dist_to_next, .direction = "down") %>%
      distinct(read_id, meth_pattern, dist_to_next, alpha, total) %>%
      filter(!grepl("A", .$meth_pattern))
nrow(alpha_distance)
alpha_distance[1:10,]
```

* 2 cpgs close to each other 1 or 0, far away = 0.5
* so filter to total = 2
* alpha is the factor because only 3 options

```{r}
alpha_distance %>%
  filter(total == 2) %>%
  ggplot(aes(x = as.factor(alpha), y = dist_to_next)) +
  geom_boxplot()

alpha_distance %>%
  filter(total == 2) %>%
  ggplot(aes(x = as.factor(alpha), y = dist_to_next)) +
  geom_violin()
```

```{r}
alpha_distance %>%
  filter(total == 2) %>%
  ggplot(aes(fill = as.factor(alpha), x = dist_to_next)) +
  geom_density(alpha = 0.3)
```

* doesn't look how I thought

* maybe with cpg features

# Split by in CGI or not

```{r}
annots <- c("hg38_cpgs")

annotations <- annotatr::build_annotations(genome = 'hg38', annotations = annots)
seqs <- annotations %>% data.frame() %>% distinct(seqnames) %>% .[1:22,]
annotations <- annotations %>% data.frame() %>% filter(seqnames %in% seqs)
```

```{r}
for_beta <- find_overlaps(as_granges(for_beta), as_granges(annotations)) %>% data.frame() %>%
  mutate(cgi = ifelse(type == "hg38_cpg_islands", "island", "not_island"))
```

```{r}
alpha_distance <- for_beta %>%
      group_by(cgi, read_id) %>%
      mutate(dist_to_next = lead(start) - start,
            meth_pattern = paste0(meth_status, lead(meth_status))) %>%
      ungroup() %>%
      fill(dist_to_next, .direction = "down") %>%
      distinct(read_id, cgi, meth_pattern, dist_to_next, alpha, total) %>%
      filter(!grepl("A", .$meth_pattern)) 

alpha_distance[1:10,]
```

```{r}
alpha_distance %>%
  filter(total == 2) %>%
  ggplot(aes(x = as.factor(alpha), y = dist_to_next, fill = cgi)) +
  geom_boxplot()
```

```{r}
alpha_distance %>%
  filter(total == 2) %>%
  ggplot(aes(x = as.factor(alpha), y = dist_to_next, fill = cgi)) +
  geom_violin()
```


```{r}
alpha_distance %>%
  filter(total == 2) %>%
  ggplot(aes(fill = as.factor(alpha), x = dist_to_next)) +
  geom_density(alpha = 0.3) +
  facet_wrap(~ cgi)
```

# low, medium, and high
* split the mantel-haensz calculation by the alpha on the read
* Low: 0-0.2 
* Medium: 0.4-0.6
* High: 0.8-1
* Inclusive for all of these

```{r}
alpha_distance %>%
  filter(alpha == 0 | alpha == 0.5 | alpha == 1) %>%
  ggplot(aes(x = as.factor(alpha), y = dist_to_next, fill = cgi)) +
  geom_boxplot()

alpha_distance %>%
  mutate(alpha_level = case_when(alpha <= 0.2 ~ "Low",
                                 alpha > 0.2 & alpha < 0.4 ~ "Low-med",
                                 alpha >= 0.4 & alpha <= 0.6 ~ "Medium",
                                 alpha > 0.6 & alpha < 0.8 ~ "Med-high",
                                 alpha >= 0.8 ~ "High", TRUE ~ "Other"),
         alpha_level = factor(alpha_level, levels = c("Low", "Low-med", "Medium", "Med-high", "High"))) %>%
  ggplot(aes(x = alpha_level, y = dist_to_next, fill = cgi)) +
  geom_boxplot()
```

```{r}
alpha_distance %>%
  filter(alpha == 0 | alpha == 0.5 | alpha == 1) %>%
  ggplot(aes(x = as.factor(alpha), y = dist_to_next)) +
  geom_boxplot()

alpha_distance %>%
  mutate(alpha_level = case_when(alpha <= 0.2 ~ "Low",
                                 alpha > 0.2 & alpha < 0.4 ~ "Low-med",
                                 alpha >= 0.4 & alpha <= 0.6 ~ "Medium",
                                 alpha > 0.6 & alpha < 0.8 ~ "Med-high",
                                 alpha >= 0.8 ~ "High", TRUE ~ "Other"),
         alpha_level = factor(alpha_level, levels = c("Low", "Low-med", "Medium", "Med-high", "High"))) %>%
  ggplot(aes(x = alpha_level, y = dist_to_next)) +
  geom_boxplot()
```


* according to perplexity
* if OR > 1 - positive assoc, OR < 1 negative assoc ~ 1 little/no assoc

```{r}
mantel_low <- mantel_haen_dist_next_cpg(filter(for_beta_all, alpha <= 0.2))
mantel_med <- mantel_haen_dist_next_cpg(filter(for_beta_all, alpha >= 0.4 , alpha <= 0.6))
mantel_high <- mantel_haen_dist_next_cpg(filter(for_beta_all, alpha >= 0.8))
```

```{r}
cbind(mutate(mantel_low, OR_low = OR) %>% .[,c(2,7:8)], OR_med = mantel_med$OR, OR_high = mantel_high$OR) %>%
  ggplot(aes(x = OR_low, y = OR_high)) +
  geom_point(alpha = 0.4)

cbind(mutate(mantel_low, OR_low = OR) %>% .[,c(2,7:8)], OR_med = mantel_med$OR, OR_high = mantel_high$OR) %>%
  ggplot(aes(x = OR_low, y = OR_med)) +
  geom_point(alpha = 0.4)

cbind(mutate(mantel_low, OR_low = OR) %>% .[,c(2,7:8)], OR_med = mantel_med$OR, OR_high = mantel_high$OR) %>%
  ggplot(aes(x = OR_med, y = OR_high)) +
  geom_point(alpha = 0.4)
```
```{r}
cbind(mutate(mantel_low, OR_low = OR) %>% .[,c(2,7:8)], OR_med = mantel_med$OR, OR_high = mantel_high$OR) %>%
  ggplot(aes(x = OR_low, y = OR_high, colour = dist)) +
  geom_point(alpha = 0.4) +
  geom_hline(yintercept = 1, linetype = "dashed") +
  geom_vline(xintercept = 1, linetype = "dashed") +
  scale_x_continuous(limits = c(0,7.5)) +
  scale_y_continuous(limits = c(0,7.5))

cbind(mutate(mantel_low, OR_low = OR) %>% .[,c(2,7:8)], OR_med = mantel_med$OR, OR_high = mantel_high$OR) %>%
  ggplot(aes(x = OR_low, y = OR_med, colour = dist)) +
  geom_point(alpha = 0.4) +
  geom_hline(yintercept = 1, linetype = "dashed") +
  geom_vline(xintercept = 1, linetype = "dashed") +
  scale_x_continuous(limits = c(0,7.5)) +
  scale_y_continuous(limits = c(0,7.5))

cbind(mutate(mantel_low, OR_low = OR) %>% .[,c(2,7:8)], OR_med = mantel_med$OR, OR_high = mantel_high$OR) %>%
  ggplot(aes(x = OR_med, y = OR_high, colour = dist)) +
  geom_point(alpha = 0.4) +
  geom_hline(yintercept = 1, linetype = "dashed") +
  geom_vline(xintercept = 1, linetype = "dashed") +
  scale_x_continuous(limits = c(0,7.5)) +
  scale_y_continuous(limits = c(0,7.5))
```
* overall this is telling us there's not a lot of positive associations
* not sure if the ones close to 0 count as negative associations

```{r}
try[1:5,]
try %>%
  ggplot(aes(x = OR)) +
  geom_density()

try %>%
  ggplot(aes(x = OR)) +
  geom_histogram(bins = 50)
```
```{r}
try %>% 
  mutate(less_than_one = ifelse(OR < 1, TRUE, FALSE)) %>% .[order(.$IPD, decreasing = TRUE),] %>% .[1:6,]
```

```{r}
try %>% 
  mutate(less_than_one = ifelse(OR < 1, TRUE, FALSE)) %>%
  ggplot(aes(x = IPD, y = OR, colour = less_than_one)) +
  geom_point()
```

```{r}
mantel_low %>% 
  mutate(less_than_one = ifelse(OR < 1, TRUE, FALSE)) %>%
  ggplot(aes(x = IPD, y = OR, colour = less_than_one)) +
  geom_point()

mantel_med %>% 
  mutate(less_than_one = ifelse(OR < 1, TRUE, FALSE)) %>%
  ggplot(aes(x = IPD, y = OR, colour = less_than_one)) +
  geom_point()

mantel_high %>% 
  mutate(less_than_one = ifelse(OR < 1, TRUE, FALSE)) %>%
  ggplot(aes(x = IPD, y = OR, colour = less_than_one)) +
  geom_point()
```
* yeah didn't work splitting it by alpha

## Conclusion
** reminder: all analysis has been done on chr22, not representative of whole genome


* Assumption: 
Distribution of alpha by distance: 2 cpgs close to each other should be 1 or 0, 2 cpgs far away should be 0.5 

This does occur
It's more visible when not in a CpG island, but still occurs in other

```{r}
alpha_distance %>%
  filter(alpha == 0 | alpha == 0.5 | alpha == 1) %>%
  ggplot(aes(x = as.factor(alpha), y = dist_to_next, fill = cgi)) +
  geom_boxplot()
```


* Find CpG density on read

density = num_cpgs on read / length of read * 100

* bulk of reads under 25, then spikes at 100
```{r}
for_beta %>% distinct(read_id, total, read_start, read_end) %>%
  mutate(width = read_end - read_start + 1,
         cpg_density = (total/width)*100) %>%
  ggplot(aes(x = cpg_density)) +
  geom_density()
```


