---
title: "mp1_methPattern"
author: "caitlinpage"
date: "2025-09-25"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: inline
---

## Description


## Packages
```{r}
library(plyranges)
library(data.table)
library(tidyr)
library(stringr)
library(reshape2)
library(dplyr)
library(ggplot2)
```

## Data files
```{r}
all_samples <- readRDS("/researchers/caitlin.page/cf_nano/r_output/all_samples.rds")
overlap_bins_reads <- readRDS("/researchers/caitlin.page/cf_nano/r_output/overlap_bins_reads.rds")
cg_sites <- readRDS("/researchers/caitlin.page/cf_nano/r_output/cg_sites.rds")
```


## Analysis copied from (alphaAnalysis.html)


# how long is a piece of string? - what's the longest for the same alpha

```{r}
all_samples %>% distinct(meth_pattern, total, alpha) %>% .[order(.$total, decreasing = TRUE),] %>% mutate(num = 1:n()) %>% head()
```


```{r}
all_samples %>% distinct(meth_pattern, total, alpha) %>% .[order(.$total, decreasing = TRUE),] %>% mutate(num = 1:n()) %>% filter(alpha == 1 | alpha == 0) %>% head()
```
* longest is 70 unmethylated cpgs
* then 68 methylated cpgs
* 7/10 in top 10 are unmethylated strings - interesting

```{r}
all_samples %>% group_by(meth_pattern, total, alpha) %>% summarise(n_times = n()) %>% ungroup() %>% .[order(.$total, decreasing = TRUE),] %>% mutate(num = 1:n()) %>% filter(alpha == 1 | alpha == 0) %>% head()
```
* unsurprisingly - they are rare

# are there any meth patterns that happen in only cancer or only normal?

```{r}
all_samples %>% group_by(type) %>% distinct(meth_pattern) %>% ungroup() %>% group_by(meth_pattern) %>% mutate(n_groups = n()) %>% ungroup() %>% filter(n_groups == 1) %>% group_by(type) %>% summarise(n_unique = n())
```

## what questions do I have?

* The spatial relationship

** so we know methylation is spatially correlated
** but what does that look like on a read and when does that end?
** what about distances between cpgs on the same read? (and there are some very long reads)
** is it that when cpgs are close together - that's when it's the same
** and how do we define close together?

* genomic context
** are the super long stretches repeats/ends of chroms??
** do we see difference in betweens between groups in oncogenes? / and promoters?



## some basic questions

# how many patterns are there in this data set?
```{r}
all_samples %>% distinct(meth_pattern) %>% nrow()
```

* how many patterns for each type?
```{r}
all_samples %>% distinct(type, meth_pattern) %>% group_by(type) %>% summarise(n=n())
```

* how many patterns for each sample?
```{r}
all_samples %>% distinct(sample, meth_pattern) %>% group_by(sample) %>% summarise(n=n())
```


# what is the most common pattern?
```{r}
all_samples %>% group_by(meth_pattern, total) %>% summarise(n = n()) %>% ungroup() %>% mutate(prop = n/sum(n)) %>% .[order(.$n, decreasing = TRUE),] %>% .[1:10,]
```
A: C ~ 18% of all reads, followed by CC ~ 12%, T ~ 8%

* is it different between cancer and normal?
```{r}
all_samples %>% group_by(meth_pattern, total, type) %>% summarise(n = n()) %>% ungroup() %>% group_by(type) %>% mutate(prop = n/sum(n)) %>% .[order(.$n, decreasing = TRUE),] %>% .[1:10,]
```
Top 3 luad: C ~ 18%, CC ~ 12%, T ~ 8%
Top 3 healthy: C ~ 16%, CC ~ 11%, CCC ~ 7%
So not really different

* does it vary much between samples?
```{r}
all_samples %>% group_by(meth_pattern, total, sample, type) %>% summarise(n = n()) %>% ungroup() %>% group_by(sample) %>% mutate(prop = n/sum(n)) %>% .[order(.$n, decreasing = TRUE),] %>% mutate(rank = 1:n()) %>% filter(rank %in% 1:3) %>% .[order(.$rank),]
```
Top is CC, second is CC, third varies slightly

# what about the least common?

```{r}
all_samples %>% group_by(meth_pattern, total) %>% summarise(n = n()) %>% ungroup() %>% mutate(prop = n/sum(n)) %>% .[order(.$n, decreasing = FALSE),] %>% .[1:10,] 
```

```{r}
all_samples %>% group_by(meth_pattern, total) %>% summarise(n = n()) %>% ungroup() %>% filter(n == 1) %>% nrow()
```
* there are > 18k distinct methylation patterns that only occur once - so they are all the least likely

* what about the distribution of those patterns? - for number of cpgs and for alpha?
```{r}
all_samples %>% group_by(meth_pattern, total, alpha) %>% summarise(n = n()) %>% ungroup() %>% filter(n == 1) %>% summary(.$alpha)
```

* differences between types
```{r}
all_samples %>% group_by(meth_pattern, total, type) %>% summarise(n = n()) %>% ungroup() %>% group_by(type) %>% mutate(prop = n/sum(n)) %>% .[order(.$n, decreasing = FALSE),] %>% filter(n == 1) %>% summarise(n_per_type = n())
```
* cancer has more distinct ones

```{r}
all_samples %>% group_by(meth_pattern, total, type) %>% summarise(n = n()) %>% ungroup() %>% group_by(type) %>% mutate(prop = n/sum(n)) %>% .[order(.$n, decreasing = FALSE),] %>% filter(n == 1) %>%
  ggplot(aes(x = type, y = log2(total), colour = type)) +
  geom_boxplot() +
  labs(title = "Distrib of length pattern cancer/normal in meth patterns that only appear once") +
  theme(legend.position = "none")

all_samples %>% group_by(meth_pattern, total, type, alpha) %>% summarise(n = n()) %>% ungroup() %>% group_by(type) %>% mutate(prop = n/sum(n)) %>% .[order(.$n, decreasing = FALSE),] %>% filter(n == 1) %>%
  ggplot(aes(x = type, y = alpha, colour = type)) +
  geom_boxplot() +
  labs(title = "Distrib of alpha cancer/normal in meth patterns that only appear once") +
  theme(legend.position = "none")
```
```{r}
all_samples %>% group_by(meth_pattern, total, type, alpha) %>% summarise(n = n()) %>% ungroup() %>% group_by(type) %>% mutate(prop = n/sum(n)) %>% .[order(.$n, decreasing = FALSE),] %>% filter(n == 1) %>%
  ggplot(aes(x = total, colour = type)) +
  geom_density() +
  labs(title = "Distrib of length pat cancer/normal in meth patterns that only appear once")

all_samples %>% group_by(meth_pattern, total, type, alpha) %>% summarise(n = n()) %>% ungroup() %>% group_by(type) %>% mutate(prop = n/sum(n)) %>% .[order(.$n, decreasing = FALSE),] %>% filter(n == 1) %>%
  ggplot(aes(x = alpha, colour = type)) +
  geom_density() +
  labs(title = "Distrib of alpha cancer/normal in meth patterns that only appear once")
```


# frequency of pattern
* be careful that when I group by type - that means some patterns will show up twice - and any n value is from the meth
```{r}
all_samples %>% group_by(meth_pattern, total, alpha, type) %>% summarise(n_by_type = n()) %>% ungroup() %>% group_by(type) %>% mutate(prop_type = n_by_type/sum(n_by_type)) %>% ungroup() %>% group_by(meth_pattern) %>% mutate(n_overall = sum(n_by_type)) %>% ungroup() %>% mutate(prop_overall = n_overall/sum(n_overall))
```

# change in frequency of pattern?
can I do something cool 
like a proportion change? - this is probs for genom context
but I do have my bins right now
so could see if I can do differential proportion across 2 diff bins?

# most/least common lengths of patterns and groupings





################

## new stuff

* idea is that we could smooth the graph a bit
* shaded narrow and wide - spread of reads
* which is variation in alpha - because can't do that with beta

* what is beta when alpha is at its highest var?
```{r}
alpha_and_beta2
alpha_and_beta2_pat
```
```{r}
for_beta
got_beta <- got_beta %>% filter(seqnames == "chr22")
```

```{r}
got_beta %>% .[order(.$meth_count_pos, decreasing = TRUE),]
got_beta %>% mutate(cov = meth_count_pos + unmeth_count_pos) %>% .[order(.$cov, decreasing = TRUE),]
```

```{r}
got_beta %>%
  ggplot(aes(x = meth_count_pos, y = unmeth_count_pos)) +
  geom_point(alpha = 0.3)
```

```{r}
got_beta %>% filter(start == 12693698)
for_beta %>% filter(start == 12693698)
```
```{r}
for_beta %>% filter(start >= 12693660, start <= 12693800)
got_beta %>% filter(start >= 12693660, start <= 12693800)
for_beta %>% filter(start >= 12693660, start <= 12693800) %>% distinct(read_id, meth_pattern)
```

```{r}
for_beta %>% filter(start >= 12693660, start <= 12693800) %>% group_by(read_id) %>% mutate(read_start = min(start), read_end = max(end), genom_positions = paste0(start, collapse = ",")) %>% ungroup() %>%
  distinct(seqnames, read_start, read_end, genom_positions, alpha) %>%
  ggplot() +
  geom_segment(aes(x = read_start, xend = read_end, y = alpha, yend = alpha), alpha = 0.3)
```
```{r}
got_beta <- data.frame(got_beta)
```

```{r}
for_beta <- for_beta %>% group_by(read_id) %>% mutate(read_start = min(start), read_end = max(end), genom_positions = paste0(start, collapse = ",")) %>% ungroup()
for_beta <- for_beta %>% data.frame() %>% mutate(beta = got_beta[match(.$start, got_beta$start), "beta"])
```
```{r}
rrbs_for_beta_pat <- for_beta %>% 
  mutate(start_pos = sapply(str_split(genom_positions, ","), head, 1),
         end_pos = sapply(str_split(genom_positions, ","), tail, 1)) %>% 
  filter(start == start_pos) %>%
  distinct(read_id, seqnames, start_pos, meth_pattern, alpha, end_pos, beta) %>% 
  group_by(seqnames, start_pos, meth_pattern, alpha, end_pos, beta) %>% 
  summarise(n=n()) %>%
  ungroup()  %>%
  mutate(start_pos = as.integer(start_pos),
         end_pos = as.integer(end_pos))
```
* pat can help us find positions with interesting groups?
```{r}
rrbs_for_beta_pat %>% group_by(start_pos) %>% summarise(n_groups = n(), sum_reads = sum(n)) %>% .[order(.$n_groups, decreasing = TRUE),]
```

```{r}
rrbs_for_beta_pat %>% filter(start_pos == 43411502)
```

```{r}
rrbs_for_beta_pat %>% filter(start_pos >= 12693660, end_pos <= 12693800) %>% 
  ggplot() +
  geom_segment(aes(x = start_pos, xend = end_pos, y = alpha, yend = alpha), alpha = 0.3) +
  geom_line(aes(x = start_pos, y = beta), colour = "red")
```
```{r}
rrbs_region <- for_beta %>% filter(start>= 12693660, end <= 12693800) %>% .$read_id %>% unique()
```

```{r}
for_beta %>% filter(read_id %in% rrbs_region[1:15]) %>% distinct(read_id, read_start, read_end, alpha) %>%
  ggplot() +
  geom_segment(aes(x = read_start, xend = read_end, y = alpha, yend = alpha), alpha = 0.2) +
  geom_point(data = filter(for_beta, read_id %in% rrbs_region[1:15]), 
             aes(x = start, y = alpha, colour = meth_status)) +
  geom_point(data = distinct(filter(for_beta, read_id %in% rrbs_region[1:15]), start, beta), 
            aes(x = start, y = beta), shape = 4, size = 3) +
  geom_segment(data = filter(cg_sites, start >= min(filter(for_beta, read_id %in% rrbs_region[1:15])$start), end <= max(filter(for_beta, read_id %in% rrbs_region[1:15])$end)), aes(x = start, xend = start, y = -0.15, yend = -0.05), colour = "green")
```

```{r}
for_beta %>% filter(read_id %in% rrbs_region) %>% distinct(read_id, read_start, read_end, alpha) %>%
  ggplot() +
  geom_segment(aes(x = read_start, xend = read_end, y = alpha, yend = alpha), alpha = 0.2) +
  geom_point(data = filter(for_beta, read_id %in% rrbs_region), 
             aes(x = start, y = alpha, colour = meth_status)) +
  geom_point(data = distinct(filter(for_beta, read_id %in% rrbs_region), start, beta), 
            aes(x = start, y = beta), shape = 4, size = 3) +
  geom_segment(data = filter(cg_sites, start >= min(filter(for_beta, read_id %in% rrbs_region)$start), end <= max(filter(for_beta, read_id %in% rrbs_region)$end)), aes(x = start, xend = start, y = -0.15, yend = -0.05), colour = "green")
```
```{r}
for_beta %>% filter(read_id %in% rrbs_region) %>% distinct(read_id, read_start, read_end, alpha) %>%
  ggplot() +
  geom_segment(aes(x = read_start, xend = read_end, y = alpha, yend = alpha), alpha = 0.2) +
  geom_count(data = filter(for_beta, read_id %in% rrbs_region), 
             aes(x = start, y = alpha, colour = meth_status)) +
  geom_point(data = distinct(filter(for_beta, read_id %in% rrbs_region), start, beta), 
            aes(x = start, y = beta), shape = 4, size = 3) +
  geom_segment(data = filter(cg_sites, start >= min(filter(for_beta, read_id %in% rrbs_region)$start), end <= max(filter(for_beta, read_id %in% rrbs_region)$end)), aes(x = start, xend = start, y = -0.15, yend = -0.05), colour = "green")
```

```{r}
for_beta %>% filter(read_id %in% rrbs_region) %>% distinct(read_id, start, alpha, beta) %>%
  ggplot(aes(x = as.factor(start), y = alpha)) +
  geom_violin(scale = "count")
?geom_boxplot
?geom_violin
```
```{r}
for_beta %>% filter(read_id %in% rrbs_region) %>% distinct(read_id, start, alpha, beta) %>%
  ggplot(aes(x = as.factor(beta), y = alpha)) +
  geom_boxplot()
```

```{r}
for_beta %>% distinct(read_id, start, alpha, beta) %>% .[order(.$start),] %>% .[300:800,] %>%
  ggplot(aes(x = as.factor(beta), y = alpha)) +
  geom_boxplot()
```
```{r}
for_beta %>% filter(beta == 0.5) %>%
  ggplot(aes(x = as.factor(beta), y = alpha)) +
  geom_violin()
```

```{r}
for_beta %>%
  ggplot(aes(x = beta)) +
  geom_density()
```
```{r}
rrbs_for_beta_pat %>% filter(beta == 0.5) %>% group_by(start_pos) %>% mutate(num_pat = n())
```
```{r}
rrbs_for_beta_pat %>% filter(start == 48646638)
```

```{r}
rrbs_for_beta_pat %>% group_by(start_pos) %>% summarise(sum = sum(n)) %>% .[order(.$sum, decreasing = TRUE),]
```

```{r}
rrbs_plot <- function(long_data, read_ids, cpgs = cg_sites) {
  long_data %>% filter(read_id %in% read_ids) %>% 
    distinct(read_id, read_start, read_end, alpha) %>%
  ggplot() +
  geom_segment(aes(x = read_start, xend = read_end, y = alpha, yend = alpha), alpha = 0.2) +
  geom_point(data = filter(long_data, read_id %in% read_ids), 
             aes(x = start, y = alpha, colour = meth_status)) +
  geom_point(data = distinct(filter(long_data, read_id %in% read_ids), start, beta), 
            aes(x = start, y = beta), shape = 4, size = 3) +
  geom_segment(data = filter(cg_sites, start >= min(filter(long_data, read_id %in% read_ids)$start), 
                             end <= max(filter(long_data, read_id %in% read_ids)$end)), 
               aes(x = start, xend = start, y = -0.15, yend = -0.05), colour = "green")
}
```

```{r}
rrbs_plot(for_beta, rrbs_region)
```

```{r}
rrbs_for_beta_pat
got_beta %>% .[order(.$meth_count_pos, decreasing = TRUE),]
```

```{r}
rrbs_for_beta_pat %>% filter(start_pos >= 20918800, end_pos <= 20919100)
```

```{r}
quartiles <- filter(for_beta, start >= 20918800, end <= 20919100) %>% summarise(q1 = quantile(alpha, 0.25), q3 = quantile(alpha, 0.75))
```

```{r}
rrbs_plot(for_beta, unique(filter(for_beta, start >= 20918800, end <= 20919100)$read_id)) +
  geom_ribbon(data = quartiles, aes(x = 20918800, ymin = q1, ymax = q3))
```

```{r}
filter(for_beta, start >= 20918800, end <= 20919100) %>% summary(.$alpha)
```

