---
title: "researchSprint"
output: html_document
date: "2025-08-21"
---

## Description
* 2 day research sprint on long read data from the 21-22 August 2025
* Team leader: Dineika
* Interests: fragmentomics, methylation, deconvolution, [slide](https://docs.google.com/presentation/d/117TR8EFgIXXJawOTsnHjjHFak8TveyES_2RBhb3LtaQ/edit?slide=id.g3763afce800_1_316#slide=id.g3763afce800_1_316)
* Team members: Dineika, Max, Jovana, Gunjan, Caitlin, Therese, Song, Kevin
* Team name: Goldilocks

## What was done
* beta and pat files from wgbs_tools (Max)
* tumour fraction estimates - ichorCNA (Song)
* Fragle (fragment based deconv - short read designed) (Dineika)
* end motif analysis (Gunjan)
* processing data, variance analysis, genome binning (Jovana)
* correlate tumour fraction and proportion of meth/unmeth sites at luad genes? (Therese - I helped)

## what did I do?
* some basic plots
* binning the genome
* plots of luad genes
* cpg islands, gene promoters plots
* tried and failed to make beta files using the alpha, then made again from bedgraphs

## summary of sample information
* bc10 and bc11 have the highest tumour fraction
* bc8 and bc9 have the lowest tumour fraction - these 2 are grouped together as low - everything else is high

# my analysis code

```{r}
library(biomaRt)
library(EnsDb.Hsapiens.v86)

library(plyranges)
library(tidyr)
library(ggplot2)
library(BSgenome.Hsapiens.UCSC.hg38)
library(stringr)
library(data.table)
library(dplyr)
```

```{r}
healthy_epireads <- readRDS("/researchers/caitlin.page/cf_nano/r_output/fixed_healthy_epireads.rds")
luad_epireads <- readRDS("/researchers/caitlin.page/cf_nano/r_output/fixed_luad_epireads.rds")
```


* cpg and genom positions on luad samples were wrong 
  * was a bug in the for loop
  * has now been fixed

```{r}
cg_sites <- cbind(data.frame(Biostrings::matchPattern("CG", BSgenome.Hsapiens.UCSC.hg38[["chr22"]])), seqnames = "chr22") %>%
  mutate(pos = paste0(seqnames, "-", start)) %>%
  relocate(pos, seqnames)
cg_sites$index <- 1:nrow(cg_sites)
```


# read lengths between cancer and normal

```{r}
columns <- c("read_id", "seqnames", "start", "end", "read_length", "type", "sample")
```
```{r}
names(healthy_epireads)
names(luad_epireads)
```

```{r}
read_lengths <- rbind(healthy_epireads$hu5.10 %>% mutate(type = "healthy", sample = "hu5.10") %>% .[,columns], 
      healthy_epireads$hu5.11 %>% mutate(type = "healthy", sample = "hu5.11") %>% .[,columns],
      healthy_epireads$hu5.12 %>% mutate(type = "healthy", sample = "hu5.12") %>% .[,columns],
      healthy_epireads$ispro.bc2 %>% mutate(type = "healthy", sample = "ispro.bc2") %>% .[,columns],
      healthy_epireads$ispro.bc3 %>% mutate(type = "healthy", sample = "ispro.bc3") %>% .[,columns],
      healthy_epireads$ispro.bc4 %>% mutate(type = "healthy", sample = "ispro.bc4") %>% .[,columns],
      healthy_epireads$ispro.bc5 %>% mutate(type = "healthy", sample = "ispro.bc5") %>% .[,columns],
      luad_epireads$ispro.bc1 %>% mutate(type = "luad", sample = "ispro.bc1") %>% .[,columns],
      luad_epireads$ispro.bc10 %>% mutate(type = "luad", sample = "ispro.bc10") %>% .[,columns],
      luad_epireads$ispro.bc11 %>% mutate(type = "luad", sample = "ispro.bc11") %>% .[,columns],
      luad_epireads$ispro.bc8 %>% mutate(type = "luad", sample = "ispro.bc8") %>% .[,columns],
      luad_epireads$ispro.bc9 %>% mutate(type = "luad", sample = "ispro.bc9") %>% .[,columns],
      luad_epireads$ispro.s1 %>% mutate(type = "luad", sample = "ispro.s1") %>% .[,columns])

head(read_lengths)
```
```{r}
read_lengths %>% group_by(type) %>% summarise(n())
```
```{r}
summary(read_lengths$read_length)
```

```{r}
read_lengths %>%
  ggplot(aes(x = read_length, colour = type)) +
  geom_density(linewidth = 1.5) +
  scale_x_continuous(limits = c(0,500)) +
  theme_classic()
```


```{r}
read_lengths %>%
  ggplot(aes(x = log2(read_length), colour = type)) +
  geom_density()

read_lengths %>%
  ggplot(aes(x = log2(read_length), colour = sample)) +
  geom_density() +
  facet_wrap(~type)
```

```{r}
read_lengths %>%
  ggplot(aes(x = read_length, colour = type)) +
  geom_density() +
  scale_x_continuous(limits = c(0,500))
```
* dineika and max expected to see bigger shift to smaller read lengths in luad but nice to see it in the dinuc
* also if you cut with cartesian coordinates

# plot alpha across a region
bc11

```{r}
healthy_epireads$hu5.10[110:125,]
```

```{r}
healthy_epireads$hu5.10 %>% filter(start >= 10718212, end <= 10721909) %>%
  ggplot(aes(x = start, y = alpha, colour = read_id)) +
  geom_segment(aes(x = start, xend = end, y = alpha, yend = alpha)) +
  theme(legend.position = "none")
```


```{r}
luad_epireads$ispro.bc11 %>% filter(start >= 10718212, end <= 10721909) %>%
  ggplot(aes(x = start, y = alpha, colour = read_id)) +
  geom_segment(aes(x = start, xend = end, y = alpha, yend = alpha)) +
  theme(legend.position = "none")
```
* nice that there's a difference between the cancer and healthy
* bc11 has highest tumour purity


# maybe beta?
* I don't really know what I should be doing...
* I think dineika needs team meeting let's set some goals
* I don't want to try and run deconvolution stuff
* other people can haggle with scripts and then I just take their instructions and run with it
* which does kinda leave the more exploratory stuff again
* or I could bin the genome - though if jovana was going to do that?



# let's just combine all the samples but have more columns

```{r}
healthy_epireads[3]
```


```{r}
columns <- c("read_id", "seqnames", "start", "end", "read_length", "type", "sample", "index_cg", "meth_pattern", "num_meth", "total", "alpha")

all_samples <- rbind(healthy_epireads$hu5.10 %>% mutate(type = "healthy", sample = "hu5.10") %>% .[,columns], 
      healthy_epireads$hu5.11 %>% mutate(type = "healthy", sample = "hu5.11") %>% .[,columns],
      healthy_epireads$hu5.12 %>% mutate(type = "healthy", sample = "hu5.12") %>% .[,columns],
      healthy_epireads$ispro.bc2 %>% mutate(type = "healthy", sample = "ispro.bc2") %>% .[,columns],
      healthy_epireads$ispro.bc3 %>% mutate(type = "healthy", sample = "ispro.bc3") %>% .[,columns],
      healthy_epireads$ispro.bc4 %>% mutate(type = "healthy", sample = "ispro.bc4") %>% .[,columns],
      healthy_epireads$ispro.bc5 %>% mutate(type = "healthy", sample = "ispro.bc5") %>% .[,columns],
      luad_epireads$ispro.bc1 %>% mutate(type = "luad", sample = "ispro.bc1") %>% .[,columns],
      luad_epireads$ispro.bc10 %>% mutate(type = "luad", sample = "ispro.bc10") %>% .[,columns],
      luad_epireads$ispro.bc11 %>% mutate(type = "luad", sample = "ispro.bc11") %>% .[,columns],
      luad_epireads$ispro.bc8 %>% mutate(type = "luad", sample = "ispro.bc8") %>% .[,columns],
      luad_epireads$ispro.bc9 %>% mutate(type = "luad", sample = "ispro.bc9") %>% .[,columns],
      luad_epireads$ispro.s1 %>% mutate(type = "luad", sample = "ispro.s1") %>% .[,columns])

head(all_samples)
```


```{r}
all_samples %>%
  ggplot(aes(x = alpha, colour = type)) +
  geom_density(linewidth = 1.5) +
  theme_classic()
```

```{r}
all_samples %>%
  ggplot(aes(x = alpha, colour = sample)) +
  geom_density() +
  facet_wrap(~ type)
```

```{r}
all_samples %>% filter(type == "healthy") %>%
  ggplot(aes(x = alpha)) +
  geom_density() +
  labs(title = "All healthy")
```


```{r}
all_samples %>% filter(sample %in% c("ispro.bc10", "ispro.bc11") | type == "healthy") %>%
  ggplot(aes(x = alpha, colour = sample)) +
  geom_density() +
  facet_wrap(~ type)
```
* good we see less methylation of 1 in the cancer samples


* redo cpg and genom positions - but it's b/c I want beta
* so maybe I don't need it to get beta


# can I code a way to find out how many reads overlap??
* perplexity

```{r}
dt <- as.data.table(all_samples)

# Create all possible pairs; exclude self-pairs (i != j)
setkey(dt, start, end)
overlaps <- foverlaps(dt[, .(row_id = .I, start, end)], dt[, .(row_id = .I, start, end)],
                      by.x = c("start", "end"), by.y = c("start", "end"), nomatch = 0L)
overlaps <- overlaps[row_id != i.row_id]

# Get unique rows with overlaps
unique_overlap_rows <- unique(c(overlaps$row_id, overlaps$i.row_id))
num_overlap_rows <- length(unique_overlap_rows)
print(num_overlap_rows)
```

```{r}
head(unique_overlap_rows)
```
* ok so most of the rows overlap - but if I want a dense region or find matching positions I need binning
* which is a tomorrow problem - Max is trying binning now




# get beta vals

```{r}
beta_format <- all_samples %>% separate_longer_position(cols = meth_pattern, width = 1) %>% 
  group_by(read_id) %>% 
  mutate(cpg_pos = index_cg:c(index_cg + n() -1)) %>% ungroup() %>%
  mutate(genom_pos = cg_sites[match(.$cpg_pos, cg_sites$index), "start"])
```
* can't run on everything
* good news is ok that cpg pos and genome pos are wrong because I redid them from the cg site - all I need is first index
```{r}
beta_format <- healthy_epireads$hu5.10 %>% separate_longer_position(cols = meth_pattern, width = 1) %>% 
  group_by(read_id) %>% 
  mutate(cpg_pos = index_cg:c(index_cg + n() -1)) %>% ungroup() %>%
  mutate(genom_pos = cg_sites[match(.$cpg_pos, cg_sites$index), "start"])
```


* oh I'll have to modify this because I have combined all the samples
* i think it's ok I just group by sample
* but also then how do I do that for the columns 
* can't visualise how the data looks right now :(
```{r}
beta_format <- beta_format %>% group_by(cpg_pos) %>%
  mutate(site_pattern = paste(meth_pattern, collapse = "")) %>% ungroup() %>% distinct(cpg_pos, genom_pos, site_pattern) %>%
  mutate(num_meth = str_count(.$site_pattern, "C"), num_unmeth = str_count(.$site_pattern, "T"))
beta_format <- data.frame(beta_format)
head(beta_format)
```

```{r}
beta_vals <- cg_sites
beta_vals <- beta_vals %>% mutate(he_hu5.10_meth = beta_format[match(.$index, beta_format$cpg_pos), "num_meth"],
                     he_hu5.10_unmeth = beta_format[match(.$index, beta_format$cpg_pos), "num_unmeth"])
head(beta_vals)
```

* let's make a loop so it repeats for the other samples
```{r}
length(healthy_epireads)
```

```{r}
names(healthy_epireads)
```

```{r}
try <- beta_vals
try[,paste0("he_", names(healthy_epireads)[2], "_meth")] <- 5
head(try)
```


```{r}
for(i in 2:length(healthy_epireads)) {
  beta_format <- healthy_epireads[[i]]
  beta_format <-  beta_format %>% separate_longer_position(cols = meth_pattern, width = 1) %>% 
    group_by(read_id) %>% 
    mutate(cpg_pos = index_cg:c(index_cg + n() -1)) %>% ungroup() %>%
    mutate(genom_pos = cg_sites[match(.$cpg_pos, cg_sites$index), "start"])
  beta_format <- beta_format %>% group_by(cpg_pos) %>%
    mutate(site_pattern = paste(meth_pattern, collapse = "")) %>% ungroup() %>% distinct(cpg_pos, genom_pos, site_pattern) %>%
    mutate(num_meth = str_count(.$site_pattern, "C"), num_unmeth = str_count(.$site_pattern, "T"))
  beta_format <- data.frame(beta_format)
  beta_vals[,paste0("he_", names(healthy_epireads)[i], "_meth")] <- beta_format[match(beta_vals$index, beta_format$cpg_pos), "num_meth"]
  beta_vals[,paste0("he_", names(healthy_epireads)[i], "_unmeth")] <- beta_format[match(beta_vals$index, beta_format$cpg_pos), "num_meth"]
}
head(beta_vals)
```

```{r}
names(luad_epireads)
```

```{r}
for(i in 1:length(luad_epireads)) {
  beta_format <- luad_epireads[[i]]
  beta_format <-  beta_format %>% separate_longer_position(cols = meth_pattern, width = 1) %>% 
    group_by(read_id) %>% 
    mutate(cpg_pos = index_cg:c(index_cg + n() -1)) %>% ungroup() %>%
    mutate(genom_pos = cg_sites[match(.$cpg_pos, cg_sites$index), "start"])
  beta_format <- beta_format %>% group_by(cpg_pos) %>%
    mutate(site_pattern = paste(meth_pattern, collapse = "")) %>% ungroup() %>% distinct(cpg_pos, genom_pos, site_pattern) %>%
    mutate(num_meth = str_count(.$site_pattern, "C"), num_unmeth = str_count(.$site_pattern, "T"))
  beta_format <- data.frame(beta_format)
  beta_vals[,paste0("luad_", names(luad_epireads)[i], "_meth")] <- beta_format[match(beta_vals$index, beta_format$cpg_pos), "num_meth"]
  beta_vals[,paste0("luad_", names(luad_epireads)[i], "_unmeth")] <- beta_format[match(beta_vals$index, beta_format$cpg_pos), "num_meth"]
}
head(beta_vals)
```


```{r}
beta_vals[is.na(beta_vals)] <- 0
```

```{r}
beta_vals <- beta_vals %>% 
  mutate(he_hu5.10_beta = he_hu5.10_meth/(he_hu5.10_meth + he_hu5.10_unmeth),
         he_hu5.11_beta = he_hu5.11_meth/(he_hu5.11_meth + he_hu5.11_unmeth),
         he_hu5.12_beta = he_hu5.12_meth/(he_hu5.12_meth + he_hu5.12_unmeth),
         he_ispro.bc2_beta = he_ispro.bc2_meth/(he_ispro.bc2_meth + he_ispro.bc2_unmeth),
         he_ispro.bc3_beta = he_ispro.bc3_meth/(he_ispro.bc3_meth + he_ispro.bc3_unmeth),
         he_ispro.bc4_beta = he_ispro.bc4_meth/(he_ispro.bc4_meth + he_ispro.bc4_unmeth),
         he_ispro.bc5_beta = he_ispro.bc5_meth/(he_ispro.bc5_meth + he_ispro.bc5_unmeth),
         luad_ispro.bc1_beta = luad_ispro.bc1_meth/(luad_ispro.bc1_meth + luad_ispro.bc1_unmeth),
         luad_ispro.bc10_beta = luad_ispro.bc10_meth/(luad_ispro.bc10_meth + luad_ispro.bc10_unmeth),
         luad_ispro.bc11_beta = luad_ispro.bc11_meth/(luad_ispro.bc11_meth + luad_ispro.bc11_unmeth),
         luad_ispro.bc8_beta = luad_ispro.bc8_meth/(luad_ispro.bc8_meth + luad_ispro.bc8_unmeth),
         luad_ispro.bc9_beta = luad_ispro.bc9_meth/(luad_ispro.bc9_meth + luad_ispro.bc9_unmeth),
         luad_ispro.s1_beta = luad_ispro.s1_meth/(luad_ispro.s1_meth + luad_ispro.s1_unmeth))

beta_vals[is.na(beta_vals)] <- 0
head(beta_vals)
```

* how do I plot this?
* I can't do the same with my all alpha because that was one column :(
* wrong format

```{r}
beta_vals_long <- beta_vals
columns <- c("pos", "seqnames", "start", "end", "index", "type", "sample", "beta")

beta_vals_long <- rbind(beta_vals %>% mutate(type = "healthy", sample = "hu5.10", beta = he_hu5.10_beta) %>% .[,columns], 
      beta_vals %>% mutate(type = "healthy", sample = "hu5.11", beta = he_hu5.11_beta) %>% .[,columns],
      beta_vals %>% mutate(type = "healthy", sample = "hu5.12", beta = he_hu5.12_beta) %>% .[,columns],
      beta_vals %>% mutate(type = "healthy", sample = "ispro.bc2", beta = he_ispro.bc2_beta) %>% .[,columns],
      beta_vals %>% mutate(type = "healthy", sample = "ispro.bc3", beta = he_ispro.bc3_beta) %>% .[,columns],
      beta_vals %>% mutate(type = "healthy", sample = "ispro.bc4", beta = he_ispro.bc4_beta) %>% .[,columns],
      beta_vals %>% mutate(type = "healthy", sample = "ispro.bc5", beta = he_ispro.bc5_beta) %>% .[,columns],
      beta_vals %>% mutate(type = "luad", sample = "ispro.bc1", beta = luad_ispro.bc1_beta) %>% .[,columns],
      beta_vals %>% mutate(type = "luad", sample = "ispro.bc10", beta = luad_ispro.bc10_beta) %>% .[,columns],
      beta_vals %>% mutate(type = "luad", sample = "ispro.bc11", beta = luad_ispro.bc11_beta) %>% .[,columns],
      beta_vals %>% mutate(type = "luad", sample = "ispro.bc8", beta = luad_ispro.bc8_beta) %>% .[,columns],
      beta_vals %>% mutate(type = "luad", sample = "ispro.bc9", beta = luad_ispro.bc9_beta) %>% .[,columns],
      beta_vals %>% mutate(type = "luad", sample = "ispro.s1", beta = luad_ispro.s1_beta) %>% .[,columns])
head(beta_vals_long)
```

```{r}
beta_vals_long %>%
  ggplot(aes(x = beta, colour = sample)) +
  geom_density() +
  facet_wrap(~ type)
```


```{r}
beta_vals_long %>%
  ggplot(aes(x = beta, colour = type)) +
  geom_density()
```
* looks suspicous because it's supposed to be high at 0 and 1, low at 0.5
* and got nothing at 1

* should have just used the bed files for this
```{r}
list.files("/researchers/caitlin.page/cf_nano/megalodonBedgraphs")
```


```{r}
he_hu5.10_bed <- read_bed("/researchers/caitlin.page/cf_nano/megalodonBedgraphs/HU005.10.meg242.remora1.edgefix.modbam2bed.5mC.cut0.667.hg38.sorted.bedgraph.gz")
he_hu5.10_bed %>% data.frame() %>% filter(seqnames == "chr22")
```

```{r}
beta_vals[,c(1:4,7:9,34)] %>% filter(start == 10522481)
```
* it does match that's good
```{r}
he_hu5.10_bed <- he_hu5.10_bed %>% data.frame() %>% filter(seqnames == "chr22")
he_hu5.10_bed$name <- as.double(he_hu5.10_bed$name)
```

```{r}
he_hu5.10_bed %>% 
  ggplot(aes(x = name)) +
  geom_density()
```
* cool yeah so I did betas wrong

* let's just try and quickly fix the betas:

```{r}
bedgraphs <- list.files("/researchers/caitlin.page/cf_nano/megalodonBedgraphs")
bedgraphs <- bedgraphs[grep("bedgraph", bedgraphs)]
bedgraphs <- bedgraphs[grep(".tbi", bedgraphs, invert = TRUE)]
head(bedgraphs)
```
```{r}
str_extract(bedgraphs, "^[^m]+")
```

```{r}
he_hu5.10_bed$sample <- "HU005.10."

bed_betas <- he_hu5.10_bed
```

```{r}
for(i in 2:length(bedgraphs)) {
  bed <- read_bed(paste0("/researchers/caitlin.page/cf_nano/megalodonBedgraphs/", bedgraphs[i]))
  bed <- bed %>% data.frame() %>% filter(seqnames == "chr22")
  bed$name <- as.double(bed$name)
  bed$sample <- str_extract(bedgraphs, "^[^m]+")[i]
  bed_betas <- rbind(bed_betas, bed)
}
head(bed_betas)
```

```{r}
bed_betas %>%
  ggplot(aes(x = name)) +
  geom_density()

bed_betas %>%
  ggplot(aes(x = name, colour = sample)) +
  geom_density()
```

* yeah that's right
* I don't have numbers for it though so that's a bit annoying



# can I add some kind of alpha to this? - meeting last week talk
- or just add the beta back to the alpha values file
- which would mean having the correct files
* made them and asked andrew but he hasn't uploaded them yet :(


Analysis from the research sprint is continued [here](researchSprintB.html)
* focus on bins, genes, and cpg features
