---
title: "rrbsData"
author: "Caitlin Page"
date: "2025-10-08"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: inline
---

## Introduction

```{r}
library(plyranges)
library(stringr)
library(dplyr)
library(tidyr)
library(ggplot2)
library(data.table)
```

```{r}
cg_sites <- cbind(data.frame(Biostrings::matchPattern("CG", BSgenome.Hsapiens.UCSC.hg38::BSgenome.Hsapiens.UCSC.hg38[["chr22"]])), seqnames = "chr22") %>% 
  mutate(pos = paste0(seqnames, "-", start)) %>%
  relocate(pos, seqnames)
cg_sites$index <- 1:nrow(cg_sites)
```

# processing wgbs/rrbs data with Bismark
* process with bismark
[Bismark steps](https://felixkrueger.github.io/Bismark/quick_reference/)
Bismark genome preparation:
** made a genome index folder in /researchers/caitlin.page called Bisulfite_Genome
** aligned the fastq file
** resulting bam stored in /researchers/caitlin.page/pancreatic_cancer_wgbs/bismark_res
** bismark_methylation_extractor

# paper - BSseq (pancreatic cancer)
[GEO](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE63123)
has 19 samples
* 4 individuals, diff replicates and situations
* processed with hg19

all from sample A_38
Lg is lung metastasis
Per is peritoneal metastasis

* SRR1646792: Lg rep 1
* SRR1646793: Lg rep 2
* SRR1646794: Per rep 1
* SRR1646795: Per rep 2

* oops there's 2 more for this sample: 2 liver metastasis: SRR3427332, SRR3427334

```{r}
list.files("/researchers/caitlin.page/pancreatic_cancer_wgbs")[grep("fastq", list.files("/researchers/caitlin.page/pancreatic_cancer_wgbs"))]
```
```{r}
list.files("/researchers/caitlin.page/pancreatic_cancer_wgbs/bismark_res")
```

```{r}
bismark_read_bottom_strand <- fread(paste0("/researchers/caitlin.page/pancreatic_cancer_wgbs/bismark_res/", "CpG_OB_SRR1646792_GSM1541788_38-Lg_rep1_BS-Seq_Homo_sapiens_Bisulfite-Seq_1_bismark_bt2_pe.txt"))
names(bismark_read_bottom_strand) <- c("seq_id", "strand", "seqnames", "start", "meth_status")
```
```{r}
bismark_read_top_strand <- fread(paste0("/researchers/caitlin.page/pancreatic_cancer_wgbs/bismark_res/", "CpG_OT_SRR1646792_GSM1541788_38-Lg_rep1_BS-Seq_Homo_sapiens_Bisulfite-Seq_1_bismark_bt2_pe.txt"))
names(bismark_read_top_strand) <- c("seq_id", "strand", "seqnames", "start", "meth_status")
bismark_read_top_strand[1:10,]
```
```{r}
nrow(bismark_read_top_strand)
```

Z: methylation
z: unmethylated
[source](https://toolshed.g2.bx.psu.edu/repository/display_tool?repository_id=0df2df1072d600d7&tool_config=%2Fsrv%2Ftoolshed-repos%2Fmain%2F000%2Frepo_463%2Fbismark_methylation_extractor.xml&changeset_revision=8c191acde702&render_repository_actions_for=tool_shed)
```{r}
distinct(bismark_read_top_strand, seq_id) %>% nrow() # number of reads
```


```{r}
bismark_read_top_strand <- bismark_read_top_strand %>% filter(seqnames == "chr22")
nrow(bismark_read_top_strand)
nrow(distinct(bismark_read_top_strand, seq_id))
bismark_read_top_strand[1:10,]
```

* sense check positions match those in my cg sites list

```{r}
bismark_read_top_strand$start %in% cg_sites$start %>% summary()
bismark_read_top_strand$start %in% cg_sites$end %>% summary()
```

```{r}
bismark_read_top_strand %>% group_by(seq_id) %>% summarise(num_cpgs_in_read=n()) %>% group_by(num_cpgs_in_read) %>% summarise(n=n()) %>%
  ggplot(aes(x = num_cpgs_in_read, y = n)) +
  geom_point()
```
* comparatively - nanopore has 1500 reads with 4 cpgs
```{r}
bismark_read_top_strand <- bismark_read_top_strand %>% .[order(.$start),]
```

* learn to use data table because dplyr is not memory efficient enough and R will crash
```{r}
bismark_read_top_strand[, meth_status := ifelse(meth_status == "Z", "M", "U")]
bismark_read_top_strand[, meth_pattern := paste0(meth_status, collapse = ""), by = seq_id]
bismark_read_top_strand[, num_meth := str_count(meth_pattern, pattern = "M")]
bismark_read_top_strand[, total := str_length(meth_pattern)]
bismark_read_top_strand[, alpha := num_meth/total]
bismark_read_top_strand[, genom_positions := paste0(start, collapse = ","), by = seq_id]

bismark_read_top_strand <- bismark_read_top_strand %>% .[order(.$seq_id),]
bismark_read_top_strand[1:10,]
```


# other sources of data

[encode rrbs fastqs avail](https://www.encodeproject.org/search/?type=Experiment&control_type%21=%2A&related_series.%40type=ReferenceEpigenome&replicates.library.biosample.donor.organism.scientific_name=Homo+sapiens&status=released&assay_title=RRBS)
* has 13 rrbs data sets

[GM12878 cell line](https://www.encodeproject.org/experiments/ENCSR000DFT/)
* it's a lymphoblastoid cell line
* picked it because it has no cancer or anything else - a few had cancer
* also immune cells - nice and have used before
* also it's blood - which ties in with my cell free analysis

* has 2 technical replicates
* single end this time

```{r}
list.files("/researchers/caitlin.page/rrbs_encode_cell_line") %>% .[grep("fastq", .)]
```

output files from bismark methylation_extractor

```{r}
list.files("/researchers/caitlin.page/rrbs_encode_cell_line") %>% .[grep("txt", .)]
```

```{r}
fread("/researchers/caitlin.page/rrbs_encode_cell_line/CpG_OB_ENCFF000LYP_bismark_bt2.txt")[1:10,]
```

```{r}
source("../code/processData.R")
bis <- fread("/researchers/caitlin.page/rrbs_encode_cell_line/CpG_OT_ENCFF000LYO_bismark_bt2.txt")
process_wgbs(bis) %>% head()
```





